---
title: "OXPHOS KEGG Genes"
author: "John Santiago"
date: "2024-06-22"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r echo = F,include = F}

library(plotly)
library(edgeR)
library(heatmaply)
library(pathview)
library(org.Dm.eg.db)

A4V.cpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.cpmdata.csv", row.names = 1)
A4V.meancpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.meancpmdata.csv", row.names = 1)
A4V.FDR = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.FDRdata.csv", row.names = 1)
A4V.FC = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.FCdata.csv", row.names = 1)
A4V.meta = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.metadata.csv", row.names = 1)

A4V.comparekey = setNames(colnames(A4V.FDR),
                          c('A3FHvS3FH', 'A3MHvS3MH', 'A9FHvS9FH', 'A9MHvS9MH', 'A40FHvS40FH',
                            'A3FHvA9FH', 'S3FHvS9FH', 'A3MHvA9MH', 'S3FHvS9MH',
                            'A9FHvA40FH', 'S9FHvS40FH', 'A3FHvA40FH', 'S3FHvS40FH',
                            
                            'A3FTvS3FT', 'A3MTvS3MT', 'A9FTvS9FT', 'A9MTvS9MT', 'A40FTvS40FT',
                            'A3FTvA9FT', 'S3FTvS9FT', 'A3MTvA9MT', 'S3FTvS9MT',
                            'A9FTvA40FT', 'S9FTvS40FT', 'A3FTvA40FT', 'S3FTvS40FT',
                            
                            'A3FAvS3FA', 'A3MAvS3MA', 'A9FAvS9FA', 'A9MAvS9MA', 'A40FAvS40FA',
                            'A3FAvA9FA', 'S3FAvS9FA', 'A3MAvA9MA', 'S3FAvS9MA',
                            'A9FAvA40FA', 'S9FAvS40FA', 'A3FAvA40FA', 'S3FAvS40FA',
                            
                            
                            'A3FHvA3MH', 'S3FHvS3MH', 'A9FHvA9MH', 'S9FHvS9MH',
                            'A3FTvA3MT', 'S3FTvS3MT', 'A9FTvA9MT', 'S9FTvS9MT',
                            'A3FAvA3MA', 'S3FAvS3MA', 'A9FAvA9MA', 'S9FAvS9MA'))

G85R.cpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT_cpmdata.csv",row.names = 1)
G85R.cpm = G85R.cpm[grep('FBgn', row.names(G85R.cpm)),]
G85R.meancpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT_meancpmdata.csv", row.names = 1)
colnames(G85R.meancpm) = c('GRFC', 'GRMC', 
                           'GRFDf', 'GRMDf', 'WTFDf', 'WTMDf',
                           'GRFOE', 'GRMOE', 'WTFOE', 'WTMOE',
                           'WTFC', 'WTMC')
G85R.FDR = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT.EdgeR.FDRTable.csv",row.names = 1)
G85R.FC = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT.EdgeR.FCTable.csv", row.names = 1)
G85R.meta  = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT.metadata.csv",row.names = 1)

G85R.comparekey = setNames(colnames(G85R.FDR),
                           c('GRFCvGRFDf', 'GRFCvGRFOE', 'WTFCvWTFDf', 'WTFCvWTFOE', 
                             'GRFCvWTFC', 'GRFDfvWTFDf', 'GRFOEvWTFOE',
                             'GRMCvGRMDf', 'GRMCvGRMOE', 'WTMCvWTMDf', 'WTMCvWTMOE', 
                             'GRMCvWTMC', 'GRMDfvWTMFDf', 'GRMOEvWTMOE',
                             'GRFCvGRMC', 'WTFCvWTMC'))

G85R.metab.comparekey = setNames(c("GR.CxDf", "GR.CxOE", "WT.CxDf", "WT.CxOE",
                                   "GRxWT.C", "GRxWT.Df", "GRxWT.OE",
                                   "GR.CxDf", "GR.CxOE", "WT.CxDf", "WT.CxOE",
                                   "GRxWT.C", "GRxWT.Df", "GRxWT.OE"), names(G85R.comparekey)[1:14])

G85R.metabnorm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/NormalizedMetabolomicData.csv", row.names = 1)
G85R.metabmean = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MeanNormalizedMetabolomicData.csv", row.names = 1)
metab.id = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetaboliteMultiKEGGIDKey.csv", row.names = 1)
metab.id = setNames(metab.id$metab, metab.id$KEGG)
G85R.metabFC = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetaboliteFCs.csv", row.names = 1)
G85R.metabFDR = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetaboliteFDRs.csv", row.names = 1)
G85R.metabp = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/Metabolitepvals.csv", row.names = 1)

GeneIDKey = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/GeneIDKey.csv", row.names = 1)
KEGG.MetabKey = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/RawMetabolomicData.csv", row.names = 1)
KEGG.MetabKey = setNames(KEGG.MetabKey[-c(1:2), 2], row.names(KEGG.MetabKey)[-c(1:2)])
genesingo = readRDS("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/genesingo.RData")
GenesInKegg = readRDS("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/kegg.symbol2path.RData")
KEGG.Names = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/KEGG.names.csv", row.names = 1)
GO.Names = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/GO.names.csv", row.names = 1)

git.dir = "https://raw.githubusercontent.com/johncsantiago/WhartonLab/master/"
kegg = readRDS(gzcon(url(paste0(git.dir,"GeneralDataFiles/kegg.symbol2path.RData"))))

nodeID = as.list(org.Dm.egENZYME)


gbbOE = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbOE_CountTable.csv", row.names = 1)
gbbOE.groups = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbOE_Metadata.csv", row.names = 1)
gbbOE = gbbOE[,row.names(gbbOE.groups)]
gbbOE.groups$group = paste0(gbbOE.groups$genotype, "_",gbbOE.groups$stage)

##normalize data
countdata=gbbOE
x <- countdata
gr <- factor(gbbOE.groups$group, levels=c(unique(gbbOE.groups$group)))
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
##normalized cpm
gbbOE.cpmdata=cpm(z)

gbbOEmean = gbbOE.cpmdata[,1:6]
colnames(gbbOEmean) = unique(gr)
i=1
while(i<=6){
  temp= gbbOE.cpmdata[,row.names(gbbOE.groups[gbbOE.groups$group == colnames(gbbOEmean)[i],])]
  gbbOEmean[,i] = apply(temp, 1, mean)
  i=i+1
}


design<-model.matrix(~0 + gr)
colnames(design) = unique(gr)
z = estimateDisp(z, design, robust = T)

fit <- glmQLFit(z, design)
plotQLDisp(fit)

compare = makeContrasts(G85R_LL3-LoxP_LL3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.LL3=DEout
sGRxWT.LL3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_ML3-LoxP_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.ML3=DEout
sGRxWT.ML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_LL3-G85R_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.LL3xML3=DEout
sGR.LL3xML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(LoxP_LL3-LoxP_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
WT.LL3xML3=DEout
sWT.LL3xML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_gbb_LL3-G85R_gbb_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
G85R_gbb.LL3xML3=DEout
sG85R_gbb.LL3xML3=DEout[DEout$FDR<.05,]

gbbKO = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbKO_CountTable.csv", row.names = 1)
gbbKO.groups = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbKO_Metadata.csv", row.names = 1)

##normalize data
countdata=gbbKO
x <- countdata
gr <- factor(gbbKO.groups$Genotype, levels=c(unique(gbbKO.groups$Genotype)))
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
##normalized cpm
gbbKO.cpmdata=cpm(z)

gbbKOmean = gbbKO.cpmdata[,1:2]
colnames(gbbKOmean) = c('WT', 'KO')
i=1
while(i<=2){
  temp= gbbKO.cpmdata[,row.names(gbbKO.groups)[gbbKO.groups$Genotype == colnames(gbbKOmean)[i]]]
  gbbKOmean[,i] = apply(temp, 1, mean)
  i=i+1
}

design<-model.matrix(~0 + gr)
colnames(design) = unique(gr)
z = estimateDisp(z, design, robust = T)

fit <- glmQLFit(z, design)
plotQLDisp(fit)

compare = makeContrasts(KO-WT, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
KOxWT=DEout
sKOxWT=DEout[DEout$FDR<.05,]

  
```


```{r echo = F, fig.height= 12}

NADmetab = names(kegg[grep('00190', kegg)])
NADmetab = row.names(GeneIDKey[GeneIDKey$Symbol %in% NADmetab, ])

geneset = NADmetab

hmdata = A4V.meancpm[geneset,]

A4V.order = c("S3FH", "S9FH", "S40FH", "A3FH", "A9FH", "A40FH",
              "S3FT", "S9FT", "S40FT", "A3FT", "A9FT", "A40FT",
              "S3FA", "S9FA", "S40FA", "A3FA", "A9FA", "A40FA",
              "S3MH", "S9MH", "A3MH", "A9MH",
              "S3MT", "S9MT", "A3MT", "A9MT",
              "S3MA", "S9MA", "A3MA", "A9MA")

hmdata = (hmdata[!is.na((GeneIDKey[row.names(hmdata), 'Symbol'])),A4V.order])
row.names(hmdata) = GeneIDKey[row.names(hmdata), 'Symbol']

heatmaply(hmdata,
          scale = 'row',
          Colv = F,
          cexRow = .5)

```


```{r echo = F, fig.height= 12}

hmdata = G85R.meancpm[geneset,]
G85R.order = c("WTFC", "WTFDf", "WTFOE",
          "GRFC", "GRFDf", "GRFOE",
          "WTMC", "WTMDf", "WTMOE",
          "GRMC", "GRMDf", "GRMOE")

hmdata = (hmdata[!is.na((GeneIDKey[row.names(hmdata), 'Symbol'])),G85R.order])
row.names(hmdata) = GeneIDKey[row.names(hmdata), 'Symbol']

heatmaply(hmdata,
          scale = 'row',
          Colv = F,
          cexRow = .5)

```


```{r echo = F, fig.height= 12}

hmdata = gbbOEmean[intersect(row.names(gbbOEmean),geneset),]
gbbOE.order = c("LoxP_ML3", "LoxP_LL3",
          "G85R_ML3", "G85R_LL3",
          "G85R_gbb_ML3", "G85R_gbb_LL3")

hmdata = (hmdata[!is.na((GeneIDKey[row.names(hmdata), 'Symbol'])),gbbOE.order])
row.names(hmdata) = GeneIDKey[row.names(hmdata), 'Symbol']

heatmaply(hmdata,
          scale = 'row',
          Colv = F,
          cexRow = .5)

```



```{r echo = F, fig.height= 12}

hmdata = gbbKO.cpmdata[intersect(row.names(gbbKOmean),geneset), row.names(gbbKO.groups)]
colnames(hmdata) = c("WT1", "WT2", "WT3", "KO1", "KO2", "KO3")

hmdata = (hmdata[!is.na((GeneIDKey[row.names(hmdata), 'Symbol'])),])
row.names(hmdata) = GeneIDKey[row.names(hmdata), 'Symbol']

heatmaply(hmdata,
          scale = 'row',
          Colv = F,
          cexRow = .5)

```

```{r warning=F,  echo = F, include = F}

##colnames(G85R.FC) = "GRF.CxDf"  "GRF.CxOE"  "WTF.CxDf" "WTF.CxOE"  "GRxWT.FC"  "GRxWT.FDf" "GRxWT.FOE" "GRM.CxDf"  "GRM.CxOE" "WTM.CxDf"  "WTM.CxOE"  "GRxWT.MC" "GRxWT.MDf" "GRxWT.MOE" "GR.FxM" "WT.FxM"   

##colnames(G85R.metabFC) = "KEGG" "GRxWT.C" "GRxWT.Df" "GRxWT.OE" "GR.CxDf"  "WT.CxDf" "GR.CxOE"  "WT.CxOE" 

FC.data = G85R.FC[,c("WTF.CxDf", "WTM.CxDf")]
metab.data = setNames(G85R.metabFC[,"WT.CxDf"], G85R.metabFC$KEGG)
out.id = "WT_ControlxDeficient_FandM_Genes with WT_ControlxDeficient_Metab"



PathwayID = "00190"
sigs = row.names(FC.data)

##File name

OutputDir = "/Users/johncsantiago/Documents/GitHub/WhartonLab/G85R OXPHOS KEGG Pathways/"
setwd(OutputDir)
img.file = paste0(OutputDir, "dme",PathwayID, ".", out.id, ".multi.png")
  
  if(class(FC.data)=="numeric"){
    deg.data=setNames(FC.data[sigs],GeneIDKey[sigs, "ensembl"])
  }
  
  if(class(FC.data)=="data.frame"){
    deg.data = as.matrix(FC.data)
    row.names(deg.data) = GeneIDKey[row.names(FC.data), "ensembl"]
  }
  
  pv.out <- pathview(gene.data =  deg.data,
                     cpd.data = metab.data,
                     pathway.id = PathwayID,
                     species = "dme",
                     kegg.native = T,
                     limit=list(gene=c(-2,2)),
                     node.sum="max.abs",
                     low="royalblue",mid = "grey",high="firebrick",
                     out.suffix = out.id,
                     bins=list(genes=30),
                     plot.col.key = T,
                     match.data = F)
  
  #knitr::include_graphics(img.file)

```

```{r warning=F,  echo = F, include = F}

##EdgeR Comparisons  = GRxWT.ML3 GR.LL3xML3 WT.LL3xML3 G85R_gbb.LL3xML3



FC.data = data.frame(FC1 = GR.LL3xML3[row.names(GRxWT.ML3), "logFC"],
                     FC2 = G85R_gbb.LL3xML3[row.names(GRxWT.ML3), "logFC"])
row.names(FC.data) = row.names(GRxWT.ML3)


out.id = "G85RandG85R.gbbOE_LL3xML3"



PathwayID = "00190"
sigs = row.names(FC.data)

##File name

OutputDir = "/Users/johncsantiago/Documents/GitHub/WhartonLab/G85R OXPHOS KEGG Pathways/"
setwd(OutputDir)
img.file = paste0(OutputDir, "dme",PathwayID, ".", out.id, ".multi.png")

    deg.data = as.matrix(FC.data)
    row.names(deg.data) = GeneIDKey[row.names(FC.data), "ensembl"]
  
  pv.out <- pathview(gene.data =  deg.data,
                     pathway.id = PathwayID,
                     species = "dme",
                     kegg.native = T,
                     limit=list(gene=c(-2,2)),
                     node.sum="max.abs",
                     low="royalblue",mid = "grey",high="firebrick",
                     out.suffix = out.id,
                     bins=list(genes=30),
                     plot.col.key = T,
                     match.data = F)
  
  #knitr::include_graphics(img.file)

```

```{r warning=F,  echo = F, include = F}

##EdgeR Comparisons  = KOxWT



FC.data = setNames(KOxWT$logFC, row.names(KOxWT))

out.id = "gbbKOxgbbWT"



PathwayID = "00190"
sigs = names(FC.data)

##File name

OutputDir = "/Users/johncsantiago/Documents/GitHub/WhartonLab/G85R OXPHOS KEGG Pathways/"
setwd(OutputDir)
img.file = paste0(OutputDir, "dme",PathwayID, ".", out.id, ".png")

    deg.data = as.vector(FC.data)
    names(deg.data) = GeneIDKey[names(FC.data), "ensembl"]
  
  pv.out <- pathview(gene.data =  deg.data,
                     pathway.id = PathwayID,
                     species = "dme",
                     kegg.native = T,
                     limit=list(gene=c(-2,2)),
                     node.sum="max.abs",
                     low="royalblue",mid = "grey",high="firebrick",
                     out.suffix = out.id,
                     bins=list(genes=30),
                     plot.col.key = T,
                     match.data = F)
  
  #knitr::include_graphics(img.file)

```
