---
title: "rac1 gene expression"
author: "John Santiago"
date: "2024-05-24"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




```{r include = F, echo = F}

library(plotly)
library(edgeR)
library(heatmaply)
library(org.Dm.eg.db)
library(KEGGREST)

##load all files from github
git.dir = "https://raw.githubusercontent.com/johncsantiago/WhartonLab/master/"
cpmdata = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT_cpmdata.csv"),row.names = 1)
groups  = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.metadata.csv"),row.names = 1)
TKT.EdgeR = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FDRTable.csv"),row.names = 1)
TKT.EdgeR.FC = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FCTable.csv"),row.names = 1)

groups = groups[colnames(cpmdata),]

gbbOE = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbOE_CountTable.csv", row.names = 1)
gbbOE.groups = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbOE_Metadata.csv", row.names = 1)
gbbOE = gbbOE[,row.names(gbbOE.groups)]
gbbOE.groups$group = paste0(gbbOE.groups$genotype, "_",gbbOE.groups$stage)

##normalize data
countdata=gbbOE
x <- countdata
gr <- factor(gbbOE.groups$group, levels=c(unique(gbbOE.groups$group)))
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
##normalized cpm
gbbOE.cpmdata=cpm(z)

gbbOEmean = gbbOE.cpmdata[,1:6]
colnames(gbbOEmean) = unique(gr)
i=1
while(i<=6){
  temp= gbbOE.cpmdata[,row.names(gbbOE.groups[gbbOE.groups$group == colnames(gbbOEmean)[i],])]
  gbbOEmean[,i] = apply(temp, 1, mean)
  i=i+1
}


design<-model.matrix(~0 + gr)
colnames(design) = unique(gr)
z = estimateDisp(z, design, robust = T)

fit <- glmQLFit(z, design)
plotQLDisp(fit)

compare = makeContrasts(G85R_LL3-LoxP_LL3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.LL3=DEout
sGRxWT.LL3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_ML3-LoxP_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.ML3=DEout
sGRxWT.ML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_LL3-G85R_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.LL3xML3=DEout
sGR.LL3xML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(LoxP_LL3-LoxP_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
WT.LL3xML3=DEout
sWT.LL3xML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_gbb_LL3-G85R_gbb_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
G85R_gbb.LL3xML3=DEout
sG85R_gbb.LL3xML3=DEout[DEout$FDR<.05,]

gbbKO = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbKO_CountTable.csv", row.names = 1)
gbbKO.groups = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbKO_Metadata.csv", row.names = 1)

##normalize data
countdata=gbbKO
x <- countdata
gr <- factor(gbbKO.groups$Genotype, levels=c(unique(gbbKO.groups$Genotype)))
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
##normalized cpm
gbbKO.cpmdata=cpm(z)

GeneIDKey = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/GeneIDKey.csv", row.names = 1)

A4V.cpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.cpmdata.csv",row.names = 1)

A4V.FDR = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.FDRdata.csv")

nodeID = as.list(org.Dm.egENZYME)

```



```{r echo = F}

##subsetting genes for patterns



```


```{r echo = F}

##relish FBgn0014018
##attacin FBgn0041579
##cecropin FBgn0000276
##diptericin FBgn0004240
##drosocin FBgn0010388
##defensin FBgn0010385
##drosomycin FBgn0283461
##metchnikowin FBgn0014865
##sod1 FBgn0003462
##sod2 FBgn0010213
##catalase FBgn0000261
##gstd1 FBgn0001149
## arc1 FBgn0033926
##rac1
##cgs = c('CG10592','CG10827','CG1462','CG16771','CG1809','CG3264','CG3290','CG3292','CG5150','CG5361','CG5656','CG8105','CG8147')
##mle 'FBgn0002774'
##gstD FBgn0001149
##Tudor-SN FBgn0035121
##atf4 FBgn0000370
##thor FBgn0261560
##sdhbl FBgn0030975
##ND-51L FBgn0034251
##kayak (AP-1) FBgn0001297
## Jra (AP-1) FBgn0001291
#sad FBgn0003312

##Specific enzyme
enzyme.ens = nodeID[grep('3.1.3.5',nodeID)]
enzyme.genes = GeneIDKey[GeneIDKey$ensembl %in% names(enzyme.ens)[c(1,3:5,7)],]

```

> GeneIDKey[intersect(s1,intersect(s2,intersect(s3,s4))),'Symbol']
 [1] "Cyp4p1"  "CG13962" "Cpr51A"  "CG11438" "Sgs5bis" "sad"    
 [7] "Lip2"    "CG4757"  "CG14229" "CG15406" "CG3270"  "CG9877" 
[13] "CG13905" "Zip71B"  "CG17027" "nom"     "CG16727" "CG9702" 
[19] "CG43740"
> GeneIDKey[intersect(s1,intersect(s2,intersect(s3,s4))),'FBgn']
 [1] "FBgn0015037" "FBgn0032824" "FBgn0033942" "FBgn0037164"
 [5] "FBgn0038523" "FBgn0003312" "FBgn0024740" "FBgn0027584"
 [9] "FBgn0031059" "FBgn0031517" "FBgn0033093" "FBgn0034819"
[13] "FBgn0035176" "FBgn0036461" "FBgn0036553" "FBgn0037617"
[17] "FBgn0038719" "FBgn0039787" "FBgn0263997"

```{r echo = F}



FBgn = 'FBgn0032513'
name = 'STARD7'
FBgn = 'FBgn0028932'
name = 'CG16890'




gene = data.frame(cpm = as.numeric(cpmdata[FBgn,]), condition = groups$Group, group = paste0(groups$Genotype, groups$Sex))

gene$group = factor(gene$group, levels = c(unique(gene$group)))
gene$condition = factor(gene$condition, levels = c('WT.F', "TktDfWT.F", "TktOEWT.F",
                                                   'GR.F', "TktDfGR.F", "TktOEGR.F",
                                                   'WT.M', "TktDfWT.M", "TktOEWT.M",
                                                   'GR.M', "TktDfGR.M", "TktOEGR.M"))

par(mar = c(5,7,5,2))
plot(x = NA,
      y = NA,
     type = 'n',
     ylim = c(min(gene$cpm), max(gene$cpm)),
     xlim = c(.5, 12.5),
     xaxt = "n",
     ylab="",
     yaxt = "n",
     xlab = NA)

rect(0, 0-(1.5*max(gene$cpm)), 13, 1.5*max(gene$cpm), col = 'gray95')

boxplot(gene$cpm~gene$condition, 
        xlab="",
        las = 2, 
        cex.axis = .8,
        xaxt = 'none',
        yaxt = 'none',
        boxwex = .75,
        boxlwd = 2,
        lwd = 2,
        cex.ticks = 2,
        add = T,
        col = c(rep("honeydew", 3),
                rep("thistle1", 3),
                rep("lightcyan", 3),
                rep("lightyellow", 3)))

box(lwd = 2)

points(x=as.numeric(factor(gene$condition)),
       y=gene$cpm,
       cex=1,
       pch=21,
       bg=c('firebrick',"gold","darkgreen","dodgerblue")[as.numeric(factor(gene$group))])

axis(side = 1,
     at = 1:12,
     labels = rep(c("Control", "Df", "OE"), 4),
     line = 0,
     cex.axis = .75,
     lwd.ticks = 2)

axis(side = 2,
     line = 0,
     cex.axis = 1,
     lwd.ticks = 2,
     las = 2)

axis(side = 2,
     at = .5*(max(gene$cpm) + min(gene$cpm)),
     label = "Counts per million reads",
     tick = F,
     padj = -3,
     cex.axis = 1.25)

axis(side = 1,
     at = c(2, 5, 8, 11),
     labels = c("Silent", "G85R","Silent", "G85R"),
     tick = F,
     padj = 2,
     cex.axis = 1)
axis(side = 3,
     at = c(3.5, 9.5),
     labels = c("Female", "Male"),
     tick = F,
     padj = .5,
     cex.axis = 1.5)

axis(side = 3,
     at = 6.5,
     labels = name,
     tick = F,
     padj = -1.5,
     cex.axis = 2)

axis(side = 3,
     at = c(3.5,9.5),
     tick = T,
     outer = T,
     labels = F,
     line = -2.25,
     lwd.ticks = 0)

xline1 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.04
xline2 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.19
lines(x=c(3.5,3.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(6.5,6.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(9.5,9.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
abline(v = 6.5, lty = 2, lwd = 2)


```


```{r echo = F}

gene = data.frame(cpm = as.numeric(cpmdata[FBgn,]), 
                  condition = groups$Group, 
                  group = paste0(groups$Genotype, groups$Sex),
                  TKT = groups$TKT)

gene = gene[gene$TKT != 'OE',]

gene$group = factor(gene$group, levels = c(unique(gene$group)))
gene$condition = factor(gene$condition, levels = c('GR.F', "TktDfGR.F",
                                                 'WT.F', "TktDfWT.F",
                                                 'GR.M', "TktDfGR.M",
                                                 'WT.M', "TktDfWT.M"))

boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"gold","darkgreen","dodgerblue")[as.numeric(factor(gene$group))])

```


```{r echo = F}

A4V.groups = as.vector(colnames(A4V.cpm))

A4V.groups = substr(A4V.groups,1, nchar(A4V.groups)-1)

par(mar=c(5.1, 4.1, 4.1, 8.1), xpd=TRUE)
gene = data.frame(cpm = as.numeric(A4V.cpm[FBgn,]), 
                  condition = A4V.groups, 
                  group = paste0(substr(A4V.groups, 1,1), substr(A4V.groups, nchar(as.vector(A4V.groups))-1,nchar(as.vector(A4V.groups))-1)))

gene$condition = factor(gene$condition, levels = c("A3FH", "A9FH", "A40FH",
                                                   "S3FH", "S9FH", "S40FH",
                                                   "A3FT", "A9FT", "A40FT",
                                                   "S3FT", "S9FT", "S40FT",
                                                   "A3FA", "A9FA", "A40FA",
                                                   "S3FA", "S9FA", "S40FA",
                                                   "A3MH", "A9MH",
                                                   "S3MH", "S9MH",
                                                   "A3MT", "A9MT",
                                                   "S3MT", "S9MT",
                                                   "A3MA", "A9MA",
                                                   "S3MA", "S9MA"))
gene$group = factor(gene$group, levels = c(unique(gene$group)))

boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"gold","darkgreen","dodgerblue")[as.numeric(factor(gene$group))])
legend('topright',
       inset=c(-0.275,0),
       legend = c('A4V F', 'Silent F', 'A4V M', 'Silent F'), 
       fill = c('firebrick',"darkgreen", "gold","dodgerblue"),
       cex = .65,
       bty = 'n',
       pt.cex = .5)


```

```{r}

gene = data.frame(cpm = as.numeric(gbbOE.cpmdata[FBgn,]), condition = gbbOE.groups$group, group = gbbOE.groups$genotype)

gene$group = factor(gene$group, levels = c(unique(gene$group)))
gene$condition = factor(gene$condition, levels = c('LoxP_ML3', "LoxP_LL3",
                                                 'G85R_ML3', "G85R_LL3",
                                                 'G85R_gbb_ML3', "G85R_gbb_LL3"))

boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"dodgerblue","gold")[as.numeric(factor(gene$group))])

```


```{r}

## WT = W1118
## KO = GBB1/GBB2 (both null alleles)
## Aaron and Catherine legend for RNAseq
## Whole larvae

gene = data.frame(cpm = as.numeric(gbbKO.cpmdata[FBgn,]), condition = gbbKO.groups$Genotype)

gene$condition = factor(gene$condition, levels = c('WT', "KO"))

boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"dodgerblue","gold")[as.numeric(factor(gene$condition))])

```




```{r echo = F}


rawdata = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetabolomicsForMetaboanalyst.txt", sep = "\t", row.names = 1)
KEGG.Key = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/RawMetabolomicData.csv", row.names = 1)
KEGG.Key = setNames(KEGG.Key[-c(1:2), 2], row.names(KEGG.Key)[-c(1:2)])

genotypes = setNames(c("WT", "GR", "TktOEWT", "TktOEGR", "TktDfWT", "TktDfGR"), c("C", "E", "A", "F", "B", "D"))

metadata = data.frame(Sample = colnames(rawdata), 
                      Group = as.character(rawdata[1,]), 
                      TIC = as.numeric(rawdata[2,]), 
                      Genotypes = genotypes[as.character(rawdata[1,])])

row.names(metadata) = metadata$Sample

norm.func = function(data){
  nd = (as.numeric(data)/metadata$TIC)*1000
  return(nd)
}

norm.data = t(apply(rawdata[3:nrow(rawdata),], 1, norm.func))
colnames(norm.data) = colnames(rawdata)

group.mean=function(data){
  gm = mean(na.omit(data))
}

mean.data = matrix(0, nrow=nrow(norm.data), ncol = length(genotypes))
row.names(mean.data) = row.names(norm.data)
colnames(mean.data) = unique(metadata$Group)

i=1
while(i<=ncol(mean.data)){
  mean.data[,i] = apply(norm.data[,metadata[metadata$Group == colnames(mean.data)[i], "Sample"]], 1, group.mean)
  i = i+1
}

mean.metab = mean.data

compare.conditions = function(condition1, condition2){
  columns1 = metadata[metadata$Group == condition1, "Sample"]
  columns2 = metadata[metadata$Group == condition2, "Sample"]
  p=setNames(rep(NA, nrow(norm.data)), row.names(norm.data))
  i=1
  while(i<=length(p)){
    if(length(na.omit(norm.data[i,columns1]))>2 & 
       length(na.omit(norm.data[i,columns2]))>2){
      temp = t.test(na.omit(norm.data[i,columns1]),
                    na.omit(norm.data[i,columns2]), paired = F, var.equal = T)
      p[i] = temp[[3]]
      }
    i=i+1
    }

  fdr = p.adjust(p, "BH", length(p))

  fc = mean.metab[,condition1]/mean.metab[,condition2]

  comparison.table = data.frame(p = p, 
                                FDR = fdr, 
                                FC = fc, 
                                KEGG = KEGG.Key[row.names(mean.metab)])
  
  comparison.table = comparison.table[order(comparison.table$FDR),]
}

GRxWT.C = compare.conditions("E", "C")
GRxWT.Df = compare.conditions("D", "B")[row.names(GRxWT.C),]
GRxWT.OE = compare.conditions("F", "A")[row.names(GRxWT.C),]

GR.CxDf = compare.conditions("D", "E")[row.names(GRxWT.C),]
WT.CxDf = compare.conditions("B", "C")[row.names(GRxWT.C),]

GR.CxOE = compare.conditions("F", "E")[row.names(GRxWT.C),]
WT.CxOE = compare.conditions("A", "C")[row.names(GRxWT.C),]

GR.DfxWT.C = compare.conditions("D", "C")[row.names(GRxWT.C),]
GR.OExWT.C = compare.conditions("F", "C")[row.names(GRxWT.C),]



comparison.means = list(GRF.CxDf = c("TktDfGR.F","GR.F"), 
                 GRF.CxOE = c("TktOEGR.F","GR.F"),
                 WTF.CxDf = c("TktDfWT.F","WT.F"),
                 WTF.CxOE = c("TktOEWT.F","WT.F"),
                 GRxWT.FC = c("GR.F","WT.F"), 
                 GRxWT.FDf = c("TktDfGR.F","TktDfWT.F"),
                 GRxWT.FOE = c("TktOEGR.F","TktOEWT.F"), 
                 GRM.CxDf = c("TktDfGR.M","GR.M"), 
                 GRM.CxOE = c("TktOEGR.M","GR.M"),
                 WTM.CxDf = c("TktDfWT.M","WT.M"),
                 WTM.CxOE = c("TktOEWT.M","WT.M"),
                 GRxWT.MC = c("GR.M","WT.M"), 
                 GRxWT.MDf = c("TktDfGR.M","TktDfWT.M"),
                 GRxWT.MOE = c("TktOEGR.M","TktOEWT.M"),
                 GR.FxM = c("GR.F","GR.M"),
                 WT.FxM = c("WT.F","WT.M"))


metab.FC = data.frame('GRxWT.C' = GRxWT.C$FC,
                      'GRxWT.Df' = GRxWT.Df$FC,
                      'GRxWT.OE' = GRxWT.OE$FC,
                      'GR.CxDf' = GR.CxDf$FC,
                      'WT.CxDf' = WT.CxDf$FC,
                      'GR.CxOE' = GR.CxOE$FC,
                      'WT.CxOE' = WT.CxOE$FC,
                      'GR.DfxWT.C' = GR.DfxWT.C$FC,
                      'GR.OExWT.C' = GR.OExWT.C$FC)
row.names(metab.FC) = row.names(GRxWT.C)
metab.FC = log2(metab.FC)

metab.FDR = data.frame(GRxWT.C = GRxWT.C$p,
                       GRxWT.Df = GRxWT.Df$p,
                       GRxWT.OE = GRxWT.OE$p,
                       GR.CxDf = GR.CxDf$p,
                       WT.CxDf = WT.CxDf$p,
                       GR.CxOE = GR.CxOE$p,
                       WT.CxOE = WT.CxOE$p,
                       GR.DfxWT.C = GR.DfxWT.C$p,
                       GR.OExWT.C = GR.OExWT.C$p)
row.names(metab.FDR) = row.names(GRxWT.C)

colnames(mean.metab) = as.character(genotypes[colnames(mean.metab)])

colnames(norm.data) = as.character(genotypes[substr(colnames(norm.data),1,1)])


```

```{r echo = F}



metab = 'D-sedoheptulose-7-phosphate'
name = 'D-sedoheptulose-7-phosphate'



metab = data.frame(data = as.numeric(norm.data[metab,]), 
                   group = colnames(norm.data), 
                   geno = colnames(norm.data))


metab$group = factor(metab$group, levels = c("WT", "TktDfWT", "TktOEWT",
                                             "GR", "TktDfGR", "TktOEGR"))

metab$geno = factor(substr(metab$group, (nchar(as.character(metab$group)) -1), nchar(as.character(metab$group))), 
                     levels = c("WT", "GR"))

par(mar = c(5,7,5,2))
plot(x = NA,
      y = NA,
     type = 'n',
     ylim = c(min(metab$data), max(metab$data)),
     xlim = c(.5, 6.5),
     xaxt = "n",
     ylab="",
     yaxt = "n",
     xlab = NA)

rect(0, 0-(1.5*max(gene$cpm)), 13, 1.5*max(gene$cpm), col = 'gray95')

boxplot(metab$data~metab$group, 
        xlab="",
        las = 2, 
        cex.axis = .8,
        xaxt = 'none',
        yaxt = 'none',
        boxwex = .75,
        boxlwd = 2,
        lwd = 2,
        cex.ticks = 2,
        add = T,
        col = c(rep("lightyellow", 3),
                rep("lightgreen", 3)))

box(lwd = 2)

points(x=as.numeric(factor(metab$group)),
       y=metab$data,
       cex=1.25,
       pch=21,
       bg=c('gold3',"darkgreen")[as.numeric(factor(metab$geno))])

axis(side = 1,
     at = 1:12,
     labels = rep(c("Control", "Df", "OE"), 4),
     line = 0,
     cex.axis = .75,
     lwd.ticks = 2)

axis(side = 2,
     line = 0,
     cex.axis = 1,
     lwd.ticks = 2,
     las = 2)

axis(side = 2,
     at = .5*(max(gene$cpm) + min(gene$cpm)),
     label = "Counts per million reads",
     tick = F,
     padj = -3,
     cex.axis = 1.25)

axis(side = 1,
     at = c(2, 5, 8, 11),
     labels = c("Silent", "G85R","Silent", "G85R"),
     tick = F,
     padj = 2,
     cex.axis = 1)

axis(side = 3,
     at = 3.5,
     labels = name,
     tick = F,
     padj = -.5,
     cex.axis = 2)


xline1 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.04
xline2 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.19
lines(x=c(3.5,3.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(6.5,6.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(9.5,9.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)



```