---
title: "Specifgic Gene or Metabolite Levels"
author: "John Santiago"
date: "2024-05-24"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




```{r include = F, echo = F}
##Load Libraries and Data

library(plotly)
library(edgeR)

GeneIDKey = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/GeneIDKey.csv", row.names = 1)

##load all files from github
git.dir = "https://raw.githubusercontent.com/johncsantiago/WhartonLab/master/"

TKT.cpm = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT_cpmdata.csv"),row.names = 1)

TKT.groups  = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.metadata.csv"),row.names = 1)
TKT.FDR = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FDRTable.csv"),row.names = 1)
TKT.FC = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FCTable.csv"),row.names = 1)

TKT.groups = TKT.groups[colnames(TKT.cpm),]

A4V.cpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.cpmdata.csv",row.names = 1)

A4V.FDR = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/A4V_BodySections_RNAseq/A4V.FDRdata.csv")




gbbOE = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbOE_CountTable.csv", row.names = 1)
gbbOE.groups = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbOE_Metadata.csv", row.names = 1)
gbbOE = gbbOE[,row.names(gbbOE.groups)]
gbbOE.groups$group = paste0(gbbOE.groups$genotype, "_",gbbOE.groups$stage)

##normalize data
countdata=gbbOE
x <- countdata
gr <- factor(gbbOE.groups$group, levels=c(unique(gbbOE.groups$group)))
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
##normalized cpm
gbbOE.cpmdata=cpm(z)

gbbOEmean = gbbOE.cpmdata[,1:6]
colnames(gbbOEmean) = unique(gr)
i=1
while(i<=6){
  temp= gbbOE.cpmdata[,row.names(gbbOE.groups[gbbOE.groups$group == colnames(gbbOEmean)[i],])]
  gbbOEmean[,i] = apply(temp, 1, mean)
  i=i+1
}


design<-model.matrix(~0 + gr)
colnames(design) = unique(gr)
z = estimateDisp(z, design, robust = T)

fit <- glmQLFit(z, design)
plotQLDisp(fit)

compare = makeContrasts(G85R_LL3-LoxP_LL3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.LL3=DEout
sGRxWT.LL3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_ML3-LoxP_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.ML3=DEout
sGRxWT.ML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_LL3-G85R_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.LL3xML3=DEout
sGR.LL3xML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(LoxP_LL3-LoxP_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
WT.LL3xML3=DEout
sWT.LL3xML3=DEout[DEout$FDR<.05,]

compare = makeContrasts(G85R_gbb_LL3-G85R_gbb_ML3, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
G85R_gbb.LL3xML3=DEout
sG85R_gbb.LL3xML3=DEout[DEout$FDR<.05,]

gbbKO = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbKO_CountTable.csv", row.names = 1)
gbbKO.groups = read.csv("/Users/johncsantiago/Documents/GitHub/Datasets/gbbKO_Metadata.csv", row.names = 1)

##normalize data
countdata=gbbKO
x <- countdata
gr <- factor(gbbKO.groups$Genotype, levels=c(unique(gbbKO.groups$Genotype)))
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
##normalized cpm
gbbKO.cpmdata=cpm(z)

rawdata = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetabolomicsForMetaboanalyst.txt", sep = "\t", row.names = 1)
KEGG.Key = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/RawMetabolomicData.csv", row.names = 1)
KEGG.Key = setNames(KEGG.Key[-c(1:2), 2], row.names(KEGG.Key)[-c(1:2)])

genotypes = setNames(c("WT", "GR", "TktOEWT", "TktOEGR", "TktDfWT", "TktDfGR"), c("C", "E", "A", "F", "B", "D"))

metadata = data.frame(Sample = colnames(rawdata), 
                      Group = as.character(rawdata[1,]), 
                      TIC = as.numeric(rawdata[2,]), 
                      Genotypes = genotypes[as.character(rawdata[1,])])

row.names(metadata) = metadata$Sample

norm.func = function(data){
  nd = (as.numeric(data)/metadata$TIC)*1000
  return(nd)
}

norm.data = t(apply(rawdata[3:nrow(rawdata),], 1, norm.func))
colnames(norm.data) = colnames(rawdata)

group.mean=function(data){
  gm = mean(na.omit(data))
}

mean.data = matrix(0, nrow=nrow(norm.data), ncol = length(genotypes))
row.names(mean.data) = row.names(norm.data)
colnames(mean.data) = unique(metadata$Group)

i=1
while(i<=ncol(mean.data)){
  mean.data[,i] = apply(norm.data[,metadata[metadata$Group == colnames(mean.data)[i], "Sample"]], 1, group.mean)
  i = i+1
}

mean.metab = mean.data

compare.conditions = function(condition1, condition2){
  columns1 = metadata[metadata$Group == condition1, "Sample"]
  columns2 = metadata[metadata$Group == condition2, "Sample"]
  p=setNames(rep(NA, nrow(norm.data)), row.names(norm.data))
  i=1
  while(i<=length(p)){
    if(length(na.omit(norm.data[i,columns1]))>2 & 
       length(na.omit(norm.data[i,columns2]))>2){
      temp = t.test(na.omit(norm.data[i,columns1]),
                    na.omit(norm.data[i,columns2]), paired = F, var.equal = T)
      p[i] = temp[[3]]
      }
    i=i+1
    }

  fdr = p.adjust(p, "BH", length(p))

  fc = mean.metab[,condition1]/mean.metab[,condition2]

  comparison.table = data.frame(p = p, 
                                FDR = fdr, 
                                FC = fc, 
                                KEGG = KEGG.Key[row.names(mean.metab)])
  
  comparison.table = comparison.table[order(comparison.table$FDR),]
}

GRxWT.C = compare.conditions("E", "C")
GRxWT.Df = compare.conditions("D", "B")[row.names(GRxWT.C),]
GRxWT.OE = compare.conditions("F", "A")[row.names(GRxWT.C),]

GR.CxDf = compare.conditions("D", "E")[row.names(GRxWT.C),]
WT.CxDf = compare.conditions("B", "C")[row.names(GRxWT.C),]

GR.CxOE = compare.conditions("F", "E")[row.names(GRxWT.C),]
WT.CxOE = compare.conditions("A", "C")[row.names(GRxWT.C),]

GR.DfxWT.C = compare.conditions("D", "C")[row.names(GRxWT.C),]
GR.OExWT.C = compare.conditions("F", "C")[row.names(GRxWT.C),]



comparison.means = list(GRF.CxDf = c("TktDfGR.F","GR.F"), 
                 GRF.CxOE = c("TktOEGR.F","GR.F"),
                 WTF.CxDf = c("TktDfWT.F","WT.F"),
                 WTF.CxOE = c("TktOEWT.F","WT.F"),
                 GRxWT.FC = c("GR.F","WT.F"), 
                 GRxWT.FDf = c("TktDfGR.F","TktDfWT.F"),
                 GRxWT.FOE = c("TktOEGR.F","TktOEWT.F"), 
                 GRM.CxDf = c("TktDfGR.M","GR.M"), 
                 GRM.CxOE = c("TktOEGR.M","GR.M"),
                 WTM.CxDf = c("TktDfWT.M","WT.M"),
                 WTM.CxOE = c("TktOEWT.M","WT.M"),
                 GRxWT.MC = c("GR.M","WT.M"), 
                 GRxWT.MDf = c("TktDfGR.M","TktDfWT.M"),
                 GRxWT.MOE = c("TktOEGR.M","TktOEWT.M"),
                 GR.FxM = c("GR.F","GR.M"),
                 WT.FxM = c("WT.F","WT.M"))


metab.FC = data.frame('GRxWT.C' = GRxWT.C$FC,
                      'GRxWT.Df' = GRxWT.Df$FC,
                      'GRxWT.OE' = GRxWT.OE$FC,
                      'GR.CxDf' = GR.CxDf$FC,
                      'WT.CxDf' = WT.CxDf$FC,
                      'GR.CxOE' = GR.CxOE$FC,
                      'WT.CxOE' = WT.CxOE$FC,
                      'GR.DfxWT.C' = GR.DfxWT.C$FC,
                      'GR.OExWT.C' = GR.OExWT.C$FC)
row.names(metab.FC) = row.names(GRxWT.C)
metab.FC = log2(metab.FC)

metab.FDR = data.frame(GRxWT.C = GRxWT.C$p,
                       GRxWT.Df = GRxWT.Df$p,
                       GRxWT.OE = GRxWT.OE$p,
                       GR.CxDf = GR.CxDf$p,
                       WT.CxDf = WT.CxDf$p,
                       GR.CxOE = GR.CxOE$p,
                       WT.CxOE = WT.CxOE$p,
                       GR.DfxWT.C = GR.DfxWT.C$p,
                       GR.OExWT.C = GR.OExWT.C$p)
row.names(metab.FDR) = row.names(GRxWT.C)

colnames(mean.metab) = as.character(genotypes[colnames(mean.metab)])

colnames(norm.data) = as.character(genotypes[substr(colnames(norm.data),1,1)])


```


```{r echo = F}
##Function to plot all G85R groups

plot.G85R = function(ID){
ID.row = GeneIDKey[grep(ID, GeneIDKey$FBgn),]


if(nrow(ID.row) == 1){
FBgn = ID.row$FBgn
name = ID.row$Symbol


gene = data.frame(cpm = as.numeric(TKT.cpm[FBgn,]), condition = TKT.groups$Group, group = paste0(TKT.groups$Genotype, TKT.groups$Sex))

gene$group = factor(gene$group, levels = c(unique(gene$group)))
gene$condition = factor(gene$condition, levels = c('WT.F', "TktDfWT.F", "TktOEWT.F",
                                                   'GR.F', "TktDfGR.F", "TktOEGR.F",
                                                   'WT.M', "TktDfWT.M", "TktOEWT.M",
                                                   'GR.M', "TktDfGR.M", "TktOEGR.M"))

par(mar = c(5,7,5,2))
plot(x = NA,
      y = NA,
     type = 'n',
     ylim = c(min(gene$cpm), max(gene$cpm)),
     xlim = c(.5, 12.5),
     xaxt = "n",
     ylab="",
     yaxt = "n",
     xlab = NA)

rect(0, 0-(1.5*max(gene$cpm)), 13, 1.5*max(gene$cpm), col = 'gray95')

boxplot(gene$cpm~gene$condition, 
        xlab="",
        las = 2, 
        cex.axis = .8,
        xaxt = 'none',
        yaxt = 'none',
        boxwex = .75,
        boxlwd = 2,
        lwd = 2,
        cex.ticks = 2,
        add = T,
        col = c(rep("honeydew", 3),
                rep("thistle1", 3),
                rep("lightcyan", 3),
                rep("lightyellow", 3)))

box(lwd = 2)

points(x=as.numeric(factor(gene$condition)),
       y=gene$cpm,
       cex=1,
       pch=21,
       bg=c('firebrick',"gold","darkgreen","dodgerblue")[as.numeric(factor(gene$group))])

axis(side = 1,
     at = 1:12,
     labels = rep(c("Control", "Df", "OE"), 4),
     line = 0,
     cex.axis = .75,
     lwd.ticks = 2)

axis(side = 2,
     line = 0,
     cex.axis = 1,
     lwd.ticks = 2,
     las = 2)

axis(side = 2,
     at = .5*(max(gene$cpm) + min(gene$cpm)),
     label = "Counts per million reads",
     tick = F,
     padj = -3,
     cex.axis = 1.25)

axis(side = 1,
     at = c(2, 5, 8, 11),
     labels = c("Silent", "G85R","Silent", "G85R"),
     tick = F,
     padj = 2,
     cex.axis = 1)
axis(side = 3,
     at = c(3.5, 9.5),
     labels = c("Female", "Male"),
     tick = F,
     padj = .5,
     cex.axis = 1.5)

axis(side = 3,
     at = 6.5,
     labels = name,
     tick = F,
     padj = -1.5,
     cex.axis = 2)

axis(side = 3,
     at = c(3.5,9.5),
     tick = T,
     outer = T,
     labels = F,
     line = -2.25,
     lwd.ticks = 0)

xline1 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.04
xline2 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.19
lines(x=c(3.5,3.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(6.5,6.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(9.5,9.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
abline(v = 6.5, lty = 2, lwd = 2)
}

if(nrow(ID.row) != 1){
  ID.row
}
}

```

```{r echo = F}
##Function to plot a specific gene expression levels in all A4V groups

  
plot.A4V = function(ID){
ID.row = GeneIDKey[grep(ID, GeneIDKey$FBgn),]

if(nrow(ID.row) == 1){
  FBgn = ID.row$FBgn
  name = ID.row$Symbol
  A4V.groups = as.vector(colnames(A4V.cpm))

  A4V.groups = substr(A4V.groups,1, nchar(A4V.groups)-1)

  par(mar=c(5.1, 4.1, 4.1, 8.1), xpd=TRUE)
  gene = data.frame(cpm = as.numeric(A4V.cpm[FBgn,]),
                    condition = A4V.groups,
                    group = paste0(substr(A4V.groups, 1,1), substr(A4V.groups, start = nchar(as.vector(A4V.groups))-1,nchar(as.vector(A4V.groups))-1)))

  gene$condition = factor(gene$condition, levels = c("A3FH", "A9FH", "A40FH",
                                                   "S3FH", "S9FH", "S40FH",
                                                   "A3FT", "A9FT", "A40FT",
                                                   "S3FT", "S9FT", "S40FT",
                                                   "A3FA", "A9FA", "A40FA",
                                                   "S3FA", "S9FA", "S40FA",
                                                   "A3MH", "A9MH",
                                                   "S3MH", "S9MH",
                                                   "A3MT", "A9MT",
                                                   "S3MT", "S9MT",
                                                   "A3MA", "A9MA",
                                                   "S3MA", "S9MA"))
  gene$group = factor(gene$group, levels = c(unique(gene$group)))

  boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
  points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"gold","darkgreen","dodgerblue")[as.numeric(factor(gene$group))])
  legend('topright',
       inset=c(-0.275,0),
       legend = c('A4V F', 'Silent F', 'A4V M', 'Silent F'), 
       fill = c('firebrick',"darkgreen", "gold","dodgerblue"),
       cex = .65,
       bty = 'n',
       pt.cex = .5)
}

if(nrow(ID.row) != 1){
  ID.row
}
}

```

```{r echo = F}
##Plot levels of a specific metabolite

plot.metab = function(metabID){

ID.row = row.names(norm.data)[grep(ID, row.names(norm.data))]


if(length(ID.row) == 1){
metab = ID.row
name = ID.row

metab = data.frame(data = as.numeric(norm.data[metab,]), 
                   group = colnames(norm.data), 
                   geno = colnames(norm.data))


metab$group = factor(metab$group, levels = c("WT", "TktDfWT", "TktOEWT",
                                             "GR", "TktDfGR", "TktOEGR"))

metab$geno = factor(substr(metab$group, (nchar(as.character(metab$group)) -1), nchar(as.character(metab$group))), 
                     levels = c("WT", "GR"))

par(mar = c(5,7,5,2))
plot(x = NA,
      y = NA,
     type = 'n',
     ylim = c(min(na.omit(metab$data)), max(na.omit(metab$data))),
     xlim = c(.5, 6.5),
     xaxt = "n",
     ylab="",
     yaxt = "n",
     xlab = NA)

rect(0, 0-(1.5*max(gene$cpm)), 13, 1.5*max(gene$cpm), col = 'gray95')

boxplot(metab$data~metab$group, 
        xlab="",
        las = 2, 
        cex.axis = .8,
        xaxt = 'none',
        yaxt = 'none',
        boxwex = .75,
        boxlwd = 2,
        lwd = 2,
        cex.ticks = 2,
        add = T,
        col = c(rep("lightyellow", 3),
                rep("lightgreen", 3)))

box(lwd = 2)

points(x=as.numeric(factor(metab$group)),
       y=metab$data,
       cex=1.25,
       pch=21,
       bg=c('gold3',"darkgreen")[as.numeric(factor(metab$geno))])

axis(side = 1,
     at = 1:12,
     labels = rep(c("Control", "Df", "OE"), 4),
     line = 0,
     cex.axis = .75,
     lwd.ticks = 2)

axis(side = 2,
     line = 0,
     cex.axis = 1,
     lwd.ticks = 2,
     las = 2)

axis(side = 1,
     at = c(2, 5, 8, 11),
     labels = c("Silent", "G85R","Silent", "G85R"),
     tick = F,
     padj = 2,
     cex.axis = 1)

axis(side = 3,
     at = 3.5,
     labels = name,
     tick = F,
     padj = -.5,
     cex.axis = 2)

}

if(length(ID.row) != 1){
  ID.row
}

}

```


```{r echo = F}

ID = "0000018"
plot.G85R(ID)

```


```{r echo = F}
##Tailared code to plot a specific gene expression levels in all A4V groups

ID = "0000018"
  
plot.A4V(ID)
```


```{r echo = F}

metabID = 'NADP'

plot.metab(metabID)

```

```{r echo = F}
##Tailared code to plot a subset of groups in the TKT dataset

##ID = "0000018"
  
ID.row = GeneIDKey[grep(ID, GeneIDKey$FBgn),]

if(nrow(ID.row) == 1){
FBgn = ID.row$FBgn
name = ID.row$Symbol

gene = data.frame(cpm = as.numeric(TKT.cpm[FBgn,]), 
                  condition = TKT.groups$Group, 
                  group = paste0(TKT.groups$Genotype, TKT.groups$Sex),
                  TKT = TKT.groups$TKT)

gene = gene[gene$TKT != 'OE',]

gene$group = factor(gene$group, levels = c(unique(gene$group)))
gene$condition = factor(gene$condition, levels = c('GR.F', "TktDfGR.F",
                                                 'WT.F', "TktDfWT.F",
                                                 'GR.M', "TktDfGR.M",
                                                 'WT.M', "TktDfWT.M"))

boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"gold","darkgreen","dodgerblue")[as.numeric(factor(gene$group))])

}

if(nrow(ID.row) != 1){
  ID.row
}

```



```{r echo = F}
##Plot expression levels for a specific gene in the gbbOE dataset

##ID = "0000018"
  
ID.row = GeneIDKey[grep(ID, GeneIDKey$FBgn),]

if(nrow(ID.row) == 1){
FBgn = ID.row$FBgn
name = ID.row$Symbol

gene = data.frame(cpm = as.numeric(gbbOE.cpmdata[FBgn,]), condition = gbbOE.groups$group, group = gbbOE.groups$genotype)

gene$group = factor(gene$group, levels = c(unique(gene$group)))
gene$condition = factor(gene$condition, levels = c('LoxP_ML3', "LoxP_LL3",
                                                 'G85R_ML3', "G85R_LL3",
                                                 'G85R_gbb_ML3', "G85R_gbb_LL3"))

boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"dodgerblue","gold")[as.numeric(factor(gene$group))])

}

if(nrow(ID.row) != 1){
  ID.row
}
```


```{r echo = F}
##Plot expression levels for a specific gene in the gbb-null dataset


## WT = W1118
## KO = GBB1/GBB2 (both null alleles)
## Aaron and Catherine legend for RNAseq
## Whole larvae

##ID = "0000018"
  
ID.row = GeneIDKey[grep(ID, GeneIDKey$FBgn),]

if(nrow(ID.row) == 1){
FBgn = ID.row$FBgn
name = ID.row$Symbol

gene = data.frame(cpm = as.numeric(gbbKO.cpmdata[FBgn,]), condition = gbbKO.groups$Genotype)

gene$condition = factor(gene$condition, levels = c('WT', "KO"))

boxplot(gene$cpm~gene$condition, xlab="",     
        main= name,ylab="Counts per Million Reads", las = 2, cex.axis = .8)
points(x=as.numeric(factor(gene$condition)),y=gene$cpm,cex=1,pch=21,bg=c('firebrick',"dodgerblue","gold")[as.numeric(factor(gene$condition))])

}

if(nrow(ID.row) != 1){
  ID.row
}
```



