---
title: "TKT RNAseq Experiment: Differential Expression Analysis"
author: "John Santiago"
date: "2022-03-23"
output: 
  bookdown::html_document2:
    number_sections: false
---

```{r setup, include=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```


## [Return to Home Page](https://johncsantiago.github.io/TKT_RNAseq_preliminary.html)

<br /><br />
\newpage

## <span style="text-decoration:underline">Table of Contents</span> 


- #### Figure \@ref(fig:fig1): MDS plot

- #### Figure \@ref(fig:fig2): Barplot of Total DEGs

### TKT effect in female G85R
- #### Figure \@ref(fig:fig3): Heatmaps of Total DEGs in GR.FC x GR.FDf

- #### Figure \@ref(fig:fig4): Heatmaps of Total DEGs in GR.FC x GR.FOE

### TKT effect in female WT
- #### Figure \@ref(fig:fig5): Heatmaps of Total DEGs in WT.FC x WT.FDf

- #### Figure \@ref(fig:fig6): Heatmaps of Total DEGs in WT.FC x WT.FOE

### Genotype effect between female G85R and WT
- #### Figure \@ref(fig:fig7): Heatmaps of Total DEGs in GR.FC x WT.FC

- #### Figure \@ref(fig:fig8): Heatmaps of Total DEGs in GR.FDf x WT.FDf

- #### Figure \@ref(fig:fig9): Heatmaps of Total DEGs in GR.FOE x WT.FOE

### TKT effect in Male G85R
- #### Figure \@ref(fig:fig10): Heatmaps of Total DEGs in GR.MC x GR.MDf

- #### Figure \@ref(fig:fig11): Heatmaps of Total DEGs in GR.MC x GR.MOE

### TKT effect in Male WT
- #### Figure \@ref(fig:fig12): Heatmaps of Total DEGs in WT.MC x WT.MDf

- #### Figure \@ref(fig:fig13): Heatmaps of Total DEGs in WT.MC x WT.MOE

### Genotype effect between Male G85R and WT
- #### Figure \@ref(fig:fig14): Heatmaps of Total DEGs in GR.MC x WT.MC

- #### Figure \@ref(fig:fig15): Heatmaps of Total DEGs in GR.MDf x WT.MDf

- #### Figure \@ref(fig:fig16): Heatmaps of Total DEGs in GR.MOE x WT.MOE

<br /><br /><br />
\newpage

```{r echo=F, include=FALSE}

library(edgeR)
library(plotly)
library(heatmaply)

##Clean Trait Data and calculate CPMs
countdata=read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT.counttable.csv",row.names=1)
groups=read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT.metadata.csv",row.names=1)

convert=read.csv("https://raw.githubusercontent.com/DavidRandLab/Santiago-et-al-2021-BMC-Genomics/main/Data%20Files/FBgnConversionTable.csv",row.names=1)

##normalize data
countdata=countdata[, row.names(groups)]
x <- countdata
gr <- factor(groups$Group)
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
##normalized cpm
cpmdata=cpm(z)

design<-model.matrix(~0 + gr)
colnames(design) = unique(gr)
z = estimateDisp(z, design, robust = T)

fit <- glmQLFit(z, design)
plotQLDisp(fit)

##Will calculate fold-change with the first over the second

##GR.F TKT control vs TKT altered
compare = makeContrasts(TktDfGR.F-GR.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.F.CxDf=DEout
sGR.F.CxDf=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktOEGR.F-GR.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.F.CxOE=DEout
sGR.F.CxOE=DEout[DEout$FDR<.05,]

##WT.F TKT control vs TKT altered
compare = makeContrasts(TktDfWT.F-WT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
WT.F.CxDf=DEout
sWT.F.CxDf=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktOEWT.F-WT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
WT.F.CxOE=DEout
sWT.F.CxOE=DEout[DEout$FDR<.05,]


##WT.F vs GR.F
compare = makeContrasts(GR.F-WT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.F=DEout
sGRxWT.F=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktDfGR.F-TktDfWT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.FDf=DEout
sGRxWT.FDf=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktOEGR.F-TktOEWT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.FOE=DEout
sGRxWT.FOE=DEout[DEout$FDR<.05,]



##GR.M TKT control vs TKT altered
compare = makeContrasts(TktDfGR.M-GR.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.M.CxDf=DEout
sGR.M.CxDf=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktOEGR.M-GR.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.M.CxOE=DEout
sGR.M.CxOE=DEout[DEout$FDR<.05,]

##WT.M TKT control vs TKT altered
compare = makeContrasts(TktDfWT.M-WT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
WT.M.CxDf=DEout
sWT.M.CxDf=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktOEWT.M-WT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
WT.M.CxOE=DEout
sWT.M.CxOE=DEout[DEout$FDR<.05,]


##WT.M vs GR.M
compare = makeContrasts(GR.M-WT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.M=DEout
sGRxWT.M=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktDfGR.M-TktDfWT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.MDf=DEout
sGRxWT.MDf=DEout[DEout$FDR<.05,]

compare = makeContrasts(TktOEGR.M-TktOEWT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GRxWT.MOE=DEout
sGRxWT.MOE=DEout[DEout$FDR<.05,]


MDSdata = plotMDS(z, col=as.numeric(factor(gr)))

temp=convert[,"Symbol"]
temp2=setdiff(row.names(cpmdata),convert[,2])
names(temp)=convert[,2]
names(temp2)=temp2
convert=c(temp,temp2)

```

```{r fig1, echo = F, warning=F, fig.height=10, fig.width=10, fig.cap= 'MDS plot.'}

MDSplot = data.frame(x = MDSdata$x,
                     y = MDSdata$y,
                     sample = row.names(groups))
MDSplot = cbind(MDSplot, groups[MDSplot$sample,])
MDSplot$GenoSex = paste0(MDSplot$Genotype,MDSplot$Sex)
MDSplot = MDSplot[order(MDSplot$TKT),]
MDSplot = MDSplot[order(MDSplot$Sex),]
MDSplot = MDSplot[order(MDSplot$GenoSex),]
MDSplot$Group = factor(MDSplot$Group, levels = unique(MDSplot$Group))
##MDSplot$GenoSex = factor(MDSplot$GenoSex, levels = unique(MDSplot$GenoSex))

  fig  = plot_ly(data = MDSplot, 
                 x = ~x, 
                 y = ~y,
                 name = ~Group,
                 type = 'scatter',
                 mode = 'markers',
                 showlegend = T,
                 color = ~Group,
                 symbol = ~Group,
                 ##legendrank = ~ as.numeric(Group),
                 symbols = rep(c('circle', 'triangle-down', 'triangle-up'),4),
                 colors = c("red","firebrick3","deeppink",
                            "lightskyblue","royalblue","cyan",
                            "gold", "darkgoldenrod1", "yellow",
                            "limegreen", "darkgreen", "chartreuse"),
                 marker = list(size = 25,
                               line = list(color = "black", width = 1.5)),
                 hoverinfo = "text",
                 hovertext = paste("Sample:", MDSplot$Sample)) %>%
    layout(xaxis = list(title = paste0("Leading logFC dim 1 ", "(", round((MDSdata$var.explained[1]*100),1), "%)")),
           yaxis = list(title = paste0("Leading logFC dim 2 ", "(", round((MDSdata$var.explained[2]*100),1), "%)")))
  
fig

```

<br /><br /><br />
\newpage

```{r fig2, echo = F, fig.cap='Barplot of Total DEGs.'}

Female = c(nrow(sGR.F.CxDf), nrow(sGR.F.CxOE),
           nrow(sWT.F.CxDf), nrow(sWT.F.CxOE),
           nrow(sGRxWT.F), nrow(sGRxWT.FDf), nrow(sGRxWT.FOE))

Male = c(nrow(sGR.M.CxDf), nrow(sGR.M.CxOE),
         nrow(sWT.M.CxDf), nrow(sWT.M.CxOE),
         nrow(sGRxWT.M), nrow(sGRxWT.MDf), nrow(sGRxWT.MOE))
  
Comparison = c("GR.C x GR.Df", "GR.C x GR.OE",
               "WT.C x WT.Df", "WT.C x WT.OE",
               "GR.C x WT.C", "GR.Df x WT.Df", "GR.OE x WT.OE")

data = data.frame(Female, Male, Comparison)

data$Comparison = factor(data$Comparison, levels = data$Comparison)

fig <- plot_ly(data, 
               x = ~Comparison, y = ~Female, 
               type = 'bar',
               name = "Female",
               marker = list(color =c("firebrick"),
                             line = list(color = "black", width = 1.5)))

fig <- fig %>% add_trace(y = ~Male, name = 'Males',
                         marker = list(color = c("royalblue"),
                                       line = list(color = "black", width = 1.5)))

fig <- fig %>% layout(yaxis = list(title ="Total DEGs"),
                      xaxis = list(title = NA),
                      title = "Total DEGs")

fig

```


<br /><br /><br />
\newpage

```{r echo = F}

Comparisons = list(row.names(sGR.F.CxDf), row.names(sGR.F.CxOE),
                row.names(sWT.F.CxDf), row.names(sWT.F.CxOE),
                row.names(sGRxWT.F), row.names(sGRxWT.FDf), row.names(sGRxWT.FOE),
                row.names(sGR.M.CxDf), row.names(sGR.M.CxOE),
                row.names(sWT.M.CxDf), row.names(sWT.M.CxOE),
                row.names(sGRxWT.M), row.names(sGRxWT.MDf), row.names(sGRxWT.MOE))

```

<br /><br /><br />
\newpage

```{r fig3, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.FC x GR.FDf.'}

Samples = row.names(groups)[groups$Sex =='F'
                            & groups$Genotype == 'GR']

hmdata = cpmdata[Comparisons[[1]],Samples]
##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.FC x GR.FDf",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig4, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.FC x GR.FOE.'}

Samples = row.names(groups)[groups$Sex =='F'
                            & groups$Genotype == 'GR']

hmdata = cpmdata[Comparisons[[2]],Samples]


##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.FC x GR.FOE",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig5, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in WT.FC x WT.FDf.'}

Samples = row.names(groups)[groups$Sex =='F'
                            & groups$Genotype == 'WT']

hmdata = cpmdata[Comparisons[[3]],Samples]


##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "WT.FC x WT.FDf",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig6, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in WT.FC x WT.FOE.'}

Samples = row.names(groups)[groups$Sex =='F'
                            & groups$Genotype == 'WT']

hmdata = cpmdata[Comparisons[[4]],Samples]


##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "WT.FC x WT.FOE",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig7, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.FC x WT.FC.'}

Samples = row.names(groups)[groups$Sex =='F'
                            & groups$TKT == 'Control']

hmdata = cpmdata[Comparisons[[5]],Samples]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.FC x WT.FC",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig8, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.FDf x WT.FDf.'}

Samples = row.names(groups)[groups$Sex =='F'
                            & groups$TKT == 'DF']

hmdata = cpmdata[Comparisons[[6]],Samples]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.FDf x WT.FDf",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig9, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.FOE x WT.FOE.'}

Samples = row.names(groups)[groups$Sex =='F'
                            & groups$TKT == 'OE']

hmdata = cpmdata[Comparisons[[7]],Samples]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.FOE x WT.FOE",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig10, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.MC x GR.MDf.'}

Samples = row.names(groups)[groups$Sex =='M'
                            & groups$Genotype == 'GR']

hmdata = cpmdata[Comparisons[[8]],Samples]
##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.MC x GR.MDf",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig11, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.MC x GR.MOE.'}

Samples = row.names(groups)[groups$Sex =='M'
                            & groups$Genotype == 'GR']

hmdata = cpmdata[Comparisons[[9]],Samples]


##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.MC x GR.MOE",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig12, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in WT.MC x WT.MDf.'}

Samples = row.names(groups)[groups$Sex =='M'
                            & groups$Genotype == 'WT']

hmdata = cpmdata[Comparisons[[10]],Samples]


##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "WT.MC x WT.MDf",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig13, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in WT.MC x WT.MOE.'}

Samples = row.names(groups)[groups$Sex =='M'
                            & groups$Genotype == 'WT']

hmdata = cpmdata[Comparisons[[11]],Samples]


##max.hmdata = apply(hmdata, 1, max)
##hmdata = hmdata[max.hmdata > 5,]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "WT.MC x WT.MOE",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig14, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.MC x WT.MC.'}

Samples = row.names(groups)[groups$Sex =='M'
                            & groups$TKT == 'Control']

hmdata = cpmdata[Comparisons[[12]],Samples]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.MC x WT.MC",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig15, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.MDf x WT.MDf.'}

Samples = row.names(groups)[groups$Sex =='M'
                            & groups$TKT == 'DF']

hmdata = cpmdata[Comparisons[[13]],Samples]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.MDf x WT.MDf",
            custom_hovertext = cpm.text)

```

<br /><br /><br />
\newpage

```{r fig16, echo = F, fig.height=10, fig.cap='Heatmaps of Total DEGs in GR.MOE x WT.MOE.'}

Samples = row.names(groups)[groups$Sex =='M'
                            & groups$TKT == 'OE']

hmdata = cpmdata[Comparisons[[14]],Samples]

row.names(hmdata) = convert[row.names(hmdata)]

cpm.text = round(hmdata,1)
i=1
while(i<=ncol(cpm.text)){
  cpm.text[,i] = paste0("CPM: ", cpm.text[,i])
  i = i+1
}

  heatmaply(hmdata,
            trace="none",
            col=RdYlBu(100)[100:1],
            scale="row",
            dendrogram = "both",
            Rowv=T,
            Colv=T,
            cexRow = .75, 
            na.color="grey",
            cexCol = .75,
            labRow = row.names(hmdata),
            key = T,
            column_text_angle = 90,
            col_side_colors=groups[Samples,1:3],
            col_side_palette=Spectral,
            main= "GR.MOE x WT.MOE",
            custom_hovertext = cpm.text)

```




















