---
title: "G85R vs WT (M + F)"
author: "John Santiago"
date: "2023-07-11"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r echo = F, include=F}
##Load Libraries

library(biomaRt)
library(org.Dm.eg.db)
library(goseq)
library(GO.db)
library(visNetwork)
library(rrvgo)
library(plotly)
library(KEGGREST)
library(pathview)
library(VennDiagram)
library(heatmaply)
```

```{r, echo = F, include = F}

##load all files from github
git.dir = "https://raw.githubusercontent.com/johncsantiago/WhartonLab/master/"
cpmdata = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT_cpmdata.csv"),row.names = 1)
groups  = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.metadata.csv"),row.names = 1)
TKT.EdgeR = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FDRTable.csv"),row.names = 1)
TKT.EdgeR.FC = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FCTable.csv"),row.names = 1)

GeneIDKey = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/GeneIDKey.csv", row.names = 1)
kegg = readRDS(gzcon(url(paste0(git.dir,"GeneralDataFiles/kegg.symbol2path.RData"))))
kegg.names = read.csv(paste0(git.dir, "GeneralDataFiles/KEGG.names.csv"), row.names = 1)

Synaptic = read.csv("/Users/johncsantiago/Documents/SynapticGenesandTFs.csv")

genesingo=as.list(org.Dm.egGO2ALLEGS)


##mean.cpm; Mean cpm for each gene in each condition (groups$Group). Used for filtering
mean.cpm = matrix(0, ncol = length(unique(groups$Group)), nrow = nrow(cpmdata))
colnames(mean.cpm) = unique(groups$Group)
row.names(mean.cpm) = row.names(cpmdata)
i=1
while(i<=ncol(mean.cpm)){
  mean.cpm[,i]= apply(cpmdata[,row.names(groups[groups$Group == colnames(mean.cpm)[i],])], 1, mean)
  i=i+1
}

i=1
for(i in 1:ncol(TKT.EdgeR)){
  TKT.EdgeR[TKT.EdgeR[,i] == 0,i] = 2
  i=i+1
}


```

##GO and KEGG Enrichment function
```{r echo = F, include = F}

Enrichment = function(sigs, bg){
  genes = setNames(rep(0, length(bg)), bg)
  genes[sigs] = 1
  pwf=nullp(genes,"dm3","geneSymbol")
  GO.wall=goseq(pwf,"dm3","geneSymbol")
  GO.wall$adjp=p.adjust(GO.wall$over_represented_pvalue,method="BH")
  row.names(GO.wall) = GO.wall$category
  KEGG=goseq(pwf,gene2cat=kegg)
  KEGG$adjp=p.adjust(KEGG$over_represented_pvalue,method="BH")
  row.names(KEGG)=substr(as.character(KEGG[,1]),6,13)
  KEGG$Name=kegg.names[row.names(KEGG),1]

  enrichment.data = list(GO.wall, KEGG)
  names(enrichment.data) = c("GO", "KEGG")
  return(enrichment.data)
}

```

##GO.Sim.PCA function
```{r echo = F, include = F}
GO.Sim.PCA = function(use.data){
    pca <- prcomp(t(use.data), scale.=TRUE) 
    pc1=scale(pca$x[,1])
    pc2=scale(pca$x[,2])
    eigs <- pca$sdev^2
    ve=signif(((eigs / sum(eigs))*100)[1:3],4)

    pca.data=data.frame(PC1=pc1,
                        PC2=pc2,
                        sdev=pca$sdev,
                        Percent1=ve[1],
                        Percent2=ve[2],
                        Group = reducedTerms[row.names(use.data), "parentTerm"],
                        Score = reducedTerms[row.names(use.data), "score"],
                        Term = reducedTerms[row.names(use.data), "term"],
                        TotalDE = GO.data[row.names(use.data), "numDEInCat"],
                        Total = GO.data[row.names(use.data), "numInCat"],
                        SigVal = signif(GO.data[row.names(use.data), "adjp"],3))

    pca.data$Group = factor(pca.data$Group, levels = unique(pca.data$Group))
    pca.data$Score = 20*(1+pca.data$Score/mean(pca.data$Score))
    
  fig  = plot_ly(data = pca.data, 
                 x = ~PC1, 
                 y = ~PC2,
                 name = ~Group,
                 type = 'scatter',
                 mode = 'markers',
                 showlegend = T,
                 color = ~Group,
                 ##symbol = ~Group,
                 ##symbols = as.character(seq_along(unique(point.data$Set))),
                 ##colors = c("firebrick","gold","deepskyblue"),
                 marker = list(size = pca.data$Score,
                               line = list(color = "black", width = 1.5)),
                 hoverinfo = "text",
                 hovertext = paste("Parent Group: ", pca.data$Group,
                                   "\nSample: ", pca.data$Term, 
                                   "\nTotal DE: ", pca.data$TotalDE, 
                                   "\nTotal in Cat.: ", pca.data$Total, 
                                   "\nFDR: ", pca.data$SigVal)) %>%
    layout(xaxis = list(title = paste0("PC1", "(", ve[1], "%)")),
           yaxis = list(title = paste0("PC2", "(", ve[2], "%)")),
           title = "Enriched GO term similarity")
           
  return(fig)
}

```

##GO20 Function
```{r echo = F, include =F}

GO20 = function(GO.data, GO.ontology){
GO.data = GO.data[GO.data$ontology == GO.ontology,
                  c("category", "term","adjp", "numDEInCat", "numInCat")]


data = GO.data[20:1,c("term","adjp","numDEInCat", "numInCat", "category")]
data$Score = -log10(data$adjp)
data$term = factor(data$term, levels = c(unique(data$term)))

m <- list(
  l = 50,
  r = 50,
  b = 100,
  t = 100,
  pad = 4
)

fig <- plot_ly(data, 
               x = ~numDEInCat, y = ~term, 
               type = 'bar',
               name = "term",
               marker = list(colorscale = list(c(0,.5,1), c("brown","gold", "steelblue")),
    reversescale =T,
                             colorbar = list(title = "-log10(FDR)",
                                      len = .8, outlinewidth = 0,
                                      tickformat = ".2f",
                                      tick0 = 0),##, dtick = .01),
                             color = ~Score,
                             line = list(color = "black", width = 1.5)),
    hoverinfo = "text",
    hovertext = paste("Sample:", data$term,
                      "\nGO ID: ", data$category,
                      "\nTotal DE: ", data$numDEInCat,
                      "\nTotal in Cat.: ", data$numInCat,
                      "\nFDR: ", signif(data$adjp, 3))) %>%
    layout(xaxis = list(title = "Number of Genes"),
           yaxis = list(title = ""),
           title = "Top 20 Most Enriched GO Terms",
           margin = m)

  return(fig)
}

```

##KEGG20 Function
```{r, echo = F, include = F}

KEGG20 = function(data){
  
  data = data[1:20, c("category", "Name","adjp", "numDEInCat", "numInCat")]
  
data$term = data$Name
if(length(unique(data$term)) < length(data$term)){
  data$term[duplicated(data$term)] = paste(data$term[duplicated(data$term)], "2")
}  
data = data[20:1,]
data$Score = -log10(data$adjp)
data$term = factor(data$term, levels = c(unique(data$term)))


m <- list(
  l = 50,
  r = 50,
  b = 100,
  t = 100,
  pad = 4
)

fig <- plot_ly(data, 
               x = ~numDEInCat, y = ~term, 
               type = 'bar',
               name = "Name",
               marker = list(colorscale = list(c(0,.5,1), c("brown","gold", "steelblue")),
    reversescale =T,
                             colorbar = list(title = "-log10(FDR)",
                                      len = .8, outlinewidth = 0,
                                      tickformat = ".2f",
                                      tick0 = 0),##, dtick = .01),
                             color = ~Score,
                             line = list(color = "black", width = 1.5)),
    hoverinfo = "text",
    hovertext = paste("Sample:", data$Name,
                      "\nGO ID: ", data$category,
                      "\nTotal DE: ", data$numDEInCat,
                      "\nTotal in Cat.: ", data$numInCat,
                      "\nFDR: ", signif(data$adjp, 3))) %>%
    layout(xaxis = list(title = "Number of Genes"),
           yaxis = list(title = ""),
           title = "Top 20 Most Enriched KEGG Terms",
           margin = m)

return(fig)

}

```
  
```{r echo = F, include = F}

##GR vs WT Females
sigs1 = row.names(TKT.EdgeR)[TKT.EdgeR[,"GRxWT.FC"] <= .05]
bg1 = row.names(TKT.EdgeR)[TKT.EdgeR[, "GRxWT.FC"]<= 1]

sigs2 = row.names(TKT.EdgeR)[TKT.EdgeR[,"GRxWT.MC"] <= .05]
bg2 = row.names(TKT.EdgeR)[TKT.EdgeR[, "GRxWT.MC"]<= 1]

sigs = intersect(sigs1, sigs2)
bg = unique(c(bg1, bg2))

sigsKEY = GeneIDKey[sigs,]
sigs = sigsKEY$Symbol
bgKEY = GeneIDKey[bg,]
bg = bgKEY$Symbol

sig.cats = Enrichment(sigs, bg)

```

```{r echo =F, include =F}

GO.data = sig.cats[["GO"]]
GO.ontology = "BP"
go_analysis = GO.data$category[GO.data$adjp <= .05 & GO.data$ontology == GO.ontology]
scores = setNames(-log10(GO.data$adjp), go_analysis)

simMatrix <- calculateSimMatrix(go_analysis,
                                orgdb="org.Dm.eg.db",
                                ont=GO.ontology,
                                method="Rel")

reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Dm.eg.db")

```


```{r echo = F, include =F}

GO.data = sig.cats[["GO"]]
GO.ontology = "BP"
BP.GO.fig = GO20(GO.data, GO.ontology)

GO.ontology = "CC"
CC.GO.fig = GO20(GO.data, GO.ontology)

GO.ontology = "MF"
MF.GO.fig = GO20(GO.data, GO.ontology)

KEGG.fig = KEGG20(sig.cats[["KEGG"]])

```

```{r, fig.width= 10}

BP.GO.fig

```

```{r, fig.width= 10}

CC.GO.fig

```

```{r, fig.width= 10}

MF.GO.fig

```

```{r, fig.width= 10}

KEGG.fig

```

```{r fig.height=7, fig.width=10}

heatmapPlot(simMatrix,
            reducedTerms,
            annotateParent=TRUE,
            annotationLabel="parentTerm",
            fontsize=6,
            main = "Repressor Gene Enriched Biological Process GO terms")

```


```{r warning=F, fig.width=10}

use.data   = simMatrix
GO.Sim.PCA(use.data)

```


```{r fig.align='left'}

treemapPlot(reducedTerms)

```

```{r}

##GR.F vs WT.F specific volcano

specific.cat = c('none', 'parent.category', 'specific.kegg', 'specific.go')

specific.cat = specific.cat[1]
parent.category = 'neurogenesis'
specific.kegg = "DNA replication"
specific.go =  '32553'
Synaptic.genes = F

set.size = F
  
GOterm.genes = vector('list', length = length(unique(reducedTerms$parent)))
names(GOterm.genes) = unique(reducedTerms$parent)
i=1
while(i<=length(GOterm.genes)){
  goinparent = reducedTerms[grep(names(GOterm.genes)[i], reducedTerms$parent),'go']
  genesinparent = unlist(genesingo[goinparent])
  GOterm.genes[[i]] = GeneIDKey[GeneIDKey$ensembl %in% genesinparent, 'FBgn']
  i = i +1
}

volcano.data = data.frame(Symbol = GeneIDKey[row.names(TKT.EdgeR), "Symbol"], 
                          FDR = -log2(TKT.EdgeR[, 'GRxWT.FC']),
                          FC = TKT.EdgeR.FC[row.names(TKT.EdgeR), 'GRxWT.FC'],
                          ##Color = 0)
                          Color = 'grey',
                          size = 5*log(mean.cpm[row.names(TKT.EdgeR),"GR.F"]),
                          GR.F.cpm = mean.cpm[row.names(TKT.EdgeR),"GR.F"],
                          WT.F.cpm = mean.cpm[row.names(TKT.EdgeR),"WT.F"],
                          GO.terms = "none")

main.title = "Significant DEG in G85R Females vs. WT Females"

volcano.data$Color[volcano.data$FDR >= -log2(.05) & volcano.data$FC < 0] = 'lightblue'
volcano.data$Color[volcano.data$FDR >= -log2(.0001) & volcano.data$FC < 0] = 'steelblue'
volcano.data$Color[volcano.data$FDR >= -log2(.000001) & volcano.data$FC < 0] = 'dodgerblue'

volcano.data$Color[volcano.data$FDR >= -log2(.05) & volcano.data$FC > 0] = 'lightcoral'
volcano.data$Color[volcano.data$FDR >= -log2(.0001) & volcano.data$FC > 0] = 'tomato'
volcano.data$Color[volcano.data$FDR >= -log2(.000001) & volcano.data$FC > 0] = 'firebrick'

volcano.data$size[volcano.data$FDR <= -log2(.05)] = 3

if(set.size == T){
  volcano.data$size = 15
  volcano.data$size[volcano.data$FDR <= -log2(.05)] = 5
}

volcano.data$Color = 'grey'
volcano.data$Color[volcano.data$Symbol %in% Synaptic$Symbol] = 'deeppink'

if(specific.cat == 'parent.category'){
  parent.genes = GOterm.genes[[intersect(reducedTerms[reducedTerms$parentTerm == parent.category, "go"], names(GOterm.genes))]]
  volcano.data$Color = 'honeydew'
  volcano.data[parent.genes, 'Color'] = 'deeppink'
  volcano.data = volcano.data[order(volcano.data$Color, decreasing = T),]
  main.title = sig.cats$GO[c(grep(parent.category, sig.cats$GO$term), grep(parent.category, sig.cats$GO$category)), 'term']
}


if(specific.cat == 'specific.kegg'){
  kegg.id = sig.cats$KEGG[c(grep(specific.kegg, sig.cats$KEGG$Name), grep(specific.kegg, sig.cats$KEGG$category)), 'category']
  kegg.genes = names(kegg[grep(kegg.id, kegg)])
  volcano.data$Color = 'honeydew'
  row.names(volcano.data) = volcano.data$Symbol
    volcano.data[kegg.genes, 'Color'] = 'deeppink'
  volcano.data = volcano.data[order(volcano.data$Color, decreasing = T),]
  main.title = sig.cats$KEGG[c(grep(specific.kegg, sig.cats$KEGG$Name), grep(specific.kegg, sig.cats$KEGG$category)), 'Name']
}


if(specific.cat == 'specific.go'){
  go.id = sig.cats$GO[c(grep(specific.go, sig.cats$GO$term), grep(specific.go, sig.cats$GO$category)), 'category']
  volcano.data$Color = 'honeydew'
  go.genes = genesingo[[go.id]]
  go.genes = GeneIDKey[GeneIDKey$ensembl %in% go.genes, "FBgn"]
  go.genes = intersect(go.genes, row.names(volcano.data))
  volcano.data[go.genes, 'Color'] = 'deeppink'
  volcano.data = volcano.data[order(volcano.data$Color, decreasing = T),]
  main.title = sig.cats$GO[c(grep(specific.go, sig.cats$GO$term), grep(specific.go, sig.cats$GO$category)), 'term']
}

##volcano.data


fig = plot_ly(data = volcano.data,
              x = ~FC,
              y = ~FDR,
              type = 'scatter',
              mode = 'markers',
              marker = list(color = ~Color, colors = ~Color, size = volcano.data$size,
                            line = list(color = 'black', width = .5)),
              hoverinfo = "text",
              hovertext = paste("Gene:", volcano.data$Symbol,
                      "\n-log2(FDR): ", round(volcano.data$FDR,2),
                      "\nFC: ", round(volcano.data$FC,2),
                      "\nG85R mean cpm: ", round(volcano.data$GR.F.cpm, 1),
                      "\nWT mean cpm: ", round(volcano.data$WT.F.cpm, 1)))
fig = fig %>% layout(title = main.title)

              ##color = ~Color,
              ##colors = 'Spectral')
              ##color = ~Color,
              ##colors = ~Color)
              ##marker = list(colorscale = list(c(0,.5,1), c("blue","yellow", "red")), color = ~Color))

fig

```

```{r}
specific.cat = c('none', 'parent.category', 'specific.kegg', 'specific.go')

specific.cat = specific.cat[1]
parent.category = 'chromosome organization'
specific.kegg = "DNA replication"
specific.go =  '006457'

set.size = F
  
GOterm.genes = vector('list', length = length(unique(reducedTerms$parent)))
names(GOterm.genes) = unique(reducedTerms$parent)
i=1
while(i<=length(GOterm.genes)){
  goinparent = reducedTerms[grep(names(GOterm.genes)[i], reducedTerms$parent),'go']
  genesinparent = unlist(genesingo[goinparent])
  GOterm.genes[[i]] = GeneIDKey[GeneIDKey$ensembl %in% genesinparent, 'FBgn']
  i = i +1
}

volcano.data = data.frame(Symbol = GeneIDKey[row.names(TKT.EdgeR), "Symbol"], 
                          FDR = -log2(TKT.EdgeR[, 'GRxWT.MC']),
                          FC = TKT.EdgeR.FC[row.names(TKT.EdgeR), 'GRxWT.MC'],
                          ##Color = 0)
                          Color = 'grey',
                          size = 5*log(mean.cpm[row.names(TKT.EdgeR),"GR.M"]),
                          GR.F.cpm = mean.cpm[row.names(TKT.EdgeR),"GR.M"],
                          WT.F.cpm = mean.cpm[row.names(TKT.EdgeR),"WT.M"],
                          GO.terms = "none")

main.title = "Significant DEG in G85R Males vs. WT Males"

volcano.data$Color[volcano.data$FDR >= -log2(.05) & volcano.data$FC < 0] = 'lightblue'
volcano.data$Color[volcano.data$FDR >= -log2(.0001) & volcano.data$FC < 0] = 'steelblue'
volcano.data$Color[volcano.data$FDR >= -log2(.000001) & volcano.data$FC < 0] = 'dodgerblue'

volcano.data$Color[volcano.data$FDR >= -log2(.05) & volcano.data$FC > 0] = 'lightcoral'
volcano.data$Color[volcano.data$FDR >= -log2(.0001) & volcano.data$FC > 0] = 'tomato'
volcano.data$Color[volcano.data$FDR >= -log2(.000001) & volcano.data$FC > 0] = 'firebrick'

volcano.data$size[volcano.data$FDR <= -log2(.05)] = 3

if(set.size == T){
  volcano.data$size = 15
  volcano.data$size[volcano.data$FDR <= -log2(.05)] = 5
}


if(specific.cat == 'parent.category'){
  parent.genes = GOterm.genes[[intersect(reducedTerms[reducedTerms$parentTerm == parent.category, "go"], names(GOterm.genes))]]
  volcano.data$Color = 'honeydew'
  volcano.data[parent.genes, 'Color'] = 'deeppink'
  volcano.data = volcano.data[order(volcano.data$Color, decreasing = T),]
  main.title = sig.cats$GO[c(grep(parent.category, sig.cats$GO$term), grep(parent.category, sig.cats$GO$category)), 'term']
}


if(specific.cat == 'specific.kegg'){
  kegg.id = sig.cats$KEGG[c(grep(specific.kegg, sig.cats$KEGG$Name), grep(specific.kegg, sig.cats$KEGG$category)), 'category']
  kegg.genes = names(kegg[grep(kegg.id, kegg)])
  volcano.data$Color = 'honeydew'
  row.names(volcano.data) = volcano.data$Symbol
    volcano.data[kegg.genes, 'Color'] = 'deeppink'
  volcano.data = volcano.data[order(volcano.data$Color, decreasing = T),]
  main.title = sig.cats$KEGG[c(grep(specific.kegg, sig.cats$KEGG$Name), grep(specific.kegg, sig.cats$KEGG$category)), 'Name']
}


if(specific.cat == 'specific.go'){
  go.id = sig.cats$GO[c(grep(specific.go, sig.cats$GO$term), grep(specific.go, sig.cats$GO$category)), 'category']
  volcano.data$Color = 'honeydew'
  go.genes = genesingo[[go.id]]
  go.genes = GeneIDKey[GeneIDKey$ensembl %in% go.genes, "FBgn"]
  go.genes = intersect(go.genes, row.names(volcano.data))
  volcano.data[go.genes, 'Color'] = 'deeppink'
  volcano.data = volcano.data[order(volcano.data$Color, decreasing = T),]
  main.title = sig.cats$GO[c(grep(specific.go, sig.cats$GO$term), grep(specific.go, sig.cats$GO$category)), 'term']
}

##volcano.data


fig = plot_ly(data = volcano.data,
              x = ~FC,
              y = ~FDR,
              type = 'scatter',
              mode = 'markers',
              marker = list(color = ~Color, colors = ~Color, size = volcano.data$size,
                            line = list(color = 'black', width = .5)),
              hoverinfo = "text",
              hovertext = paste("Gene:", volcano.data$Symbol,
                      "\n-log2(FDR): ", round(volcano.data$FDR,2),
                      "\nFC: ", round(volcano.data$FC,2),
                      "\nG85R mean cpm: ", round(volcano.data$GR.F.cpm, 1),
                      "\nWT mean cpm: ", round(volcano.data$WT.F.cpm, 1)))
fig = fig %>% layout(title = main.title)

fig

```


```{r}

rawdata = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetabolomicsForMetaboanalyst.txt", sep = "\t", row.names = 1)

KEGG.Key = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/RawMetabolomicData.csv", row.names = 1)
KEGG.Key = setNames(KEGG.Key[-c(1:2), 2], row.names(KEGG.Key)[-c(1:2)])

genotypes = setNames(c("WT", "GR", "TktOEWT", "TktOEGR", "TktDfWT", "TktDfGR"), c("C", "E", "A", "F", "B", "D"))

metadata = data.frame(Sample = colnames(rawdata), 
                      Group = as.character(rawdata[1,]), 
                      TIC = as.numeric(rawdata[2,]), 
                      Genotypes = genotypes[as.character(rawdata[1,])])



norm.func = function(data){
  nd = (as.numeric(data)/metadata$TIC)*1000
  return(nd)
}

norm.data = t(apply(rawdata[3:nrow(rawdata),], 1, norm.func))
colnames(norm.data) = colnames(rawdata)

group.mean=function(data){
  gm = mean(na.omit(data))
}

mean.data = matrix(0, nrow=nrow(norm.data), ncol = length(genotypes))
row.names(mean.data) = row.names(norm.data)
colnames(mean.data) = unique(metadata$Group)

i=1
while(i<=ncol(mean.data)){
  mean.data[,i] = apply(norm.data[,metadata[metadata$Group == colnames(mean.data)[i], "Sample"]], 1, group.mean)
  i = i+1
}

compare.conditions = function(condition1, condition2){
  columns1 = metadata[metadata$Group == condition1, "Sample"]
  columns2 = metadata[metadata$Group == condition2, "Sample"]
  p=setNames(rep(NA, nrow(norm.data)), row.names(norm.data))
  i=1
  while(i<=length(p)){
    if(length(na.omit(norm.data[i,columns1]))>2 & 
       length(na.omit(norm.data[i,columns2]))>2){
      temp = t.test(na.omit(norm.data[i,columns1]),
                    na.omit(norm.data[i,columns2]), paired = F, var.equal = T)
      p[i] = temp[[3]]
      }
    i=i+1
    }

  fdr = p.adjust(p, "BH", length(p))

  fc = mean.data[,condition1]/mean.data[,condition2]

  comparison.table = data.frame(p = p, 
                                FDR = fdr, 
                                FC = fc, 
                                KEGG = KEGG.Key[row.names(mean.data)])
  
  comparison.table = comparison.table[order(comparison.table$FDR),]
}

GRxWT.C = compare.conditions("E", "C")
GRxWT.Df = compare.conditions("D", "B")
GRxWT.OE = compare.conditions("F", "A")

```


```{r}

set.size == F
volcano.data = data.frame(Symbol = row.names(GRxWT.C), 
                          FDR = -log2(GRxWT.C$FDR),
                          FC = log2(GRxWT.C$FC),
                          Color = 'honeydew',
                          size = log(mean.data[row.names(GRxWT.C),'E']),
                          GR.Level = mean.data[row.names(GRxWT.C), "E"],
                          WT.Level = mean.data[row.names(GRxWT.C), "C"],
                          GO.terms = "none")

main.title = "FC for metabolites in G85R vs. WT"

volcano.data$Color[volcano.data$FDR >= -log2(.05) & volcano.data$FC < 0] = 'lightblue'
volcano.data$Color[volcano.data$FDR >= -log2(.0001) & volcano.data$FC < 0] = 'steelblue'
volcano.data$Color[volcano.data$FDR >= -log2(.000001) & volcano.data$FC < 0] = 'dodgerblue'

volcano.data$Color[volcano.data$FDR >= -log2(.05) & volcano.data$FC > 0] = 'lightcoral'
volcano.data$Color[volcano.data$FDR >= -log2(.0001) & volcano.data$FC > 0] = 'tomato'
volcano.data$Color[volcano.data$FDR >= -log2(.000001) & volcano.data$FC > 0] = 'firebrick'

##volcano.data$size[volcano.data$FDR <= -log2(.05)] = 3

if(set.size == T){
  volcano.data$size = 15
  volcano.data$size[volcano.data$FDR <= -log2(.05)] = 5
}

fig = plot_ly(data = volcano.data,
              x = ~FC,
              y = ~FDR,
              type = 'scatter',
              mode = 'markers',
              marker = list(color = ~Color, colors = ~Color, size = volcano.data$size,
                            line = list(color = 'black', width = .5)),
              hoverinfo = "text",
              hovertext = paste("Metabolite:", volcano.data$Symbol,
                      "\n-log2(FDR): ", round(volcano.data$FDR,2),
                      "\nFC: ", round(volcano.data$FC,2),
                      "\nG85R mean cpm: ", round(volcano.data$GR.Level, 1),
                      "\nWT mean cpm: ", round(volcano.data$WT.Level, 1)))
fig = fig %>% layout(title = main.title)

fig

```



```{r, include = F, echo = F}

if(intersect(list.files(), "KEGG_Image_Files") != "KEGG_Image_Files"){
  dir.create(paste0(dirname(rstudioapi::getSourceEditorContext()$path), "/KEGG_Image_Files/"))
}

Home = dirname(rstudioapi::getSourceEditorContext()$path)
OutputDir = paste0(Home, "/KEGG_Image_Files/")

ShowKEGG = function(PathwayID){
  
download.kegg(pathway.id= PathwayID, species = "dme", kegg.dir = OutputDir)
xml.file=paste0(OutputDir, "dme",PathwayID,".xml")
node.data=node.info(xml.file)

all.path = names(kegg[grep(PathwayID,kegg)])

up = TKT.EdgeR.FC[TKT.EdgeR.FC$Moderate == 1, "Symbol"]
Moderate = intersect(Moderate, all.path)
Moderate = paste("Dmel_", GeneIDKey[GeneIDKey$Symbol %in% Moderate, "CG"],sep="")

Strong = Df.data[Df.data$Strong == 1, "Symbol"]
Strong = intersect(Strong, all.path)
Strong = paste("Dmel_", GeneIDKey[GeneIDKey$Symbol %in% Strong, "CG"],sep="")

Enhancer = Df.data[Df.data$Enhancer == 1, "Symbol"]
Enhancer = intersect(Enhancer, all.path)
Enhancer = paste("Dmel_", GeneIDKey[GeneIDKey$Symbol %in% Enhancer, "CG"],sep="")

Lethal = Df.data[Df.data$Lethal == 1, "Symbol"]
Lethal = intersect(Lethal, all.path)
Lethal = paste("Dmel_", GeneIDKey[GeneIDKey$Symbol %in% Strong, "CG"],sep="")

plot.data.gene=node.map(mol.data=c(Moderate, Strong, Enhancer, Lethal), node.data, node.types="gene")
cols.ts.gene = node.color(plot.data.gene, limit=2, bins=2)
cols.ts.gene[1:length(cols.ts.gene)] = 0

pdg.moderate=node.map(mol.data=Moderate, node.data, node.types="gene")
temp = node.color(pdg.moderate, limit=2, bins=2)
ctg.moderate = cols.ts.gene
ctg.moderate[temp!="#FFFFFF"]= 1 ##"gold2"

pdg.strong=node.map(mol.data=Strong, node.data, node.types="gene")
temp = node.color(pdg.strong, limit=2, bins=2)
ctg.strong = cols.ts.gene
ctg.strong[temp!="#FFFFFF"]= 1 ##"brown2"

pdg.enhancer=node.map(mol.data=Enhancer, node.data, node.types="gene")
temp = node.color(pdg.enhancer, limit=2, bins=2)
ctg.enhancer = cols.ts.gene
ctg.enhancer[temp!="#FFFFFF"]= 1 ##"deepskyblue"

pdg.lethal=node.map(mol.data=Lethal, node.data, node.types="gene")
temp = node.color(pdg.lethal, limit=2, bins=2)
ctg.lethal = cols.ts.gene
ctg.lethal[temp!="#FFFFFF"]= 1 ##"navyblue"


color.combos = data.frame(Moderate = as.numeric(ctg.moderate),
                          Strong = as.numeric(ctg.strong),
                          Enhancer = as.numeric(ctg.enhancer),
                          Lethal = as.numeric(ctg.lethal))
row.names(color.combos)= names(cols.ts.gene)

cols.ts.gene[cols.ts.gene=="0"]="darkseagreen1"

cols.ts.gene[row.names(color.combos)[color.combos$Moderate == 1]] = "gold2"
cols.ts.gene[row.names(color.combos)[color.combos$Strong   == 1]] = "brown1"
cols.ts.gene[row.names(color.combos)[color.combos$Enhancer   == 1]] = "lightsteelblue"
cols.ts.gene[row.names(color.combos)[color.combos$Lethal   == 1]] = "cornflowerblue"
cols.ts.gene[row.names(color.combos)[(color.combos$Moderate   == 1 | color.combos$Strong   == 1) & (color.combos$Enhancer   == 1 | color.combos$Lethal   == 1) ]] = "grey60"


setwd(OutputDir)
pv.pars= keggview.native(plot.data.gene=plot.data.gene,
                         cols.ts.gene=cols.ts.gene, 
                         node.data=node.data,
                         pathway.name=paste0("dme", PathwayID),
                         same.layer=T, plot.col.key=F, out.suffix = kegg.names[paste0("dme", PathwayID),1],
                         kegg.dir = OutputDir)
setwd(Home)
}

```
