---
title: "G85R Analysis: TKT Signaling Pathway"
author: "John Santiago"
date: "2024-09-18"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r include = F, echo = F}

library(plotly)
library(edgeR)
library(heatmaply)
library(org.Dm.eg.db)
library(KEGGREST)
library(RcisTarget)
library(DT)
source('/Users/johncsantiago/Documents/GitHub/WhartonLab/GOandKEGGPlottingFunctions.R')
library(VennDiagram)

```


```{r include = F, echo = F}
##load all files from github
git.dir = "https://raw.githubusercontent.com/johncsantiago/WhartonLab/master/"
cpmdata = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT_cpmdata.csv"),row.names = 1)
groups  = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.metadata.csv"),row.names = 1)
TKT.EdgeR = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FDRTable.csv"),row.names = 1)
TKT.EdgeR.FC = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FCTable.csv"),row.names = 1)

G85R.meancpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT_meancpmdata.csv", row.names = 1)

groups = groups[colnames(cpmdata),]


GeneIDKey = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/GeneIDKey.csv", row.names = 1)

##Clean Trait Data and calculate CPMs
countdata=read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT.counttable.csv",row.names=1)


##normalize data
countdata=countdata[, row.names(groups)]
x <- countdata
gr <- factor(groups$Group)
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
design<-model.matrix(~0 + gr)
colnames(design) = unique(gr)
z = estimateDisp(z, design, robust = T)
fit <- glmQLFit(z, design)

compare = makeContrasts(TktDfGR.F-WT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.FDfxWT.FC=DEout
sGR.FDfxWT.FC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.FDfxWT.FC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[17] = 'GR.FDfxWT.FC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.FDfxWT.FC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[17] = 'GR.FDfxWT.FC'

compare = makeContrasts(TktDfGR.M-WT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.MDfxWT.MC=DEout
sGR.MDfxWT.MC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.MDfxWT.MC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[18] = 'GR.MDfxWT.MC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.MDfxWT.MC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[18] = 'GR.MDfxWT.MC'


compare = makeContrasts(TktOEGR.F-WT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.FOExWT.FC=DEout
sGR.FOExWT.FC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.FOExWT.FC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[19] = 'GR.FOExWT.FC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.FOExWT.FC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[19] = 'GR.FOExWT.FC'

compare = makeContrasts(TktOEGR.M-WT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.MOExWT.MC=DEout
sGR.MOExWT.MC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.MOExWT.MC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[20] = 'GR.MOExWT.MC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.MOExWT.MC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[20] = 'GR.MOExWT.MC'

rawdata = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetabolomicsForMetaboanalyst.txt", sep = "\t", row.names = 1)
KEGG.Key = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/RawMetabolomicData.csv", row.names = 1)
KEGG.Key = setNames(KEGG.Key[-c(1:2), 2], row.names(KEGG.Key)[-c(1:2)])

genotypes = setNames(c("WT", "GR", "TktOEWT", "TktOEGR", "TktDfWT", "TktDfGR"), c("C", "E", "A", "F", "B", "D"))

metadata = data.frame(Sample = colnames(rawdata), 
                      Group = as.character(rawdata[1,]), 
                      TIC = as.numeric(rawdata[2,]), 
                      Genotypes = genotypes[as.character(rawdata[1,])])

row.names(metadata) = metadata$Sample

norm.func = function(data){
  nd = (as.numeric(data)/metadata$TIC)*1000
  return(nd)
}

norm.data = t(apply(rawdata[3:nrow(rawdata),], 1, norm.func))
colnames(norm.data) = colnames(rawdata)

group.mean=function(data){
  gm = mean(na.omit(data))
}

mean.data = matrix(0, nrow=nrow(norm.data), ncol = length(genotypes))
row.names(mean.data) = row.names(norm.data)
colnames(mean.data) = unique(metadata$Group)

i=1
while(i<=ncol(mean.data)){
  mean.data[,i] = apply(norm.data[,metadata[metadata$Group == colnames(mean.data)[i], "Sample"]], 1, group.mean)
  i = i+1
}

mean.metab = mean.data

compare.conditions = function(condition1, condition2){
  columns1 = metadata[metadata$Group == condition1, "Sample"]
  columns2 = metadata[metadata$Group == condition2, "Sample"]
  p=setNames(rep(NA, nrow(norm.data)), row.names(norm.data))
  i=1
  while(i<=length(p)){
    if(length(na.omit(norm.data[i,columns1]))>2 & 
       length(na.omit(norm.data[i,columns2]))>2){
      temp = t.test(na.omit(norm.data[i,columns1]),
                    na.omit(norm.data[i,columns2]), paired = F, var.equal = T)
      p[i] = temp[[3]]
      }
    i=i+1
    }

  fdr = p.adjust(p, "BH", length(p))

  fc = mean.metab[,condition1]/mean.metab[,condition2]

  comparison.table = data.frame(p = p, 
                                FDR = fdr, 
                                FC = fc, 
                                KEGG = KEGG.Key[row.names(mean.metab)])
  
  comparison.table = comparison.table[order(comparison.table$FDR),]
}

GRxWT.C = compare.conditions("E", "C")
GRxWT.Df = compare.conditions("D", "B")[row.names(GRxWT.C),]
GRxWT.OE = compare.conditions("F", "A")[row.names(GRxWT.C),]

GR.CxDf = compare.conditions("D", "E")[row.names(GRxWT.C),]
WT.CxDf = compare.conditions("B", "C")[row.names(GRxWT.C),]

GR.CxOE = compare.conditions("F", "E")[row.names(GRxWT.C),]
WT.CxOE = compare.conditions("A", "C")[row.names(GRxWT.C),]

GR.DfxWT.C = compare.conditions("D", "C")[row.names(GRxWT.C),]
GR.OExWT.C = compare.conditions("F", "C")[row.names(GRxWT.C),]



comparison.means = list(GRF.CxDf = c("TktDfGR.F","GR.F"), 
                 GRF.CxOE = c("TktOEGR.F","GR.F"),
                 WTF.CxDf = c("TktDfWT.F","WT.F"),
                 WTF.CxOE = c("TktOEWT.F","WT.F"),
                 GRxWT.FC = c("GR.F","WT.F"), 
                 GRxWT.FDf = c("TktDfGR.F","TktDfWT.F"),
                 GRxWT.FOE = c("TktOEGR.F","TktOEWT.F"), 
                 GRM.CxDf = c("TktDfGR.M","GR.M"), 
                 GRM.CxOE = c("TktOEGR.M","GR.M"),
                 WTM.CxDf = c("TktDfWT.M","WT.M"),
                 WTM.CxOE = c("TktOEWT.M","WT.M"),
                 GRxWT.MC = c("GR.M","WT.M"), 
                 GRxWT.MDf = c("TktDfGR.M","TktDfWT.M"),
                 GRxWT.MOE = c("TktOEGR.M","TktOEWT.M"),
                 GR.FxM = c("GR.F","GR.M"),
                 WT.FxM = c("WT.F","WT.M"))


metab.FC = data.frame('GRxWT.C' = GRxWT.C$FC,
                      'GRxWT.Df' = GRxWT.Df$FC,
                      'GRxWT.OE' = GRxWT.OE$FC,
                      'GR.CxDf' = GR.CxDf$FC,
                      'WT.CxDf' = WT.CxDf$FC,
                      'GR.CxOE' = GR.CxOE$FC,
                      'WT.CxOE' = WT.CxOE$FC,
                      'GR.DfxWT.C' = GR.DfxWT.C$FC,
                      'GR.OExWT.C' = GR.OExWT.C$FC)
row.names(metab.FC) = row.names(GRxWT.C)
metab.FC = log2(metab.FC)

metab.FDR = data.frame(GRxWT.C = GRxWT.C$p,
                       GRxWT.Df = GRxWT.Df$p,
                       GRxWT.OE = GRxWT.OE$p,
                       GR.CxDf = GR.CxDf$p,
                       WT.CxDf = WT.CxDf$p,
                       GR.CxOE = GR.CxOE$p,
                       WT.CxOE = WT.CxOE$p,
                       GR.DfxWT.C = GR.DfxWT.C$p,
                       GR.OExWT.C = GR.OExWT.C$p)
row.names(metab.FDR) = row.names(GRxWT.C)

colnames(mean.metab) = as.character(genotypes[colnames(mean.metab)])

enzyme2KEGG = as.list(org.Dm.egENZYME2EG)
enzyme2KEGG$'3.6.1.77' = "35231"
enzyme2KEGG = enzyme2KEGG[names(enzyme2KEGG) != "`3.6.1.-`"]


```


```{r echo = F}

TKT.biplot = function(x, y, mean.data, main.title){
min.expression = 10

FDR.data = TKT.EdgeR[, c(x, y)]
x.genes = row.names(FDR.data[FDR.data[,x] < 0.05,])
y.genes = row.names(FDR.data[FDR.data[,y] < 0.05,])
xy.genes = intersect(x.genes, y.genes)
DEGs = unique(c(x.genes, y.genes))


FC.data = TKT.EdgeR.FC[DEGs, c(x, y)]
FDR.data = TKT.EdgeR[DEGs, c(x, y)]
mean.data = mean.data[DEGs, ]
  

  
  
data = data.frame(Symbol = GeneIDKey[DEGs, "Symbol"],
                  FBgn = DEGs,
                  FDR1 = FDR.data[, x],
                  FDR2 = FDR.data[, y],
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.cpm = mean.data[,1],
                  Control1.cpm = mean.data[,2],
                  Variable2.cpm = mean.data[,3],
                  Control2.cpm = mean.data[,4])
  row.names(data) = data$FBgn

  
  data[x.genes, 'Color'] = 'royalblue'
  data[y.genes, 'Color'] = 'firebrick' 
  data[xy.genes, 'Color'] = 'gold'
  
  data$size = log2(apply(mean.data, 1, max))
  data$size[data$size<3] = 1
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = 2.5*data$size
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Symbol:", data$Symbol,
                                  "\nFBgn:", data$FBgn,
                                  "\n ", colnames(FDR.data)[1], "FDR: ", round(data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "FDR: ", round(data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean CPM: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean CPM: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean CPM: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean CPM: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  return(list(fig, data))
}
```


```{r include = F, echo = F}
Subset.Enrichment = function(bg, geneset){
  
  sigs = geneset
  
  
  sigsKEY = GeneIDKey[sigs,]
  sigs = na.omit(sigsKEY$Symbol)
  bgKEY = GeneIDKey[bg,]
  bg = na.omit(bgKEY$Symbol)
  
  genes = setNames(rep(0, length(bg)), bg)
  genes[sigs] = 1
  pwf=nullp(genes,"dm3","geneSymbol")
  GO.wall=goseq(pwf,"dm3","geneSymbol")
  GO.wall$adjp=p.adjust(GO.wall$over_represented_pvalue,method="BH")
  row.names(GO.wall) = GO.wall$category
  KEGG=goseq(pwf,gene2cat=GenesInKegg)
  KEGG$adjp=p.adjust(KEGG$over_represented_pvalue,method="BH")
  row.names(KEGG)=substr(as.character(KEGG[,1]),6,13)
  KEGG$Name=KEGG.Names[row.names(KEGG),1]
  
  enrichment.data = list(GO.wall, KEGG)
  names(enrichment.data) = c("GO", "KEGG")
  return(enrichment.data)
}

```

```{r echo = F}

compoundandgenes = function(CompoundData, x, y, mean.data, Fx, My, Fmean, Mmean){
KEGG.ID = unlist(strsplit(CompoundData$compoundID, " / "))
KEGG.ID = unlist(strsplit(KEGG.ID, "-"))
  
symbols = schema(F)$traces$scatter$attributes$marker$symbol$values
symbols = symbols[c(3, 15, 27, 63, 39, 51, 159, 207, 75, 87, 99)]
i = 1
enzymes = keggGet(KEGG.ID[i])[[1]]$ENZYME
if(length(enzymes > 0)){
  enzymes = unlist(enzyme2KEGG[intersect(enzymes, names(enzyme2KEGG))])
  if(length(enzymes > 0)){
    enzymes = GeneIDKey[GeneIDKey$Name %in% enzymes,]
    enzymes$Compound = CompoundData[grep(KEGG.ID[i], CompoundData$compoundID),'Metab']
  }
}

if(length(KEGG.ID)>1){
  i = 2
  while(i <= length(KEGG.ID)){
    temp.enzymes = keggGet(KEGG.ID[i])[[1]]$ENZYME
    if(length(temp.enzymes > 0)){
      temp.enzymes = unlist(enzyme2KEGG[intersect(temp.enzymes, names(enzyme2KEGG))])
      if(length(temp.enzymes > 0)){
        temp.enzymes = GeneIDKey[GeneIDKey$Name %in% temp.enzymes,]
        temp.enzymes$Compound = CompoundData[grep(KEGG.ID[i], CompoundData$compoundID),'Metab']
        enzymes = rbind(enzymes, temp.enzymes)
      }
    }
    i = i + 1
  }
}

enzymes$symbol = symbols[as.numeric(factor(enzymes$Symbol))]

colnames(CompoundData)[1:2] = c('Molecule', 'ID')
bg = setNames(c('firebrick',"gold","darkgreen","dodgerblue", 'deeppink', 'grey', 'darkorange')[as.numeric(factor(CompoundData$Molecule))], factor(CompoundData$Molecule))
CompoundData$shape = 'hexagon'
#CompoundData$Color = bg[CompoundData$Molecule]
CompoundData$Color = 'grey'
CompoundData$size = 35
CompoundData$group = 'Metabolite'

enzymes = data.frame(Molecule = enzymes$Symbol,
                     ID = enzymes$FBgn,
                     FDR1 = TKT.EdgeR[enzymes$FBgn, Fx],
                     FDR2 = TKT.EdgeR[enzymes$FBgn, My],
                     FC1 = TKT.EdgeR.FC[enzymes$FBgn, Fx],
                     FC2 = TKT.EdgeR.FC[enzymes$FBgn, My],
                     Color = 'gold',
                     size = 20,
                     Variable1.mean = Fmean[enzymes$FBgn,1],
                     Control1.mean  = Fmean[enzymes$FBgn,2],
                     Variable2.mean = Mmean[enzymes$FBgn,1],
                     Control2.mean  = Mmean[enzymes$FBgn,2],
                     shape = enzymes$symbol,
                     group = 'Enzyme')

GeneData = na.omit(enzymes)

metabs = data.frame(Molecule = CompoundData$Molecule,
                     ID = CompoundData$ID,
                     FDR1 = CompoundData$FDR,
                     FDR2 = CompoundData$FDR,
                     FC1 = CompoundData$FC,
                     FC2 = CompoundData$FC,
                     Color = CompoundData$Color,
                     size = CompoundData$size,
                     Variable1.mean = CompoundData$Variable1.mean,
                     Control1.mean  = CompoundData$Control1.mean,
                     Variable2.mean = CompoundData$Variable1.mean,
                     Control2.mean  = CompoundData$Variable1.mean,
                     shape = 'hexagon',
                     group = 'Metabolite')

CompoundData = na.omit(metabs)

data = rbind(CompoundData, GeneData)
      
 

  fig = plot_ly(type = 'scatter',
                mode = 'markers')
  fig = fig %>% add_trace(x = enzymes$FC1,
                          y = enzymes$FC2,
                          name = 'Genes',
                          marker = list(
                            color = enzymes$Color,
                            size = enzymes$size,
                            symbol = enzymes$shape,
                            line = list(
                              color = 'black',
                              width = 2
                            )
                          ),
                hoverinfo = "text",
                hovertext = paste("Gene Symbol:", enzymes$Molecule,
                                  "\nCompound ID:", enzymes$ID,
                                  "\n", Fx, "p: ", round(2^-enzymes$FDR1, 2),
                                  "\n", My, "p: ", round(2^-enzymes$FDR2,2),
                                  "\n", Fx, "FC: ", round(enzymes$FC1, 2),
                                  "\n", My, "FC: ", round(enzymes$FC2,2),
                                  "\n", Fx, "mean: ", signif(enzymes$Variable1,2),
                                  "\n ", My, "mean: ", signif(enzymes$Control1,2)))
  
    
    
      fig = fig %>% add_trace(x = CompoundData$FC1,
                          y = CompoundData$FC2,
                          name = 'Metabolites',
                          marker = list(
                            color = CompoundData$Color,
                            size = CompoundData$size,
                            symbol = CompoundData$shape,
                            line = list(
                              color = 'black',
                              width = 2
                            )
                          ),
                hoverinfo = "text",
                hovertext = paste("Gene Symbol:", CompoundData$Molecule,
                                  "\nCompound ID:", CompoundData$ID,
                                  "\nSex: Male",
                                  "\n", x, "p: ", round(2^-CompoundData$FDR1, 2),
                                  "\n", y, "p: ", round(2^-CompoundData$FDR2,2),
                                  "\n", x, "FC: ", round(CompoundData$FC1, 2),
                                  "\n", y, "FC: ", round(CompoundData$FC2,2),
                                  "\n", colnames(mean.data)[1], "mean: ", signif(CompoundData$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean: ", signif(CompoundData$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean: ", signif(CompoundData$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean: ", signif(CompoundData$Control2,2)))
                          
        fig = fig %>% layout(title = data$Molecule[1],
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(Fx, "FC")),
                       yaxis = list(title = paste(My, "FC")),
                       showlegend = T)
  
   fig
   
}
```


```{r echo = F}

#TKT = FBgn0036784

FBgn = "FBgn0036784"
Name = GeneIDKey[FBgn, "Symbol"]
TKT = GeneIDKey[FBgn, "Name"]
enzymeID = names(enzyme2KEGG)[grep(TKT, enzyme2KEGG)]
reactions <- keggGet(c(enzymeID))

temp.substrate = reactions[[1]]["SUBSTRATE"][[1]]
temp.substrate = strsplit(temp.substrate, split = ":")
substrates = c(1:length(temp.substrate))
i=1
while(i<=length(temp.substrate)){
  substrates[i] = substring(temp.substrate[[i]][2], 1,6)
  names(substrates)[i] = substring(temp.substrate[[i]][1], 1, (nchar(temp.substrate[[i]][1]) - 4))
  i=i+1
}


temp.product = reactions[[1]]["PRODUCT"][[1]]
temp.product = strsplit(temp.product, split = ":")
products = c(1:length(temp.product))
i=1
while(i<=length(temp.product)){
  products[i] = substring(temp.product[[i]][2], 1,6)
  names(products)[i] = substring(temp.product[[i]][1], 1, (nchar(temp.product[[i]][1]) - 4))
  i=i+1
}

TKTmap = data.frame(from.Name = c(names(substrates), rep(Name, length(products))),
                    from.ID = c(substrates, rep(TKT, length(products))),
                    to.Name = c(rep(Name, length(substrates)), names(products)),
                    to.ID = c(rep(TKT, length(substrates)), products))


secondary = c(substrates, products)

secondary.enzymes = as.character()

i=1
while( i<= length(secondary)){
  temp.enzymes = keggGet(secondary[i])[[1]]["ENZYME"][[1]]
  temp.enzymes = unlist(enzyme2KEGG[intersect(temp.enzymes, names(enzyme2KEGG))])
  secondary.enzymes = c(secondary.enzymes, temp.enzymes)
  i = i +1
}

secondary.enzymes = unique(secondary.enzymes)

secondary.enzymes = row.names(GeneIDKey[GeneIDKey$Name %in% unlist(secondary.enzymes),])

tertiary.enzymes = as.character()

j = 1

while(j <= length(secondary.enzymes)){
  FBgn = secondary.enzymes[j]
  Name = GeneIDKey[FBgn, "Symbol"]
  ID = GeneIDKey[FBgn, "Name"]
  enzymeID = names(enzyme2KEGG)[grep(ID, enzyme2KEGG)]
  reactions <- keggGet(c(enzymeID))
  
  temp.substrate = reactions[[1]]["SUBSTRATE"][[1]]
  temp.substrate = strsplit(temp.substrate, split = ":")
  substrates = c(1:length(temp.substrate))
  i=1
  while(i<=length(temp.substrate)){
    substrates[i] = substring(temp.substrate[[i]][2], 1,6)
    names(substrates)[i] = substring(temp.substrate[[i]][1], 1,
                                     (nchar(temp.substrate[[i]][1]) - 4))
    i=i+1
  }
  
  temp.product = reactions[[1]]["PRODUCT"][[1]]
  temp.product = strsplit(temp.product, split = ":")
  products = c(1:length(temp.product))
  i=1
  while(i<=length(temp.product)){
    products[i] = substring(temp.product[[i]][2], 1,6)
    names(products)[i] = substring(temp.product[[i]][1], 1, (nchar(temp.product[[i]][1]) - 4))
    i=i+1
  }
  
  temp.map = data.frame(from.Name = c(names(substrates), rep(Name, length(products))),
                        from.ID = c(substrates, rep(ID, length(products))),
                        to.Name = c(rep(Name, length(substrates)), names(products)),
                        to.ID = c(rep(ID, length(substrates)), products))
  row.names(temp.map) =  1:nrow(temp.map)
  TKTmap = rbind(TKTmap, temp.map)
  tertiary.enzymes = c(tertiary.enzymes, substrates, products)
  
  j=j+1
  }



```


```{r echo = F}

visedges = TKTmap[, c("from.Name", "to.Name")]
colnames(visedges) = c("from", "to")

temp = colorRampPalette(c('royalblue',"dodgerblue",'red', "firebrick"))(401)

visedges$color = round(TKT.EdgeR.FC[purine.path$FBgn, comparison.T],2)
##visedges$width = .5 + abs(visedges$color)
visedges$color[visedges$color < (-2)] = -2
visedges$color[visedges$color > (2)] = 2
visedges$color = (visedges$color * 100) +201
visedges$color = temp[visedges$color]
visedges$symbol = GeneIDKey[visedges$FBgn, 'Symbol']
visedges$fdr = signif(TKT.EdgeR[visedges$FBgn, comparison.T], 2)
visedges$GR = round(mean.cpm[visedges$FBgn, cond1],1)
visedges$WT = round(mean.cpm[visedges$FBgn, cond2],1)
visedges$width = .01 + (1*abs(log10(visedges$WT)))

groups = setNames(c(TKTmap$from.ID, TKTmap$to.ID),
                  c(TKTmap$from.Name, TKTmap$to.Name))
groups[substr(groups, 1,1) == "C"] = "metabolite"
groups[groups != "metabolite"] = "enzyme"
groups['CG8036'] = "TKT"

nodes = data.frame(id = unique(c(TKTmap$from.Name, TKTmap$to.Name)),
                   label = unique(c(TKTmap$from.Name, TKTmap$to.Name)),
                   group = groups[unique(c(TKTmap$from.Name, TKTmap$to.Name))])

nodes$label[nodes$label == "CG8036"] = "TKT"

nodes$color = round(TKT.EdgeR.FC[p, comparison.T],2)
##visedges$width = .5 + abs(visedges$color)
visedges$color[visedges$color < (-2)] = -2
visedges$color[visedges$color > (2)] = 2
visedges$color = (visedges$color * 100) +201
visedges$color = temp[visedges$color]
visedges$symbol = GeneIDKey[visedges$FBgn, 'Symbol']
visedges$fdr = signif(TKT.EdgeR[visedges$FBgn, comparison.T], 2)
visedges$GR = round(mean.cpm[visedges$FBgn, cond1],1)
visedges$WT = round(mean.cpm[visedges$FBgn, cond2],1)
visedges$width = .01 + (1*abs(log10(visedges$WT)))

visNetwork(nodes,visedges)%>%
  visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)%>%
  #visEdges(arrows = visedges$arrows)#%>%
  visGroups(groupname = "metabolite", 
            shape = 'box', color = "gold",
            physics=T, size = 25, 
            font=list(color="black"))%>%
  visGroups(groupname = "TKT", 
            shape = 'circle', color = "firebrick",
            physics=F, size = 25, 
            font=list(color="white"))



```

```{r}

PPP = keggGet('dme00030')[[1]]
PPP.enz = PPP$GENE
PPP.met = PPP$COMPOUND

download.kegg(pathway.id = "00030",
              species = "dme",
              kegg.dir = "/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/R Code/")

node.data = node.info('/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/R Code/dme00030.xml')

node.info = data.frame(ID = unlist(node.data$kegg.names, use.names = F),
                       number = rep(names(node.data$kegg.names),
                                    lengths(node.data$kegg.names)),
                       x = 0,
                       y = 0,
                       color = 0,
                       names = "")
node.info$x = node.data$x[node.info$number]
node.info$y = node.data$y[node.info$number]

node.info = node.info[grep("C", node.info$ID),]
node.info = node.info[grep("CG", node.info$ID, invert = T),]

enzymes = as.character()

i=1
while( i<= nrow(node.info)){
  temp.enzymes = keggGet(node.info$ID[i])[[1]]["ENZYME"][[1]]
  temp.enzymes = unlist(enzyme2KEGG[intersect(temp.enzymes, names(enzyme2KEGG))])
  enzymes = c(enzymes, temp.enzymes)
  i = i +1
}


edges = data.frame(from.Name = as.character(),
                   from.ID = as.character(),
                   to.Name = as.character(),
                   to.ID = as.character())

j = 1

while(j <= length(enzymes)){
  FBgn = na.omit(GeneIDKey[GeneIDKey$Name == enzymes[j], "FBgn"])
  Name = GeneIDKey[FBgn, "Symbol"]
  ID = GeneIDKey[FBgn, "Name"]
  enzymeID = names(enzymes)[i]
  reactions <- keggGet(c(enzymeID))
  
  temp.substrate = reactions[[1]]["SUBSTRATE"][[1]]
  temp.substrate = strsplit(temp.substrate, split = ":")
  substrates = c(1:length(temp.substrate))
  i=1
  while(i<=length(temp.substrate)){
    substrates[i] = substring(temp.substrate[[i]][2], 1,6)
    names(substrates)[i] = substring(temp.substrate[[i]][1], 1,
                                     (nchar(temp.substrate[[i]][1]) - 4))
    i=i+1
  }
  
  temp.product = reactions[[1]]["PRODUCT"][[1]]
  temp.product = strsplit(temp.product, split = ":")
  products = c(1:length(temp.product))
  i=1
  while(i<=length(temp.product)){
    products[i] = substring(temp.product[[i]][2], 1,6)
    names(products)[i] = substring(temp.product[[i]][1], 1, (nchar(temp.product[[i]][1]) - 4))
    i=i+1
  }
  
  temp.edges = data.frame(from.Name = c(names(substrates), rep(Name, length(products))),
                          from.ID = c(substrates, rep(ID, length(products))),
                          to.Name = c(rep(Name, length(substrates)), names(products)),
                          to.ID = c(rep(ID, length(substrates)), products))
  row.names(temp.edges) =  1:nrow(temp.edges)
  edges = rbind(edges, temp.edges)
  j=j+1
  }

edges = unique(edges)

edges = rbind(edges[edges$from.ID %in% node.info$ID, ],
              edges[edges$to.ID %in% node.info$ID, ])

```



```{r}

visedges = edges[, c("from.Name", "to.Name")]
colnames(visedges) = c("from", "to")

nodes = data.frame(id = unique(node.info$ID))


visNetwork(nodes,visedges)

```



