---
title: "G85R Analysis: Reversal Genes"
author: "John Santiago"
date: "2024-08-19"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r include = F, echo = F}

library(plotly)
library(edgeR)
library(heatmaply)
library(org.Dm.eg.db)
library(KEGGREST)
library(RcisTarget)
library(DT)
source('/Users/johncsantiago/Documents/GitHub/WhartonLab/GOandKEGGPlottingFunctions.R')

##load all files from github
git.dir = "https://raw.githubusercontent.com/johncsantiago/WhartonLab/master/"
cpmdata = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT_cpmdata.csv"),row.names = 1)
groups  = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.metadata.csv"),row.names = 1)
TKT.EdgeR = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FDRTable.csv"),row.names = 1)
TKT.EdgeR.FC = read.csv(paste0(git.dir,"TKT_RNAseq/CountTables/TKT.EdgeR.FCTable.csv"),row.names = 1)

G85R.meancpm = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT_meancpmdata.csv", row.names = 1)

groups = groups[colnames(cpmdata),]


GeneIDKey = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/GeneralDataFiles/GeneIDKey.csv", row.names = 1)

##Clean Trait Data and calculate CPMs
countdata=read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/CountTables/TKT.counttable.csv",row.names=1)


##normalize data
countdata=countdata[, row.names(groups)]
x <- countdata
gr <- factor(groups$Group)
y <- DGEList(counts=x,group=gr)
keep <- filterByExpr(y)
y <- y[keep,,keep.lib.sizes=FALSE]
z <- calcNormFactors(y, method = "TMM")
design<-model.matrix(~0 + gr)
colnames(design) = unique(gr)
z = estimateDisp(z, design, robust = T)
fit <- glmQLFit(z, design)

compare = makeContrasts(TktDfGR.F-WT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.FDfxWT.FC=DEout
sGR.FDfxWT.FC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.FDfxWT.FC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[17] = 'GR.FDfxWT.FC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.FDfxWT.FC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[17] = 'GR.FDfxWT.FC'

compare = makeContrasts(TktDfGR.M-WT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.MDfxWT.MC=DEout
sGR.MDfxWT.MC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.MDfxWT.MC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[18] = 'GR.MDfxWT.MC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.MDfxWT.MC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[18] = 'GR.MDfxWT.MC'




compare = makeContrasts(TktOEGR.F-WT.F, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.FOExWT.FC=DEout
sGR.FOExWT.FC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.FOExWT.FC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[19] = 'GR.FOExWT.FC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.FOExWT.FC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[19] = 'GR.FOExWT.FC'

compare = makeContrasts(TktOEGR.M-WT.M, levels=design)
lrt = glmQLFTest(fit,contrast=as.vector(compare))
G_X_E<-topTags(lrt,adjust.method="BH",n = nrow(z$counts), sort.by="PValue")
DEout=G_X_E$table
GR.MOExWT.MC=DEout
sGR.MOExWT.MC=DEout[DEout$FDR<.05,]

TKT.EdgeR = cbind(TKT.EdgeR, GR.MOExWT.MC[row.names(TKT.EdgeR), "FDR"])
colnames(TKT.EdgeR)[20] = 'GR.MOExWT.MC'

TKT.EdgeR.FC = cbind(TKT.EdgeR.FC, GR.MOExWT.MC[row.names(TKT.EdgeR), "logFC"])
colnames(TKT.EdgeR.FC)[20] = 'GR.MOExWT.MC'



```


##  Are there genes with 'corrected' expression in response to rescue?
```{r echo = F}

##comparisons
primary = "GRxWT.FC"
rescue = "GRF.CxDf"

x = "GRxWT.FC"
y = "GR.FDfxWT.FC"

mean.data = G85R.meancpm[, c('GR.F', 'WT.F',
                             'TktDfGR.F', 'TktDfWT.F')]

min.expression = 10

##Reversal pattern
FC.data = TKT.EdgeR.FC[, c(primary, rescue)]
FC.data = rbind(FC.data[FC.data[,primary] < 0 & FC.data[,rescue] > 0,],
                FC.data[FC.data[,primary] > 0 & FC.data[,rescue] < 0,])
reversals = row.names(FC.data)

##FC cutoff (currently a 25% increase or decrease in at least 1 condition)
FC.cutoff = rbind(FC.data[FC.data[,primary]  < -log2(1.25) & FC.data[,rescue] > log2(1.25),],
              FC.data[FC.data[,primary]  > log2(1.25) & FC.data[,rescue] < -log2(1.25),])
FC.cutoff = row.names(FC.cutoff)

##Sig cutoff. Currently, at least 1 significant change
sig.rescue = TKT.EdgeR[TKT.EdgeR[,primary]  < .05 |
                       TKT.EdgeR[,rescue] < .05, ]
sig.rescue = row.names(sig.rescue)

reversal.genes = intersect(intersect(reversals, FC.cutoff),sig.rescue)

FC.data = TKT.EdgeR.FC[reversal.genes, c(primary, rescue)]

  FC.data = TKT.EdgeR.FC[reversal.genes, c(x, y)]
  FDR.data = TKT.EdgeR[reversal.genes, c(x, y)]
  mean.data = mean.data[reversal.genes, ]
  

  
  
data = data.frame(Symbol = GeneIDKey[reversal.genes, "Symbol"],
                  FBgn = reversal.genes,
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.cpm = mean.data[,1],
                  Control1.cpm = mean.data[,2],
                  Variable2.cpm = mean.data[,3],
                  Control2.cpm = mean.data[,4])

  main.title = "Female TKT-Df Reversal Genes"
  
  data$Color[(data$FC1 - 1) > data$FC2] = 'royalblue'
  data$Color[(data$FC2 - 1) > data$FC1] = 'firebrick' 
  #data$Color[data$Symbol == "Arc1"] = 'gold'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = 5*data$size

  female.Df.reverse = data
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Symbol:", data$Symbol,
                                  "\nFBgn:", data$FBgn,
                                  "\n ", colnames(FDR.data)[1], "FDR: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "FDR: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean CPM: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean CPM: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean CPM: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean CPM: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  fig
```


```{r echo = F}

##comparisons
primary = "GRxWT.MC"
rescue = "GRM.CxDf"

x = "GRxWT.MC"
y = "GR.MDfxWT.MC"

mean.data = G85R.meancpm[, c('GR.M', 'WT.M',
                             'TktDfGR.M', 'TktDfWT.M')]

min.expression = 10

##Reversal pattern
FC.data = TKT.EdgeR.FC[, c(primary, rescue)]
FC.data = rbind(FC.data[FC.data[,primary] < 0 & FC.data[,rescue] > 0,],
                FC.data[FC.data[,primary] > 0 & FC.data[,rescue] < 0,])
reversals = row.names(FC.data)

##FC cutoff (currently a 25% increase or decrease in at least 1 condition)
FC.cutoff = rbind(FC.data[FC.data[,primary]  < -log2(1.25) & FC.data[,rescue] > log2(1.25),],
              FC.data[FC.data[,primary]  > log2(1.25) & FC.data[,rescue] < -log2(1.25),])
FC.cutoff = row.names(FC.cutoff)

##Sig cutoff. Currently, at least 1 significant change
sig.rescue = TKT.EdgeR[TKT.EdgeR[,primary]  < .05 |
                       TKT.EdgeR[,rescue] < .05, ]
sig.rescue = row.names(sig.rescue)

reversal.genes = intersect(intersect(reversals, FC.cutoff),sig.rescue)

FC.data = TKT.EdgeR.FC[reversal.genes, c(primary, rescue)]

  FC.data = TKT.EdgeR.FC[reversal.genes, c(x, y)]
  FDR.data = TKT.EdgeR[reversal.genes, c(x, y)]
  mean.data = mean.data[reversal.genes, ]
  

  
  
data = data.frame(Symbol = GeneIDKey[reversal.genes, "Symbol"],
                  FBgn = reversal.genes,
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.cpm = mean.data[,1],
                  Control1.cpm = mean.data[,2],
                  Variable2.cpm = mean.data[,3],
                  Control2.cpm = mean.data[,4])

  main.title = "Male TKT-Df Reversal Genes"
  
  data$Color[(data$FC1 - 1) > data$FC2] = 'royalblue'
  data$Color[(data$FC2 - 1) > data$FC1] = 'firebrick' 
  #data$Color[data$Symbol == "Arc1"] = 'gold'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = 5*data$size
  
  male.Df.reverse = data
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Symbol:", data$Symbol,
                                  "\nFBgn:", data$FBgn,
                                  "\n ", colnames(FDR.data)[1], "FDR: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "FDR: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean CPM: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean CPM: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean CPM: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean CPM: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  fig
```


```{r echo = F}

##comparisons
primary = "GRxWT.FC"
rescue = "GRF.CxOE"

x = "GRxWT.FC"
y = "GR.FOExWT.FC"

mean.data = G85R.meancpm[, c('GR.F', 'WT.F',
                             'TktOEGR.F', 'TktOEWT.F')]

min.expression = 10

##Reversal pattern
FC.data = TKT.EdgeR.FC[, c(primary, rescue)]
FC.data = rbind(FC.data[FC.data[,primary] < 0 & FC.data[,rescue] > 0,],
                FC.data[FC.data[,primary] > 0 & FC.data[,rescue] < 0,])
reversals = row.names(FC.data)

##FC cutoff (currently a 25% increase or decrease in at least 1 condition)
FC.cutoff = rbind(FC.data[FC.data[,primary]  < -log2(1.25) & FC.data[,rescue] > log2(1.25),],
              FC.data[FC.data[,primary]  > log2(1.25) & FC.data[,rescue] < -log2(1.25),])
FC.cutoff = row.names(FC.cutoff)

##Sig cutoff. Currently, at least 1 significant change
sig.rescue = TKT.EdgeR[TKT.EdgeR[,primary]  < .05 |
                       TKT.EdgeR[,rescue] < .05, ]
sig.rescue = row.names(sig.rescue)

reversal.genes = intersect(intersect(reversals, FC.cutoff),sig.rescue)

FC.data = TKT.EdgeR.FC[reversal.genes, c(primary, rescue)]

  FC.data = TKT.EdgeR.FC[reversal.genes, c(x, y)]
  FDR.data = TKT.EdgeR[reversal.genes, c(x, y)]
  mean.data = mean.data[reversal.genes, ]
  

  
  
data = data.frame(Symbol = GeneIDKey[reversal.genes, "Symbol"],
                  FBgn = reversal.genes,
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.cpm = mean.data[,1],
                  Control1.cpm = mean.data[,2],
                  Variable2.cpm = mean.data[,3],
                  Control2.cpm = mean.data[,4])

  main.title = "Female TKT-OE Reversal Genes"
  
  data$Color[(data$FC1 - 1) > data$FC2] = 'royalblue'
  data$Color[(data$FC2 - 1) > data$FC1] = 'firebrick' 
  #data$Color[data$Symbol == "Arc1"] = 'gold'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = 5*data$size

  female.OE.reverse = data
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Symbol:", data$Symbol,
                                  "\nFBgn:", data$FBgn,
                                  "\n ", colnames(FDR.data)[1], "FDR: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "FDR: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean CPM: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean CPM: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean CPM: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean CPM: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  fig
```


```{r echo = F}

##comparisons
primary = "GRxWT.MC"
rescue = "GRM.CxOE"

x = "GRxWT.MC"
y = "GR.MOExWT.MC"

mean.data = G85R.meancpm[, c('GR.M', 'WT.M',
                             'TktOEGR.M', 'TktOEWT.M')]

min.expression = 10

##Reversal pattern
FC.data = TKT.EdgeR.FC[, c(primary, rescue)]
FC.data = rbind(FC.data[FC.data[,primary] < 0 & FC.data[,rescue] > 0,],
                FC.data[FC.data[,primary] > 0 & FC.data[,rescue] < 0,])
reversals = row.names(FC.data)

##FC cutoff (currently a 25% increase or decrease in at least 1 condition)
FC.cutoff = rbind(FC.data[FC.data[,primary]  < -log2(1.25) & FC.data[,rescue] > log2(1.25),],
              FC.data[FC.data[,primary]  > log2(1.25) & FC.data[,rescue] < -log2(1.25),])
FC.cutoff = row.names(FC.cutoff)

##Sig cutoff. Currently, at least 1 significant change
sig.rescue = TKT.EdgeR[TKT.EdgeR[,primary]  < .05 |
                       TKT.EdgeR[,rescue] < .05, ]
sig.rescue = row.names(sig.rescue)

reversal.genes = intersect(intersect(reversals, FC.cutoff),sig.rescue)

FC.data = TKT.EdgeR.FC[reversal.genes, c(primary, rescue)]

  FC.data = TKT.EdgeR.FC[reversal.genes, c(x, y)]
  FDR.data = TKT.EdgeR[reversal.genes, c(x, y)]
  mean.data = mean.data[reversal.genes, ]
  

  
  
data = data.frame(Symbol = GeneIDKey[reversal.genes, "Symbol"],
                  FBgn = reversal.genes,
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.cpm = mean.data[,1],
                  Control1.cpm = mean.data[,2],
                  Variable2.cpm = mean.data[,3],
                  Control2.cpm = mean.data[,4])

  main.title = "Male TKT-OE Reversal Genes"
  
  data$Color[(data$FC1 - 1) > data$FC2] = 'royalblue'
  data$Color[(data$FC2 - 1) > data$FC1] = 'firebrick' 
  #data$Color[data$Symbol == "Arc1"] = 'gold'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = 5*data$size

  male.OE.reverse = data
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Symbol:", data$Symbol,
                                  "\nFBgn:", data$FBgn,
                                  "\n ", colnames(FDR.data)[1], "FDR: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "FDR: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean CPM: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean CPM: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean CPM: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean CPM: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  fig
```

##Looking for common reversal genes
```{r echo = F}
library(VennDiagram)

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Female Df"
set2="Male Df"
set3="Female OE"
set4="Male OE"
  
##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1=(female.Df.reverse$FBgn)
s2=(male.Df.reverse$FBgn)
s3=(female.OE.reverse$FBgn)
s4=(male.OE.reverse$FBgn)

main = "Reversal Genes"

grid.newpage()
tempvenn=draw.quad.venn(area1=length(s1),area2=length(s2),area3=length(s3),area4=length(s4), n12=length(intersect(s1,s2)),n13 = length(intersect(s1,s3)),n14 = length(intersect(s1,s4)),n23 = length(intersect(s2,s3)),n24 = length(intersect(s2,s4)),n34 = length(intersect(s3,s4)),n123 =  length(intersect(s1,intersect(s2,s3))),n124 =  length(intersect(s1,intersect(s2,s4))), n134 = length(intersect(s1,intersect(s3,s4))),n234 = length(intersect(s2,intersect(s3,s4))),n1234 = length(intersect(s1,intersect(s2,intersect(s3,s4)))),category =c(set1,set2,set3,set4),fill = c("blue", "red","green","yellow"),cex=2,cat.cex = 1, title = "main")

```



##The 19 common genes:


```{r echo = F}



G85R.gene = function(FBgn, name){
gene = data.frame(cpm = as.numeric(cpmdata[FBgn,]), condition = groups$Group, group = paste0(groups$Genotype, groups$Sex))

gene$group = factor(gene$group, levels = c(unique(gene$group)))
gene$condition = factor(gene$condition, levels = c('WT.F', "TktDfWT.F", "TktOEWT.F",
                                                   'GR.F', "TktDfGR.F", "TktOEGR.F",
                                                   'WT.M', "TktDfWT.M", "TktOEWT.M",
                                                   'GR.M', "TktDfGR.M", "TktOEGR.M"))

par(mar = c(5,7,5,2))
plot(x = NA,
      y = NA,
     type = 'n',
     ylim = c(min(gene$cpm), max(gene$cpm)),
     xlim = c(.5, 12.5),
     xaxt = "n",
     ylab="",
     yaxt = "n",
     xlab = NA)

rect(0, 0-(1.5*max(gene$cpm)), 13, 1.5*max(gene$cpm), col = 'gray95')

boxplot(gene$cpm~gene$condition, 
        xlab="",
        las = 2, 
        cex.axis = .8,
        xaxt = 'none',
        yaxt = 'none',
        boxwex = .75,
        boxlwd = 2,
        lwd = 2,
        cex.ticks = 2,
        add = T,
        col = c(rep("honeydew", 3),
                rep("thistle1", 3),
                rep("lightcyan", 3),
                rep("lightyellow", 3)))

box(lwd = 2)

points(x=as.numeric(factor(gene$condition)),
       y=gene$cpm,
       cex=1,
       pch=21,
       bg=c('firebrick',"gold","darkgreen","dodgerblue")[as.numeric(factor(gene$group))])

axis(side = 1,
     at = 1:12,
     labels = rep(c("Control", "Df", "OE"), 4),
     line = 0,
     cex.axis = .75,
     lwd.ticks = 2)

axis(side = 2,
     line = 0,
     cex.axis = 1,
     lwd.ticks = 2,
     las = 2)

axis(side = 2,
     at = .5*(max(gene$cpm) + min(gene$cpm)),
     label = "Counts per million reads",
     tick = F,
     padj = -3,
     cex.axis = 1.25)

axis(side = 1,
     at = c(2, 5, 8, 11),
     labels = c("Silent", "G85R","Silent", "G85R"),
     tick = F,
     padj = 2,
     cex.axis = 1)
axis(side = 3,
     at = c(3.5, 9.5),
     labels = c("Female", "Male"),
     tick = F,
     padj = .5,
     cex.axis = 1.5)

axis(side = 3,
     at = 6.5,
     labels = name,
     tick = F,
     padj = -1.5,
     cex.axis = 2)

axis(side = 3,
     at = c(3.5,9.5),
     tick = T,
     outer = T,
     labels = F,
     line = -2.25,
     lwd.ticks = 0)

xline1 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.04
xline2 = min(gene$cpm) - (max(gene$cpm)-min(gene$cpm))*.19
lines(x=c(3.5,3.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(6.5,6.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
lines(x=c(9.5,9.5), y = c(xline1, xline2), lty = 1, lwd = 2, xpd = T)
abline(v = 6.5, lty = 2, lwd = 2)

}

```


```{r echo = F}
common.symbol = GeneIDKey[intersect(s1,intersect(s2,intersect(s3,s4))),'Symbol']

common.FBgn = GeneIDKey[intersect(s1,intersect(s2,intersect(s3,s4))),'FBgn']


```


```{r echo = F, fig.width=8}
i=1
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##Cyp4p1; Cytochrome P450 4p1; May be involved in the metabolism of insect hormones and in the breakdown of synthetic insecticides.


```{r echo = F, fig.width=8}
i=2
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG13962; Unknown function


```{r echo = F, fig.width=8}
i=3
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##Cpr51A; Cuticular protein 51A; Predicted to be a structural constituent of chitin-based larval cuticle. Involved in sexual reproduction. Located in extracellular space.

```{r echo = F, fig.width=8}
i=4
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG11438; Predicted to enable phosphatidate phosphatase activity. Predicted to be involved in phospholipid dephosphorylation; phospholipid metabolic process; and signal transduction. Predicted to be active in plasma membrane.

```{r echo = F, fig.width=8}
i=5
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##Sgs5bis; Predicted to be involved in puparial adhesion. Predicted to be located in adhesive extracellular matrix.

```{r echo = F, fig.width=8}
i=6
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##sad; shadow; shadow (sad) encodes a cytochrome P450 involved in ecdysteroid biosynthesis. It shows a mitochondrial localization and catalyzes the addition of a hydroxyl group to the 2 carbon of the cholesterol ring. sad mutants fail to undergo head involution, dorsal closure or to secrete cuticle.

```{r echo = F, fig.width=8}
i=7
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##Lip2; Lipase 2; Predicted to enable triglyceride lipase activity. Predicted to be involved in lipid metabolic process. Human ortholog(s) of this gene implicated in Wolman disease; autosomal recessive congenital ichthyosis 8; and cholesterol ester storage disease.

```{r echo = F, fig.width=8}
i=8
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG4757; Enables serine hydrolase activity. Is expressed in adult head. Human ortholog(s) of this gene implicated in colorectal cancer and gastrointestinal system cancer.

```{r echo = F, fig.width=8}
i=9
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG14229; Is expressed in organism.

```{r echo = F, fig.width=8}
i=10
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG15406; Predicted to enable transmembrane transporter activity. Predicted to be involved in transmembrane transport. Predicted to be active in membrane.

```{r echo = F, fig.width=8}
i=11
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG3270; Predicted to be involved in mitochondrial respiratory chain complex I assembly. Predicted to be active in cytoplasm. Is expressed in adult head. Human ortholog(s) of this gene implicated in nuclear type mitochondrial complex I deficiency 19. 

```{r echo = F, fig.width=8}
i=12
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG9877; Unknown function

```{r echo = F, fig.width=8}
i=13
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG13905; mat; materazzi; Enables lipid binding activity. Is expressed in adult head.

```{r echo = F, fig.width=8}
i=14
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##Zip71B; Zinc/iron regulated transporter-related protein 71B; Zinc/iron regulated transporter-related protein 71B (Zip71B) encodes a protein involved in zinc detoxification and zinc transmembrane import.

```{r echo = F, fig.width=8}
i=15
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG17027; Predicted to enable inositol monophosphate 1-phosphatase activity. Predicted to be involved in inositol metabolic process and signal transduction. Predicted to be located in cytoplasm. Is expressed in embryonic Malpighian tubule and embryonic proventriculus.

```{r echo = F, fig.width=8}
i=16
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##nom; numerous disordered muscles; Enables DNA binding activity. Involved in larval somatic muscle development. Predicted to be located in nucleus. Is expressed in embryonic/larval fat body; organism; and ring gland primordium. 

```{r echo = F, fig.width=8}
i=17
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG16727; Predicted to enable transmembrane transporter activity. Predicted to be involved in transmembrane transport. Predicted to be located in membrane. Orthologous to human SLC22A24

```{r echo = F, fig.width=8}
i=18
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```


##CG9702; Predicted to enable sulfate transmembrane transporter activity. Predicted to be involved in transmembrane transport. Predicted to be located in membrane. Predicted to be active in plasma membrane. Is expressed in adult head. Orthologous to human SLC26A11

```{r echo = F, fig.width=8}
i=19
FBgn = common.FBgn[i]
name = paste0(common.FBgn[i], ": ", common.symbol[i])
G85R.gene(FBgn, name)
```

##CG43740; Located in nucleoplasm. Is expressed in adult brain cell body rind. 

```{r echo = F, include = F}

bg = row.names(G85R.cpm)
geneset = female.Df.reverse$FBgn

FDf.functions = Subset.Enrichment(bg, geneset)

```

## GO Term analysis of reversals in each condition

```{r echo = F, include = F}

RTCC.FDf = tree(FDf.functions, "CC")
treemapPlot(RTCC.FDf)

```


##Female DF CC


```{r echo = F}

treemapPlot(RTCC.FDf)

```


```{r echo = F, include = F}

RTBP.FDf = tree(FDf.functions, "BP")

```


##Female DF BP


```{r echo = F}

treemapPlot(RTBP.FDf)

```


```{r echo = F, include = F}

RTMF.FDf = tree(FDf.functions, "MF")

```


##Female DF MF


```{r echo = F}

treemapPlot(RTMF.FDf)

```


```{r echo = F, include = F}

bg = row.names(G85R.cpm)
geneset = male.Df.reverse$FBgn

MDf.functions = Subset.Enrichment(bg, geneset)

```


```{r echo = F, include = F}

RTCC.MDf = tree(MDf.functions, "CC")

```


##Male DF CC


```{r echo = F}

treemapPlot(RTCC.MDf)

```


```{r echo = F, include = F}

RTBP.MDf = tree(MDf.functions, "BP")

```


```{r echo = F}

Df.FC = setNames(G85R.FC[, 'GRF.CxDf'], row.names(G85R.FC))
Df.FDR = setNames(G85R.FDR[, 'GRF.CxDf'], row.names(G85R.FDR))
sigs = male.Df.reverse$FBgn

cat.data = MDf.functions
GO.ontology = 'CC'
FC = Df.FC
FDR = Df.FDR
usecats = 'sigs.only'

reversal.GO20(cat.data, GO.ontology, FC, FDR, usecats, sigs)

```


##Male DF BP


```{r echo = F}

treemapPlot(RTBP.MDf)

```


```{r echo = F}

Df.FC = setNames(G85R.FC[, 'GRF.CxDf'], row.names(G85R.FC))
Df.FDR = setNames(G85R.FDR[, 'GRF.CxDf'], row.names(G85R.FDR))
sigs = male.Df.reverse$FBgn

cat.data = MDf.functions
GO.ontology = 'BP'
FC = Df.FC
FDR = Df.FDR
usecats = 'sigs.only'

reversal.GO20(cat.data, GO.ontology, FC, FDR, usecats, sigs)

```


```{r echo = F, include = F}

RTMF.MDf = tree(MDf.functions, "MF")

```


##Male DF MF


```{r echo = F}

treemapPlot(RTMF.MDf)

```


```{r echo = F}

Df.FC = setNames(G85R.FC[, 'GRF.CxDf'], row.names(G85R.FC))
Df.FDR = setNames(G85R.FDR[, 'GRF.CxDf'], row.names(G85R.FDR))
sigs = male.Df.reverse$FBgn

cat.data = MDf.functions
GO.ontology = 'MF'
FC = Df.FC
FDR = Df.FDR
usecats = 'sigs.only'

reversal.GO20(cat.data, GO.ontology, FC, FDR, usecats, sigs)

```


```{r echo = F, include = F}

data("motifAnnotations_dmel_v9")
motifRankings = importRankings("/Users/johncsantiago/Documents/GitHub/WhartonLab/TKT_RNAseq/DataFiles/dm6_v10_clust.genes_vs_motifs.rankings.feather")
motifAnnotations = motifAnnotations_dmel_v9

##motif rankings table for every TF
motif.data = getRanking(motifRankings)
##motifs associated with a TF
##TF.data = motifAnnotations[motifAnnotations$TF == 'Tudor-SN',]
###TF.data = motifAnnotations[motifAnnotations$TF == 'Mad',]

##TF.motif.data = motif.data[motif.data$motifs %in% TF.data$motif,]

##Motif collection data directory
motif.collection.dir = "/Users/johncsantiago/Documents/GitHub/WhartonLab/DrosophilaMotifsSingletonData/"

geneLists = male.Df.reverse$Symbol

motifEnrichmentTable_wGenes <- cisTarget(geneLists, 
         motifRankings,
         motifAnnot=motifAnnotations)

motifs_AUC <- calcAUC(geneLists, motifRankings, nCores=1)

auc <- getAUC(motifs_AUC)
hist(auc, xlab="dAUC histogram",
     breaks=100, col="#ff000050", border="darkred")
nes3 <- (3*sd(auc)) + mean(auc)
abline(v=nes3, col="red")

motifEnrichmentTable <- addMotifAnnotation(motifs_AUC, nesThreshold=3,
     motifAnnot=motifAnnotations)

motifEnrichmentTable_wGenes <- addSignificantGenes(motifEnrichmentTable,
                      rankings=motifRankings, 
                      geneSets=geneLists)
dim(motifEnrichmentTable_wGenes)


resultsSubset <- motifEnrichmentTable_wGenes[1:3,]

```


```{r echo = F}

showLogo(resultsSubset)

```


```{r echo = F, include = F}

anotatedTfs <- lapply(split(motifEnrichmentTable_wGenes$TF_highConf,
                            motifEnrichmentTable$geneSet),
                      function(x) {
                        genes <- gsub(" \\(.*\\). ", "; ", x, fixed=FALSE)
                        genesSplit <- unique(unlist(strsplit(genes, "; ")))
                        return(genesSplit)
                        })
                      
anotatedTfs$hypoxia


```


```{r echo = F, include = F}

bg = row.names(G85R.cpm)
geneset = female.OE.reverse$FBgn

FOE.functions = Subset.Enrichment(bg, geneset)

```




```{r echo = F, include = F}

RTCC.FOE = tree(FOE.functions, "CC")

```


##Female OE CC


```{r echo = F}

treemapPlot(RTCC.FOE)

```



```{r echo = F, include = F}

RTBP.FOE = tree(FOE.functions, "BP")

```


##Female OE BP



```{r echo = F}

treemapPlot(RTBP.FOE)

```



```{r echo = F, include = F}

RTMF.FOE = tree(FOE.functions, "MF")

```


##Female OE MF



```{r echo = F}

treemapPlot(RTMF.FOE)

```



```{r echo = F, include = F}

bg = row.names(G85R.cpm)
geneset = male.OE.reverse$FBgn

MOE.functions = Subset.Enrichment(bg, geneset)

```



```{r echo = F, include = F}

RTCC.MOE = tree(MOE.functions, "CC")

```


##Male OE CC


```{r echo = F}

treemapPlot(RTCC.MOE)

```



```{r echo = F, include = F}

RTBP.MOE = tree(MOE.functions, "BP")

```


##Male OE BP


```{r echo = F}

treemapPlot(RTBP.MOE)

```



```{r echo = F, include = F}

RTMF.MOE = tree(MOE.functions, "MF")

```


##Male OE MF


```{r echo = F}

treemapPlot(RTMF.MOE)

```

##Common significantly enriched GO Terms in each condition

```{r echo = F}
library(VennDiagram)

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Female Df GO Terms"
set2="Male Df GO Terms"
set3="Female OE GO Terms"
set4="Male OE GO Terms"
  
##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1 = FDf.functions$GO[FDf.functions$GO$adjp<.05, 'term']
s2 = MDf.functions$GO[MDf.functions$GO$adjp<.05, 'term']
s3 = FOE.functions$GO[FOE.functions$GO$adjp<.05, 'term']
s4 = MOE.functions$GO[MOE.functions$GO$adjp<.05, 'term']

main = "Reversal Genes"

grid.newpage()
tempvenn=draw.quad.venn(area1=length(s1),area2=length(s2),area3=length(s3),area4=length(s4), n12=length(intersect(s1,s2)),n13 = length(intersect(s1,s3)),n14 = length(intersect(s1,s4)),n23 = length(intersect(s2,s3)),n24 = length(intersect(s2,s4)),n34 = length(intersect(s3,s4)),n123 =  length(intersect(s1,intersect(s2,s3))),n124 =  length(intersect(s1,intersect(s2,s4))), n134 = length(intersect(s1,intersect(s3,s4))),n234 = length(intersect(s2,intersect(s3,s4))),n1234 = length(intersect(s1,intersect(s2,intersect(s3,s4)))),category =c(set1,set2,set3,set4),fill = c("blue", "red","green","yellow"),cex=2,cat.cex = 1, title = "main")

```


##Significantly enriched GO Terms in Male Df 

```{r echo = F}
library(VennDiagram)

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Female Df GO Terms"
set2="Male Df GO Terms"
set3="Female OE GO Terms"
set4="Male OE GO Terms"
  
##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1 = FDf.functions$GO[1:37, 'term']
s2 = MDf.functions$GO[MDf.functions$GO$adjp<.05, 'term']
s3 = FOE.functions$GO[1:37, 'term']
s4 = MOE.functions$GO[1:37, 'term']

main = "Reversal Genes"

grid.newpage()
tempvenn=draw.quad.venn(area1=length(s1),area2=length(s2),area3=length(s3),area4=length(s4), n12=length(intersect(s1,s2)),n13 = length(intersect(s1,s3)),n14 = length(intersect(s1,s4)),n23 = length(intersect(s2,s3)),n24 = length(intersect(s2,s4)),n34 = length(intersect(s3,s4)),n123 =  length(intersect(s1,intersect(s2,s3))),n124 =  length(intersect(s1,intersect(s2,s4))), n134 = length(intersect(s1,intersect(s3,s4))),n234 = length(intersect(s2,intersect(s3,s4))),n1234 = length(intersect(s1,intersect(s2,intersect(s3,s4)))),category =c(set1,set2,set3,set4),fill = c("blue", "red","green","yellow"),cex=2,cat.cex = 1, title = "main")

```



```{r echo = F}

data = MDf.functions$GO[MDf.functions$GO$adjp<.05,]
data = data[data$numDEInCat < 100,]
data$adjp = -log10(data$adjp)
data$FDf = FDf.functions$GO[row.names(data), "numDEInCat"]
##data$MOE = MOE.functions$GO[row.names(data), "numDEInCat"]
##data$FOE = FOE.functions$GO[row.names(data), "numDEInCat"]


fig = plot_ly(data = data,
              x = ~numDEInCat,
              y = ~FDf,
              type = 'scatter',
              mode = 'markers',
              marker = list(colorscale = list(c(0,.5,1), c("brown","gold", "grey")),
                            reversescale =T,
                            colorbar = list(title = "-log10(FDR)",
                                            len = .8, outlinewidth = 0,
                                            tickformat = ".2f",
                                            tick0 = 0),##, dtick = .01),
                            color = ~adjp,
                            size = 15,
                            line = list(color = "black", width = 1.5)),
                hoverinfo = "text",
                hovertext = paste("Term:", data$term,
                                  "\n Ontology:", data$ontology,
                                  "\n Number in Cat: ", data$numDEInCat,
                                  "\n adjusted p: ", data$adjp))

  fig = fig %>% layout(title = "Male and Female Df",
                       xaxis = list(title = 'Male numDEInCat'),
                       yaxis = list(title = 'Female numDEInCat'),
                       showlegend = F)

  fig

```


```{r echo = F}

data = MDf.functions$GO[MDf.functions$GO$adjp<.05,]
data = data[data$numDEInCat < 100,]
data$adjp = -log10(data$adjp)
data$FDf = FDf.functions$GO[row.names(data), "numDEInCat"]
data$MOE = MOE.functions$GO[row.names(data), "numDEInCat"]
data$FOE = FOE.functions$GO[row.names(data), "numDEInCat"]

data$FDf.adjp = FDf.functions$GO[row.names(data), "adjp"]
data$MOE.adjp = MOE.functions$GO[row.names(data), "adjp"]
data$FOE.adjp = FOE.functions$GO[row.names(data), "adjp"]


fig = plot_ly(type = 'scatter',
              mode = 'markers')
fig = fig %>% add_trace(x = data$numDEInCat,
                        y = data$FDf,
                        name = 'Female Df',
                        marker = list(
                          color = 'steelblue',
                          size = 10,
                          symbol = 'circle',
                          line = list(
                            color = 'black',
                            width = 2
                            )
                          ),
                hoverinfo = "text",
                hovertext = paste("Term:", data$term,
                                  "\n Ontology:", data$ontology,
                                  "\n Number in Cat: ", data$FDf,
                                  "\n adjusted p: ", data$FDf.adjp))
fig = fig %>% add_trace(x = data$numDEInCat,
                        y = data$FOE,
                        name = 'Female OE',
                        marker = list(
                          color = 'gold',
                          size = 10,
                          symbol = 'hexagon',
                          line = list(
                            color = 'black',
                            width = 2
                            )
                          ),
                        hoverinfo = "text",
                        hovertext = paste("Term:", data$term,
                                  "\n Ontology:", data$ontology,
                                  "\n Number in Cat: ", data$FOE,
                                  "\n adjusted p: ", data$FOE.adjp))
fig = fig %>% add_trace(x = data$numDEInCat,
                        y = data$MOE,
                        name = 'Male OE',
                        marker = list(
                          color = 'firebrick',
                          size = 10,
                          symbol = 'square',
                          line = list(
                            color = 'black',
                            width = 2
                            )
                          ),
                        hoverinfo = "text",
                        hovertext = paste("Term:", data$term,
                                  "\n Ontology:", data$ontology,
                                  "\n Number in Cat: ", data$MOE,
                                  "\n adjusted p: ", data$MOE.adjp))
                          
        fig = fig %>% layout(title = data$Molecule[1],
                             xaxis = list(title = paste(x, "Male Df NumDEInCat")),
                             yaxis = list(title = paste(y, "Other NumDEInCat")),
                             showlegend = T)
  
   fig

```



##  Is there normally significant similarity in the rescue response and could the rescue phenotype be due to normal changes observed in response to TKT modulations.
```{r echo = F}

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Female Df"
set2="Male Df"
set3="Female OE"
set4="Male OE"
  
##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1 = row.names(TKT.EdgeR[TKT.EdgeR$WTF.CxDf < .05, ])
s2 = row.names(TKT.EdgeR[TKT.EdgeR$WTM.CxDf < .05, ])
s3 = row.names(TKT.EdgeR[TKT.EdgeR$WTF.CxOE < .05, ])
s4 = row.names(TKT.EdgeR[TKT.EdgeR$WTM.CxOE < .05, ])

s1 = G85R.meancpm[s1, c('WT.F', 'TktDfWT.F')]
s2 = G85R.meancpm[s2, c('WT.M', 'TktDfWT.M')]
s3 = G85R.meancpm[s3, c('WT.F', 'TktOEWT.F')]
s4 = G85R.meancpm[s4, c('WT.M', 'TktOEWT.M')]

s1 = row.names(s1[apply(s1,1,max)>10,])
s2 = row.names(s2[apply(s2,1,max)>10,])
s3 = row.names(s3[apply(s3,1,max)>10,])
s4 = row.names(s4[apply(s4,1,max)>10,])

NormalResponse = data.frame(Rescue = c("FemaleDf", "MaleDf", "FemaleOE", "MaleOE"),
                            TotalGenes =  c(length(s1), length(s2), length(s3), length(s4)))


fig <- plot_ly(data = NormalResponse, x = ~Rescue, y = ~TotalGenes, type = 'bar', name = 'Normal Response',
               marker = list(color = 'lightgrey',
                      line = list(color = 'rgb(8,48,107)',
                                  width = 3)))
fig <- fig %>% layout(yaxis = list(title = 'Total Response Genes in Silent-Rescue vs Silent'))

fig


```



##Overlap in Total Response Genes in Silent-Rescue vs Silent
```{r echo = F}

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Female Df"
set2="Male Df"
set3="Female OE"
set4="Male OE"
  
##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1 = row.names(TKT.EdgeR[TKT.EdgeR$WTF.CxDf < .05, ])
s2 = row.names(TKT.EdgeR[TKT.EdgeR$WTM.CxDf < .05, ])
s3 = row.names(TKT.EdgeR[TKT.EdgeR$WTF.CxOE < .05, ])
s4 = row.names(TKT.EdgeR[TKT.EdgeR$WTM.CxOE < .05, ])

s1 = G85R.meancpm[s1, c('WT.F', 'TktDfWT.F')]
s2 = G85R.meancpm[s2, c('WT.M', 'TktDfWT.M')]
s3 = G85R.meancpm[s3, c('WT.F', 'TktOEWT.F')]
s4 = G85R.meancpm[s4, c('WT.M', 'TktOEWT.M')]

s1 = row.names(s1[apply(s1,1,max)>10,])
s2 = row.names(s2[apply(s2,1,max)>10,])
s3 = row.names(s3[apply(s3,1,max)>10,])
s4 = row.names(s4[apply(s4,1,max)>10,])

CF.Df = s1
CM.Df = s2
CF.OE = s3
CM.OE = s4

main = "Total Response Genes in Silent-Rescue vs Silent"

grid.newpage()
tempvenn=draw.quad.venn(area1=length(s1),area2=length(s2),area3=length(s3),area4=length(s4), n12=length(intersect(s1,s2)),n13 = length(intersect(s1,s3)),n14 = length(intersect(s1,s4)),n23 = length(intersect(s2,s3)),n24 = length(intersect(s2,s4)),n34 = length(intersect(s3,s4)),n123 =  length(intersect(s1,intersect(s2,s3))),n124 =  length(intersect(s1,intersect(s2,s4))), n134 = length(intersect(s1,intersect(s3,s4))),n234 = length(intersect(s2,intersect(s3,s4))),n1234 = length(intersect(s1,intersect(s2,intersect(s3,s4)))),category =c(set1,set2,set3,set4),fill = c("blue", "red","green","yellow"),cex=2,cat.cex = 1, title = main)

```


##Genes that are Reversal Genes in the "Silent-Rescue vs Silent"
##Reversal genes that normally respond to the rescue
```{r echo = F}

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Female Df"
set2="Male Df"
set3="Female OE"
set4="Male OE"
  
##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1 = intersect(female.Df.reverse$FBgn, CF.Df)
s2 = intersect(male.Df.reverse$FBgn, CM.Df)
s3 = intersect(female.OE.reverse$FBgn, CF.OE)
s4 = intersect(male.OE.reverse$FBgn, CM.OE)

main = "Reversal Genes in the Normal Response Genes"

grid.newpage()
tempvenn=draw.quad.venn(area1=length(s1),area2=length(s2),area3=length(s3),area4=length(s4), n12=length(intersect(s1,s2)),n13 = length(intersect(s1,s3)),n14 = length(intersect(s1,s4)),n23 = length(intersect(s2,s3)),n24 = length(intersect(s2,s4)),n34 = length(intersect(s3,s4)),n123 =  length(intersect(s1,intersect(s2,s3))),n124 =  length(intersect(s1,intersect(s2,s4))), n134 = length(intersect(s1,intersect(s3,s4))),n234 = length(intersect(s2,intersect(s3,s4))),n1234 = length(intersect(s1,intersect(s2,intersect(s3,s4)))),category =c(set1,set2,set3,set4),fill = c("blue", "red","green","yellow"),cex=2,cat.cex = 1, title = "main")

```


##Identifying reversal genes that are also sDEG in WT 'rescue' condition

###Grey points should indicate a 'conserved response gene'. That is G85R-rescue with FC similar to the obeserved FC in WT-rescue. Same direction and similar magnitude (within 50% of FC). 
##Red and Blue indicates Reversal genes that are not following a 'conserved response gene'
##Gold are any 'conserved response genes' that are sDEG in WT Rescue
##Pink and Cyan are any Reversal genes that are not 'conserved response' but are sDEG in WT response

```{r echo = F}


conserved.reversalgenes = function(x, y, mean.data, reversal.genes, sWT, main.title){
FC.data = TKT.EdgeR.FC[reversal.genes, c(x, y)]
FDR.data = TKT.EdgeR[reversal.genes, c(x, y)]
mean.data = mean.data[reversal.genes, ]

data = data.frame(Symbol = GeneIDKey[reversal.genes, "Symbol"],
                  FBgn = reversal.genes,
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.cpm = mean.data[,1],
                  Control1.cpm = mean.data[,2],
                  Variable2.cpm = mean.data[,3],
                  Control2.cpm = mean.data[,4],
                  Conserved.genes = F)


  
  data$Color[(data$FC1 - (.5*abs(data$FC1))) > data$FC2] = 'royalblue'
  
  data$Color[data$FC1 > 0 & data$FC2 < 0] = 'royalblue'
  
  data$Color[(data$FC1 + (.5*abs(data$FC1))) < data$FC2] = 'firebrick' 
  data$Color[data$FC1 < 0 & data$FC2 > 0] = 'firebrick'
  data[data$Color == 'grey', 'Conserved.genes'] = T
  
  data[data$FBgn %in% sWT & data$Color == 'grey', 'Color'] = 'gold'
    data[data$FBgn %in% sWT & data$Color == 'royalblue', 'Color'] = 'cyan'
      data[data$FBgn %in% sWT & data$Color == 'firebrick', 'Color'] = 'deeppink'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data$size = 3*data$size

  Conserved.Fdf = data
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Symbol:", data$Symbol,
                                  "\nFBgn:", data$FBgn,
                                  "\n ", colnames(FDR.data)[1], "FDR: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "FDR: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean CPM: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean CPM: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean CPM: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean CPM: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  return(list(fig, data))
}
```


```{r echo = F}

##comparisons
x = "GRF.CxDf"
y = "WTF.CxDf"

mean.data = G85R.meancpm[, c('GR.F', 'TktDfGR.F',
                             'WT.F', 'TktDfWT.F')]

reversal.genes = female.Df.reverse$FBgn
sWT = CF.Df
main.title = "Female TKT-Df Reversal Genes"

Conserved.Fdf = conserved.reversalgenes(x, y, mean.data, reversal.genes, sWT, main.title)

Conserved.Fdf[[1]]

Conserved.Fdf = Conserved.Fdf[[2]]

```



```{r echo = F}

data = Conserved.Fdf

conserved = nrow(data[data$Color == 'grey',])
conserved.DEGs = nrow(data[data$Color == 'gold',])
altered.down = nrow(data[data$Color == 'deeppink',])
altered.up = nrow(data[data$Color == 'cyan',])

pie(c((nrow(data)-conserved), altered.down, altered.up, conserved.DEGs, conserved),
    col = c(rgb(158/255,202/255,225/255, .75), 'deeppink', 'cyan', 'gold', 'grey'),
    labels = NA)
legend("topright",
       legend = c("Altered Response", "Conserved Response", "Conserved DEGs", "Altered DEGs: Increased", "Altered DEGs: Decreased"),
       pt.bg = c(rgb(158/255,202/255,225/255, .75), 'grey', 'gold', 'cyan', 'deeppink'),
       bty = 'n',
       cex = .85,
       pch = 22,
       pt.cex = 1.5)

```


```{r echo = F}

##comparisons
x = "GRM.CxDf"
y = "WTM.CxDf"

mean.data = G85R.meancpm[, c('GR.M', 'TktDfGR.M',
                             'WT.M', 'TktDfWT.M')]

reversal.genes = male.Df.reverse$FBgn
sWT = CM.Df

main.title = "Male TKT-Df Reversal Genes"

Conserved.Mdf = conserved.reversalgenes(x, y, mean.data, reversal.genes, sWT, main.title)

Conserved.Mdf[[1]]

Conserved.Mdf = Conserved.Mdf[[2]]

```


```{r echo = F}

data = Conserved.Mdf

conserved = nrow(data[data$Color == 'grey',])
conserved.DEGs = nrow(data[data$Color == 'gold',])
altered.down = nrow(data[data$Color == 'deeppink',])
altered.up = nrow(data[data$Color == 'cyan',])

pie(c((nrow(data)-conserved), altered.down, altered.up, conserved.DEGs, conserved),
    col = c(rgb(158/255,202/255,225/255, .75), 'deeppink', 'cyan', 'gold', 'grey'),
    labels = NA)
legend("topright",
       legend = c("Altered Response", "Conserved Response", "Conserved DEGs", "Altered DEGs: Increased", "Altered DEGs: Decreased"),
       pt.bg = c(rgb(158/255,202/255,225/255, .75), 'grey', 'gold', 'cyan', 'deeppink'),
       bty = 'n',
       cex = .85,
       pch = 22,
       pt.cex = 1.5)

```


```{r echo = F}

##comparisons
x = "GRF.CxOE"
y = "WTF.CxOE"

mean.data = G85R.meancpm[, c('GR.F', 'TktOEGR.F',
                             'WT.F', 'TktOEWT.F')]

reversal.genes = female.OE.reverse$FBgn
sWT = CF.OE

main.title = "Female TKT-OE Reversal Genes"

Conserved.FOE = conserved.reversalgenes(x, y, mean.data, reversal.genes, sWT, main.title)

Conserved.FOE[[1]]

Conserved.FOE = Conserved.FOE[[2]]

```


```{r echo = F}

data = Conserved.FOE

conserved = nrow(data[data$Color == 'grey',])
conserved.DEGs = nrow(data[data$Color == 'gold',])
altered.down = nrow(data[data$Color == 'deeppink',])
altered.up = nrow(data[data$Color == 'cyan',])

pie(c((nrow(data)-conserved), altered.down, altered.up, conserved.DEGs, conserved),
    col = c(rgb(158/255,202/255,225/255, .75), 'deeppink', 'cyan', 'gold', 'grey'),
    labels = NA)
legend("topright",
       legend = c("Altered Response", "Conserved Response", "Conserved DEGs", "Altered DEGs: Increased", "Altered DEGs: Decreased"),
       pt.bg = c(rgb(158/255,202/255,225/255, .75), 'grey', 'gold', 'cyan', 'deeppink'),
       bty = 'n',
       cex = .85,
       pch = 22,
       pt.cex = 1.5)

```


```{r echo = F}

##comparisons
x = "GRM.CxOE"
y = "WTM.CxOE"

mean.data = G85R.meancpm[, c('GR.M', 'TktOEGR.M',
                             'WT.M', 'TktOEWT.M')]

reversal.genes = male.OE.reverse$FBgn
sWT = CM.OE

main.title = "Male TKT-OE Reversal Genes"

Conserved.MOE = conserved.reversalgenes(x, y, mean.data, reversal.genes, sWT, main.title)

Conserved.MOE[[1]]

Conserved.MOE = Conserved.MOE[[2]]

```


```{r echo = F}

data = Conserved.MOE

conserved = nrow(data[data$Color == 'grey',])
conserved.DEGs = nrow(data[data$Color == 'gold',])
altered.down = nrow(data[data$Color == 'deeppink',])
altered.up = nrow(data[data$Color == 'cyan',])

pie(c((nrow(data)-conserved), altered.down, altered.up, conserved.DEGs, conserved),
    col = c(rgb(158/255,202/255,225/255, .75), 'deeppink', 'cyan', 'gold', 'grey'),
    labels = NA)
legend("topright",
       legend = c("Altered Response", "Conserved Response", "Conserved DEGs", "Altered DEGs: Increased", "Altered DEGs: Decreased"),
       pt.bg = c(rgb(158/255,202/255,225/255, .75), 'grey', 'gold', 'cyan', 'deeppink'),
       bty = 'n',
       cex = .85,
       pch = 22,
       pt.cex = 1.5)

```

##Assessing the number of Conserved Response genes in the Reversal Genes

##Distinguishing Conserved Response genes between control DEGs and control sDEGs

```{r echo = F}

response = data.frame(Rescue = c("FemaleDf", "MaleDf", "FemaleOE", "MaleOE"),
                      NormalResponse =  c(nrow(Conserved.Fdf[Conserved.Fdf$Color == "grey",]),
                                          nrow(Conserved.Mdf[Conserved.Mdf$Color == "grey",]),
                                          nrow(Conserved.FOE[Conserved.FOE$Color == "grey",]),
                                          nrow(Conserved.MOE[Conserved.MOE$Color == "grey",])),
                      AlteredResponse = c(nrow(Conserved.Fdf[Conserved.Fdf$Color != "grey",]),
                                          nrow(Conserved.Mdf[Conserved.Mdf$Color != "grey",]),
                                          nrow(Conserved.FOE[Conserved.FOE$Color != "grey",]),
                                          nrow(Conserved.MOE[Conserved.MOE$Color != "grey",])),
                      WTResponse      = c(nrow(Conserved.Fdf[Conserved.Fdf$Color == "gold",]),
                                          nrow(Conserved.Mdf[Conserved.Mdf$Color == "gold",]),
                                          nrow(Conserved.FOE[Conserved.FOE$Color == "gold",]),
                                          nrow(Conserved.MOE[Conserved.MOE$Color == "gold",])))

fig <- plot_ly(data = response, x = ~Rescue, y = ~NormalResponse, type = 'bar', name = 'Normal Response',
               marker = list(color = 'lightgrey',
                      line = list(color = 'rgb(8,48,107)',
                                  width = 3)))
fig <- fig %>% add_trace(y = ~WTResponse, name = 'Silent DEG Response',
               marker = list(color = 'gold'))
fig <- fig %>% add_trace(y = ~AlteredResponse, name = 'Altered Response',
               marker = list(color = 'rgb(158,202,225)'))
fig <- fig %>% layout(yaxis = list(title = 'Total Reversal Response Genes'), barmode = 'stack')

fig

```


```{r echo = F}

response = data.frame(Rescue = c("FemaleDf", "MaleDf", "FemaleOE", "MaleOE"),
                      NormalResponse =  c(nrow(Conserved.Fdf[Conserved.Fdf$Color == "grey",]),
                                          nrow(Conserved.Mdf[Conserved.Mdf$Color == "grey",]),
                                          nrow(Conserved.FOE[Conserved.FOE$Color == "grey",]),
                                          nrow(Conserved.MOE[Conserved.MOE$Color == "grey",])),
                      AlteredResponse = c(nrow(Conserved.Fdf[Conserved.Fdf$Color != "grey",]),
                                          nrow(Conserved.Mdf[Conserved.Mdf$Color != "grey",]),
                                          nrow(Conserved.FOE[Conserved.FOE$Color != "grey",]),
                                          nrow(Conserved.MOE[Conserved.MOE$Color != "grey",])),
                      WTResponse      = c(nrow(Conserved.Fdf[Conserved.Fdf$Color == "gold",]),
                                          nrow(Conserved.Mdf[Conserved.Mdf$Color == "gold",]),
                                          nrow(Conserved.FOE[Conserved.FOE$Color == "gold",]),
                                          nrow(Conserved.MOE[Conserved.MOE$Color == "gold",])),
                      TotalResponse =   c(nrow(Conserved.Fdf),
                                          nrow(Conserved.Mdf),
                                          nrow(Conserved.FOE),
                                          nrow(Conserved.MOE)))

percent.response = response
percent.response$NormalResponse = 100*percent.response$NormalResponse/percent.response$TotalResponse
percent.response$WTResponse = 100*percent.response$WTResponse/percent.response$TotalResponse
percent.response$AlteredResponse = 100*percent.response$AlteredResponse/percent.response$TotalResponse
fig <- plot_ly(data = percent.response, x = ~Rescue, y = ~NormalResponse, type = 'bar', name = 'Normal Response',
               marker = list(color = 'lightgrey',
                      line = list(color = 'rgb(8,48,107)',
                                  width = 3)))
fig <- fig %>% add_trace(y = ~WTResponse, name = 'Silent DEG Response',
               marker = list(color = 'gold'))
fig <- fig %>% add_trace(y = ~AlteredResponse, name = 'Altered Response',
               marker = list(color = 'rgb(158,202,225)'))
fig <- fig %>% layout(yaxis = list(title = 'Percent of Total Reversal Response Genes'), barmode = 'stack')

fig

```




##Conserved Reversal Genes

##Can be sDEG or not in Control response

```{r echo = F}

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Female Df"
set2="Male Df"
set3="Female OE"
set4="Male OE"


##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1 = (Conserved.Fdf[Conserved.Fdf$Color == "grey" | Conserved.Fdf$Color == "gold", "FBgn"])
s2 = (Conserved.Mdf[Conserved.Mdf$Color == "grey" | Conserved.Mdf$Color == "gold", "FBgn"])
s3 = (Conserved.FOE[Conserved.FOE$Color == "grey" | Conserved.FOE$Color == "gold", "FBgn"])
s4 = (Conserved.MOE[Conserved.MOE$Color == "grey" | Conserved.MOE$Color == "gold", "FBgn"])

main = "Conserved Reversal Genes"

grid.newpage()
tempvenn=draw.quad.venn(area1=length(s1),area2=length(s2),area3=length(s3),area4=length(s4), n12=length(intersect(s1,s2)),n13 = length(intersect(s1,s3)),n14 = length(intersect(s1,s4)),n23 = length(intersect(s2,s3)),n24 = length(intersect(s2,s4)),n34 = length(intersect(s3,s4)),n123 =  length(intersect(s1,intersect(s2,s3))),n124 =  length(intersect(s1,intersect(s2,s4))), n134 = length(intersect(s1,intersect(s3,s4))),n234 = length(intersect(s2,intersect(s3,s4))),n1234 = length(intersect(s1,intersect(s2,intersect(s3,s4)))),category =c(set1,set2,set3,set4),fill = c("blue", "red","green","yellow"),cex=2,cat.cex = 1, title = "main")

```


##Common conserved reversal genes

##Is in all groups

##Cpr51A; Cuticular protein 51A; Predicted to be a structural constituent of chitin-based larval cuticle. Involved in sexual reproduction. Located in extracellular space.

##Is in male and female Df but not OE

##vir-1; virus-induced RNA 1; virus-induced RNA 1 (vir-1) is produced from a promoter that is induced by viral infection. It can be used as a marker of the induction of an antiviral response. Its expression is regulated by the Jak/STAT pathway.

##CG34136; Unknown function



##  Are there metabolites with 'corrected' expression in response to rescue and can this response be associated with related gene expression?

```{r echo = F}



rawdata = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/MetabolomicsForMetaboanalyst.txt", sep = "\t", row.names = 1)
KEGG.Key = read.csv("/Users/johncsantiago/Documents/GitHub/WhartonLab/Metabolomics/RawMetabolomicData.csv", row.names = 1)
KEGG.Key = setNames(KEGG.Key[-c(1:2), 2], row.names(KEGG.Key)[-c(1:2)])

genotypes = setNames(c("WT", "GR", "TktOEWT", "TktOEGR", "TktDfWT", "TktDfGR"), c("C", "E", "A", "F", "B", "D"))

metadata = data.frame(Sample = colnames(rawdata), 
                      Group = as.character(rawdata[1,]), 
                      TIC = as.numeric(rawdata[2,]), 
                      Genotypes = genotypes[as.character(rawdata[1,])])

row.names(metadata) = metadata$Sample

norm.func = function(data){
  nd = (as.numeric(data)/metadata$TIC)*1000
  return(nd)
}

norm.data = t(apply(rawdata[3:nrow(rawdata),], 1, norm.func))
colnames(norm.data) = colnames(rawdata)

group.mean=function(data){
  gm = mean(na.omit(data))
}

mean.data = matrix(0, nrow=nrow(norm.data), ncol = length(genotypes))
row.names(mean.data) = row.names(norm.data)
colnames(mean.data) = unique(metadata$Group)

i=1
while(i<=ncol(mean.data)){
  mean.data[,i] = apply(norm.data[,metadata[metadata$Group == colnames(mean.data)[i], "Sample"]], 1, group.mean)
  i = i+1
}

mean.metab = mean.data

compare.conditions = function(condition1, condition2){
  columns1 = metadata[metadata$Group == condition1, "Sample"]
  columns2 = metadata[metadata$Group == condition2, "Sample"]
  p=setNames(rep(NA, nrow(norm.data)), row.names(norm.data))
  i=1
  while(i<=length(p)){
    if(length(na.omit(norm.data[i,columns1]))>2 & 
       length(na.omit(norm.data[i,columns2]))>2){
      temp = t.test(na.omit(norm.data[i,columns1]),
                    na.omit(norm.data[i,columns2]), paired = F, var.equal = T)
      p[i] = temp[[3]]
      }
    i=i+1
    }

  fdr = p.adjust(p, "BH", length(p))

  fc = mean.metab[,condition1]/mean.metab[,condition2]

  comparison.table = data.frame(p = p, 
                                FDR = fdr, 
                                FC = fc, 
                                KEGG = KEGG.Key[row.names(mean.metab)])
  
  comparison.table = comparison.table[order(comparison.table$FDR),]
}

GRxWT.C = compare.conditions("E", "C")
GRxWT.Df = compare.conditions("D", "B")[row.names(GRxWT.C),]
GRxWT.OE = compare.conditions("F", "A")[row.names(GRxWT.C),]

GR.CxDf = compare.conditions("D", "E")[row.names(GRxWT.C),]
WT.CxDf = compare.conditions("B", "C")[row.names(GRxWT.C),]

GR.CxOE = compare.conditions("F", "E")[row.names(GRxWT.C),]
WT.CxOE = compare.conditions("A", "C")[row.names(GRxWT.C),]

GR.DfxWT.C = compare.conditions("D", "C")[row.names(GRxWT.C),]
GR.OExWT.C = compare.conditions("F", "C")[row.names(GRxWT.C),]



comparison.means = list(GRF.CxDf = c("TktDfGR.F","GR.F"), 
                 GRF.CxOE = c("TktOEGR.F","GR.F"),
                 WTF.CxDf = c("TktDfWT.F","WT.F"),
                 WTF.CxOE = c("TktOEWT.F","WT.F"),
                 GRxWT.FC = c("GR.F","WT.F"), 
                 GRxWT.FDf = c("TktDfGR.F","TktDfWT.F"),
                 GRxWT.FOE = c("TktOEGR.F","TktOEWT.F"), 
                 GRM.CxDf = c("TktDfGR.M","GR.M"), 
                 GRM.CxOE = c("TktOEGR.M","GR.M"),
                 WTM.CxDf = c("TktDfWT.M","WT.M"),
                 WTM.CxOE = c("TktOEWT.M","WT.M"),
                 GRxWT.MC = c("GR.M","WT.M"), 
                 GRxWT.MDf = c("TktDfGR.M","TktDfWT.M"),
                 GRxWT.MOE = c("TktOEGR.M","TktOEWT.M"),
                 GR.FxM = c("GR.F","GR.M"),
                 WT.FxM = c("WT.F","WT.M"))


metab.FC = data.frame('GRxWT.C' = GRxWT.C$FC,
                      'GRxWT.Df' = GRxWT.Df$FC,
                      'GRxWT.OE' = GRxWT.OE$FC,
                      'GR.CxDf' = GR.CxDf$FC,
                      'WT.CxDf' = WT.CxDf$FC,
                      'GR.CxOE' = GR.CxOE$FC,
                      'WT.CxOE' = WT.CxOE$FC,
                      'GR.DfxWT.C' = GR.DfxWT.C$FC,
                      'GR.OExWT.C' = GR.OExWT.C$FC)
row.names(metab.FC) = row.names(GRxWT.C)
metab.FC = log2(metab.FC)

metab.FDR = data.frame(GRxWT.C = GRxWT.C$p,
                       GRxWT.Df = GRxWT.Df$p,
                       GRxWT.OE = GRxWT.OE$p,
                       GR.CxDf = GR.CxDf$p,
                       WT.CxDf = WT.CxDf$p,
                       GR.CxOE = GR.CxOE$p,
                       WT.CxOE = WT.CxOE$p,
                       GR.DfxWT.C = GR.DfxWT.C$p,
                       GR.OExWT.C = GR.OExWT.C$p)
row.names(metab.FDR) = row.names(GRxWT.C)

colnames(mean.metab) = as.character(genotypes[colnames(mean.metab)])

enzyme2KEGG = as.list(org.Dm.egENZYME2EG)

```


```{r echo = F}

##comparisons
primary = 'GRxWT.C'
rescue  = 'GR.CxDf'

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

min.expression = 0

##Reversal pattern
FC.data = metab.FC[, c(primary, rescue)]

FC.data = rbind(FC.data[FC.data[,primary] < 0 & FC.data[,rescue] > 0,],
                FC.data[FC.data[,primary] > 0 & FC.data[,rescue] < 0,])
FC.data = na.omit(FC.data)
reversals = row.names(FC.data)

##FC cutoff (currently a 25% increase or decrease in at least 1 condition)
FC.cutoff = rbind(FC.data[FC.data[,primary]  < -log2(1.25) & FC.data[,rescue] > log2(1.25),],
              FC.data[FC.data[,primary]  > log2(1.25) & FC.data[,rescue] < -log2(1.25),])
FC.cutoff = row.names(FC.cutoff)
FC.cutoff = na.omit(FC.cutoff)

##Sig cutoff. Currently, at least 1 significant change
sig.rescue = metab.FDR[metab.FDR[,primary]  < .05 |
                       metab.FDR[,rescue] < .05, ]
sig.rescue = row.names(sig.rescue)
sig.rescue = na.omit(sig.rescue)

reversal.metabs = intersect(intersect(reversals, FC.cutoff),sig.rescue)

FC.data = metab.FC[reversal.metabs, c(primary, rescue)]

  FC.data = metab.FC[reversal.metabs, c(x, y)]
  FDR.data = metab.FDR[reversal.metabs, c(x, y)]
  mean.data = mean.data[reversal.metabs, ]
  

  
  
data = data.frame(Metab = reversal.metabs,
                  compoundID = KEGG.Key[reversal.metabs],
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.mean = mean.data[,1],
                  Control1.mean = mean.data[,2],
                  Variable2.mean = mean.data[,3],
                  Control2.mean = mean.data[,4])

  main.title = "TKT-Df Reversal Metabolites"
  
  data$Color[(data$FC1 - 1) > data$FC2] = 'royalblue'
  data$Color[(data$FC2 - 1) > data$FC1] = 'firebrick' 
  #data$Color[data$Symbol == "Arc1"] = 'gold'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = data$size

  Df.reverse.metabs = data

  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Metabolite:", data$Metab,
                                  "\n Compound ID:", data$compoundID,
                                  "\n ", colnames(FDR.data)[1], "p: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "p: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  fig
  

```


```{r echo = F}

compoundandgenes = function(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean){
KEGG.ID = unlist(strsplit(CompoundData$compoundID, " / "))
KEGG.ID = unlist(strsplit(KEGG.ID, "-"))
  
symbols = schema(F)$traces$scatter$attributes$marker$symbol$values
symbols = symbols[c(3, 15, 27, 63, 39, 51, 159, 207, 75, 87, 99)]
i = 1
enzymes = keggGet(KEGG.ID[i])[[1]]$ENZYME
if(length(enzymes > 0)){
  enzymes = unlist(enzyme2KEGG[intersect(enzymes, names(enzyme2KEGG))])
  if(length(enzymes > 0)){
    enzymes = GeneIDKey[GeneIDKey$Name %in% enzymes,]
    enzymes$Compound = CompoundData[grep(KEGG.ID[i], CompoundData$compoundID),'Metab']
  }
}

if(length(KEGG.ID)>1){
  i = 2
  while(i <= length(KEGG.ID)){
    temp.enzymes = keggGet(KEGG.ID[i])[[1]]$ENZYME
    if(length(temp.enzymes > 0)){
      temp.enzymes = unlist(enzyme2KEGG[intersect(temp.enzymes, names(enzyme2KEGG))])
      if(length(temp.enzymes > 0)){
        temp.enzymes = GeneIDKey[GeneIDKey$Name %in% temp.enzymes,]
        temp.enzymes$Compound = CompoundData[grep(KEGG.ID[i], CompoundData$compoundID),'Metab']
        enzymes = rbind(enzymes, temp.enzymes)
      }
    }
    i = i + 1
  }
}

enzymes$symbol = symbols[as.numeric(factor(enzymes$Symbol))]

colnames(CompoundData)[1:2] = c('Molecule', 'ID')
bg = setNames(c('firebrick',"gold","darkgreen","dodgerblue", 'deeppink', 'grey', 'darkorange')[as.numeric(factor(CompoundData$Molecule))], factor(CompoundData$Molecule))
CompoundData$shape = 'hexagon'
#CompoundData$Color = bg[CompoundData$Molecule]
CompoundData$Color = 'grey'
CompoundData$size = 35
CompoundData$group = 'Metabolite'

Fenzymes = data.frame(Molecule = enzymes$Symbol,
                      ID = enzymes$FBgn,
                      FDR1 = TKT.EdgeR[enzymes$FBgn, Fx],
                      FDR2 = TKT.EdgeR[enzymes$FBgn, Fy],
                      FC1 = TKT.EdgeR.FC[enzymes$FBgn, Fx],
                      FC2 = TKT.EdgeR.FC[enzymes$FBgn, Fy],
                      #Color = bg[enzymes$Compound],
                      Color = 'deeppink',
                      size = 20,
                      Variable1.mean = Fmean.data[enzymes$FBgn,1],
                      Control1.mean  = Fmean.data[enzymes$FBgn,2],
                      Variable2.mean = Fmean.data[enzymes$FBgn,3],
                      Control2.mean  = Fmean.data[enzymes$FBgn,4],
                      shape = enzymes$symbol,
                      group = 'Female Enzyme')
Fenzymes = na.omit(Fenzymes)

Menzymes = data.frame(Molecule = enzymes$Symbol,
                      ID = enzymes$FBgn,
                      FDR1 = TKT.EdgeR[enzymes$FBgn, Mx],
                      FDR2 = TKT.EdgeR[enzymes$FBgn, My],
                      FC1 = TKT.EdgeR.FC[enzymes$FBgn, Mx],
                      FC2 = TKT.EdgeR.FC[enzymes$FBgn, My],
                      #Color = bg[enzymes$Compound],
                      Color = 'dodgerblue',
                      size = 20,
                      Variable1.mean = Mmean.data[enzymes$FBgn,1],
                      Control1.mean  = Mmean.data[enzymes$FBgn,2],
                      Variable2.mean = Mmean.data[enzymes$FBgn,3],
                      Control2.mean  = Mmean.data[enzymes$FBgn,4],
                      shape = enzymes$symbol,
                      group = 'Male Enzyme')
Menzymes = na.omit(Menzymes)

GeneData = (rbind(Fenzymes, Menzymes))

data = rbind(CompoundData, GeneData)
      
 

  fig = plot_ly(type = 'scatter',
                mode = 'markers')
  fig = fig %>% add_trace(x = Fenzymes$FC1,
                          y = Fenzymes$FC2,
                          name = 'Female Genes',
                          marker = list(
                            color = Fenzymes$Color,
                            size = Fenzymes$size,
                            symbol = Fenzymes$shape,
                            line = list(
                              color = 'black',
                              width = 2
                            )
                          ),
                hoverinfo = "text",
                hovertext = paste("Gene Symbol:", Fenzymes$Molecule,
                                  "\nCompound ID:", Fenzymes$ID,
                                  "\nSex: Female",
                                  "\n", x, "p: ", round(2^-Fenzymes$FDR1, 2),
                                  "\n", y, "p: ", round(2^-Fenzymes$FDR2,2),
                                  "\n", x, "FC: ", round(Fenzymes$FC1, 2),
                                  "\n", y, "FC: ", round(Fenzymes$FC2,2),
                                  "\n", colnames(mean.data)[1], "mean: ", signif(Fenzymes$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean: ", signif(Fenzymes$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean: ", signif(Fenzymes$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean: ", signif(Fenzymes$Control2,2)))
  
    fig = fig %>% add_trace(x = Menzymes$FC1,
                          y = Menzymes$FC2,
                          name = 'Male Genes',
                          marker = list(
                            color = Menzymes$Color,
                            size = Menzymes$size,
                            symbol = Menzymes$shape,
                            line = list(
                              color = 'black',
                              width = 2
                            )
                          ),
                hoverinfo = "text",
                hovertext = paste("Gene Symbol:", Menzymes$Molecule,
                                  "\nCompound ID:", Menzymes$ID,
                                  "\nSex: Male",
                                  "\n", x, "p: ", round(2^-Menzymes$FDR1, 2),
                                  "\n", y, "p: ", round(2^-Menzymes$FDR2,2),
                                  "\n", x, "FC: ", round(Menzymes$FC1, 2),
                                  "\n", y, "FC: ", round(Menzymes$FC2,2),
                                  "\n", colnames(mean.data)[1], "mean: ", signif(Menzymes$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean: ", signif(Menzymes$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean: ", signif(Menzymes$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean: ", signif(Menzymes$Control2,2)))
    
      fig = fig %>% add_trace(x = CompoundData$FC1,
                          y = CompoundData$FC2,
                          name = 'Metabolites',
                          marker = list(
                            color = CompoundData$Color,
                            size = CompoundData$size,
                            symbol = CompoundData$shape,
                            line = list(
                              color = 'black',
                              width = 2
                            )
                          ),
                hoverinfo = "text",
                hovertext = paste("Gene Symbol:", CompoundData$Molecule,
                                  "\nCompound ID:", CompoundData$ID,
                                  "\nSex: Male",
                                  "\n", x, "p: ", round(2^-CompoundData$FDR1, 2),
                                  "\n", y, "p: ", round(2^-CompoundData$FDR2,2),
                                  "\n", x, "FC: ", round(CompoundData$FC1, 2),
                                  "\n", y, "FC: ", round(CompoundData$FC2,2),
                                  "\n", colnames(mean.data)[1], "mean: ", signif(CompoundData$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean: ", signif(CompoundData$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean: ", signif(CompoundData$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean: ", signif(CompoundData$Control2,2)))
                          
        fig = fig %>% layout(title = data$Molecule[1],
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(x, "FC")),
                       yaxis = list(title = paste(y, "FC")),
                       showlegend = T)
  
   fig
   
}
```



```{r echo = F}

CompoundData = Df.reverse.metabs[1,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```


```{r echo = F}

CompoundData = Df.reverse.metabs[2,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```

```{r echo = F}

CompoundData = Df.reverse.metabs[3,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```


```{r echo = F}

CompoundData = Df.reverse.metabs[4,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```


```{r echo = F}

CompoundData = Df.reverse.metabs[6,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```


```{r echo = F}

CompoundData = Df.reverse.metabs[7,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```


```{r echo = F}

CompoundData = Df.reverse.metabs[8,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[10,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```


```{r echo = F}

CompoundData = Df.reverse.metabs[14,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[15,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[16,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[17,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[18,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[20,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[21,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[22,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = Df.reverse.metabs[23,]

x = 'GRxWT.C'
y = 'GR.DfxWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktDfWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FDfxWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MDfxWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktDfGR.F', 'TktDfWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktDfGR.M', 'TktDfWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```



```{r echo = F}

##comparisons
primary = 'GRxWT.C'
rescue  = 'GR.CxOE'

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

min.expression = 0

##Reversal pattern
FC.data = metab.FC[, c(primary, rescue)]

FC.data = rbind(FC.data[FC.data[,primary] < 0 & FC.data[,rescue] > 0,],
                FC.data[FC.data[,primary] > 0 & FC.data[,rescue] < 0,])
FC.data = na.omit(FC.data)
reversals = row.names(FC.data)

##FC cutoff (currently a 25% increase or decrease in at least 1 condition)
FC.cutoff = rbind(FC.data[FC.data[,primary]  < -log2(1.25) & FC.data[,rescue] > log2(1.25),],
              FC.data[FC.data[,primary]  > log2(1.25) & FC.data[,rescue] < -log2(1.25),])
FC.cutoff = row.names(FC.cutoff)
FC.cutoff = na.omit(FC.cutoff)

##Sig cutoff. Currently, at least 1 significant change
sig.rescue = metab.FDR[metab.FDR[,primary]  < .05 |
                       metab.FDR[,rescue] < .05, ]
sig.rescue = row.names(sig.rescue)
sig.rescue = na.omit(sig.rescue)

reversal.metabs = intersect(intersect(reversals, FC.cutoff),sig.rescue)

FC.data = metab.FC[reversal.metabs, c(primary, rescue)]

  FC.data = metab.FC[reversal.metabs, c(x, y)]
  FDR.data = metab.FDR[reversal.metabs, c(x, y)]
  mean.data = mean.data[reversal.metabs, ]
  

  
  
data = data.frame(Metab = reversal.metabs,
                  compoundID = KEGG.Key[reversal.metabs],
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.mean = mean.data[,1],
                  Control1.mean = mean.data[,2],
                  Variable2.mean = mean.data[,3],
                  Control2.mean = mean.data[,4])

  main.title = "TKT-OE Reversal Metabolites"
  
  data$Color[(data$FC1 - 1) > data$FC2] = 'royalblue'
  data$Color[(data$FC2 - 1) > data$FC1] = 'firebrick' 
  #data$Color[data$Symbol == "Arc1"] = 'gold'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = data$size

 OE.reverse.metabs = data
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Metabolite:", data$Metab,
                                  "\n ", colnames(FDR.data)[1], "p: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "p: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  fig

```


```{r echo = F}

CompoundData = OE.reverse.metabs[1,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```



```{r echo = F}

CompoundData = OE.reverse.metabs[2,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[3,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[4,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[5,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```



```{r echo = F}

CompoundData = OE.reverse.metabs[7,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[8,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[9,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[10,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[11,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```




```{r echo = F}

CompoundData = OE.reverse.metabs[12,]

x = 'GRxWT.C'
y = 'GR.OExWT.C'

mean.data = mean.metab[, c('GR', 'WT', 'TktOEGR', 'TktOEWT')]

Fx = 'GRxWT.FC'
Fy = 'GR.FOExWT.FC'

Mx = 'GRxWT.MC'
My = 'GR.MOExWT.MC'

Fmean.data = G85R.meancpm[, c('GR.F', 'WT.F', 'TktOEGR.F', 'TktOEWT.F')]
Mmean.data = G85R.meancpm[, c('GR.M', 'WT.M', 'TktOEGR.M', 'TktOEWT.M')]

compoundandgenes(CompoundData, x, y, mean.data, Fx, Fy, Mx, My, Fmean, Mmean)

```


```{r echo = F}

##Sample titles as strings. Only fill in up to your number of selected categories
set1="Df"
set2="OE"
  
##items to be compared (ex: gene names) as a 1 dimensional vectors. Only give inputs up to your number of selected categories
s1 = row.names(Df.reverse.metabs)
s2 = row.names(OE.reverse.metabs)

main = "Reversal Metabolites"

  grid.newpage()
draw.pairwise.venn(area1=length(s1),area2=length(s2),cross.area=length(intersect(s1,s2)),fill = c("blue","red"),cex=1.5,cat.cex = 1,category = c(set1,set2), main = "main")


```


```{r echo = F}

##comparisons
primary = 'GR.CxDf'
rescue  = 'GR.CxOE'

x = 'GR.CxDf'
y = 'GR.CxOE'

mean.data = mean.metab[, c('GR', 'WT', 'TktDfGR', 'TktOEGR')]

min.expression = 0


reversal.metabs = intersect(row.names(Df.reverse.metabs), row.names(OE.reverse.metabs))

  FC.data = metab.FC[reversal.metabs, c(x, y)]
  FDR.data = metab.FDR[reversal.metabs, c(x, y)]
  mean.data = mean.data[reversal.metabs, ]
  

  
  
data = data.frame(Metab = reversal.metabs,
                  compoundID = KEGG.Key[reversal.metabs],
                  FDR1 = -log2(FDR.data[, x]),
                  FDR2 = -log2(FDR.data[, y]),
                  FC1 = FC.data[, x],
                  FC2 = FC.data[, y],
                  Color = 'grey',
                  size = 1,
                  Variable1.mean = mean.data[,1],
                  Control1.mean = mean.data[,2],
                  Variable2.mean = mean.data[,3],
                  Control2.mean = mean.data[,4])

  main.title = "TKT-Rescue Common Reversal Metabolites"
  
  data$Color[(data$FC1 - 1) > data$FC2] = 'royalblue'
  data$Color[(data$FC2 - 1) > data$FC1] = 'firebrick' 
  #data$Color[data$Symbol == "Arc1"] = 'gold'
  size = data.frame(G85R = apply(mean.data[,c(1,3)], 1, mean),
                    WT = mean.data[,c(2)])
  size = size + .001
  data$size = log2(apply(size, 1, max))
  data$size[data$size<3] = 1
  
  data = na.omit(data)
  data = data[data$size > log2(min.expression),]
  data$size = data$size

 OE.reverse.metabs = data
  
  fig = plot_ly(data = data,
                x = ~FC1,
                y = ~FC2,
                type = 'scatter',
                mode = 'markers',
                marker = list(color = ~Color,
                              colors = ~Color,
                              line = list(color = "black", width = 1.5),
                              size = ~size),
                hoverinfo = "text",
                hovertext = paste("Metabolite:", data$Metab,
                                  "\n ", colnames(FDR.data)[1], "p: ", round(2^-data$FDR1, 2),
                                  "\n ", colnames(FDR.data)[2], "p: ", round(2^-data$FDR2,2),
                                  "\n ", colnames(FC.data)[1], "FC: ", round(data$FC1, 2),
                                  "\n ", colnames(FC.data)[2], "FC: ", round(data$FC2,2),
                                  "\n ", colnames(mean.data)[1], "mean: ", signif(data$Variable1,2),
                                  "\n ", colnames(mean.data)[2], "mean: ", signif(data$Control1,2),
                                  "\n ", colnames(mean.data)[3], "mean: ", signif(data$Variable2,2),
                                  "\n ", colnames(mean.data)[4], "mean: ", signif(data$Control2,2)))

  fig = fig %>% layout(title = main.title,
                       shapes = list(
                         type = "line",
                         line = list(color = "black", dash = 'dash', width = 2),
                         x0 = min(data$FC1, data$FC2) - 1,
                         x1 = max(data$FC1, data$FC2) + 1,
                         y0 = min(data$FC1, data$FC2) - 1,
                         y1 = max(data$FC1, data$FC2) + 1),
                       xaxis = list(title = paste(colnames(FC.data)[1], "FC")),
                       yaxis = list(title = paste(colnames(FC.data)[2], "FC")),
                       showlegend = F)

  fig

```


##Common Reversal Metabolites:

##pyridoxal, D-sedoheptulose-7-phosphate, 2,3-bisphospho-D-glycerate, NAD+, hydroxyphenyllactic acid / HVA, adenine, urea



